<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/><title>动态粒子群优化</title><style>
/* cspell:disable-file */
/* webkit printing magic: print all background colors */
html {
	-webkit-print-color-adjust: exact;
}
* {
	box-sizing: border-box;
	-webkit-print-color-adjust: exact;
}

html,
body {
	margin: 0;
	padding: 0;
}
@media only screen {
	body {
		margin: 2em auto;
		max-width: 900px;
		color: rgb(55, 53, 47);
	}
}

body {
	line-height: 1.5;
	white-space: pre-wrap;
}

a,
a.visited {
	color: inherit;
	text-decoration: underline;
}

.pdf-relative-link-path {
	font-size: 80%;
	color: #444;
}

h1,
h2,
h3 {
	letter-spacing: -0.01em;
	line-height: 1.2;
	font-weight: 600;
	margin-bottom: 0;
}

.page-title {
	font-size: 2.5rem;
	font-weight: 700;
	margin-top: 0;
	margin-bottom: 0.75em;
}

h1 {
	font-size: 1.875rem;
	margin-top: 1.875rem;
}

h2 {
	font-size: 1.5rem;
	margin-top: 1.5rem;
}

h3 {
	font-size: 1.25rem;
	margin-top: 1.25rem;
}

.source {
	border: 1px solid #ddd;
	border-radius: 3px;
	padding: 1.5em;
	word-break: break-all;
}

.callout {
	border-radius: 3px;
	padding: 1rem;
}

figure {
	margin: 1.25em 0;
	page-break-inside: avoid;
}

figcaption {
	opacity: 0.5;
	font-size: 85%;
	margin-top: 0.5em;
}

mark {
	background-color: transparent;
}

.indented {
	padding-left: 1.5em;
}

hr {
	background: transparent;
	display: block;
	width: 100%;
	height: 1px;
	visibility: visible;
	border: none;
	border-bottom: 1px solid rgba(55, 53, 47, 0.09);
}

img {
	max-width: 100%;
}

@media only print {
	img {
		max-height: 100vh;
		object-fit: contain;
	}
}

@page {
	margin: 1in;
}

.collection-content {
	font-size: 0.875rem;
}

.column-list {
	display: flex;
	justify-content: space-between;
}

.column {
	padding: 0 1em;
}

.column:first-child {
	padding-left: 0;
}

.column:last-child {
	padding-right: 0;
}

.table_of_contents-item {
	display: block;
	font-size: 0.875rem;
	line-height: 1.3;
	padding: 0.125rem;
}

.table_of_contents-indent-1 {
	margin-left: 1.5rem;
}

.table_of_contents-indent-2 {
	margin-left: 3rem;
}

.table_of_contents-indent-3 {
	margin-left: 4.5rem;
}

.table_of_contents-link {
	text-decoration: none;
	opacity: 0.7;
	border-bottom: 1px solid rgba(55, 53, 47, 0.18);
}

table,
th,
td {
	border: 1px solid rgba(55, 53, 47, 0.09);
	border-collapse: collapse;
}

table {
	border-left: none;
	border-right: none;
}

th,
td {
	font-weight: normal;
	padding: 0.25em 0.5em;
	line-height: 1.5;
	min-height: 1.5em;
	text-align: left;
}

th {
	color: rgba(55, 53, 47, 0.6);
}

ol,
ul {
	margin: 0;
	margin-block-start: 0.6em;
	margin-block-end: 0.6em;
}

li > ol:first-child,
li > ul:first-child {
	margin-block-start: 0.6em;
}

ul > li {
	list-style: disc;
}

ul.to-do-list {
	text-indent: -1.7em;
}

ul.to-do-list > li {
	list-style: none;
}

.to-do-children-checked {
	text-decoration: line-through;
	opacity: 0.375;
}

ul.toggle > li {
	list-style: none;
}

ul {
	padding-inline-start: 1.7em;
}

ul > li {
	padding-left: 0.1em;
}

ol {
	padding-inline-start: 1.6em;
}

ol > li {
	padding-left: 0.2em;
}

.mono ol {
	padding-inline-start: 2em;
}

.mono ol > li {
	text-indent: -0.4em;
}

.toggle {
	padding-inline-start: 0em;
	list-style-type: none;
}

/* Indent toggle children */
.toggle > li > details {
	padding-left: 1.7em;
}

.toggle > li > details > summary {
	margin-left: -1.1em;
}

.selected-value {
	display: inline-block;
	padding: 0 0.5em;
	background: rgba(206, 205, 202, 0.5);
	border-radius: 3px;
	margin-right: 0.5em;
	margin-top: 0.3em;
	margin-bottom: 0.3em;
	white-space: nowrap;
}

.collection-title {
	display: inline-block;
	margin-right: 1em;
}

.simple-table {
	margin-top: 1em;
	font-size: 0.875rem;
	empty-cells: show;
}
.simple-table td {
	height: 29px;
	min-width: 120px;
}

.simple-table th {
	height: 29px;
	min-width: 120px;
}

.simple-table-header-color {
	background: rgb(247, 246, 243);
	color: black;
}
.simple-table-header {
	font-weight: 500;
}

time {
	opacity: 0.5;
}

.icon {
	display: inline-block;
	max-width: 1.2em;
	max-height: 1.2em;
	text-decoration: none;
	vertical-align: text-bottom;
	margin-right: 0.5em;
}

img.icon {
	border-radius: 3px;
}

.user-icon {
	width: 1.5em;
	height: 1.5em;
	border-radius: 100%;
	margin-right: 0.5rem;
}

.user-icon-inner {
	font-size: 0.8em;
}

.text-icon {
	border: 1px solid #000;
	text-align: center;
}

.page-cover-image {
	display: block;
	object-fit: cover;
	width: 100%;
	max-height: 30vh;
}

.page-header-icon {
	font-size: 3rem;
	margin-bottom: 1rem;
}

.page-header-icon-with-cover {
	margin-top: -0.72em;
	margin-left: 0.07em;
}

.page-header-icon img {
	border-radius: 3px;
}

.link-to-page {
	margin: 1em 0;
	padding: 0;
	border: none;
	font-weight: 500;
}

p > .user {
	opacity: 0.5;
}

td > .user,
td > time {
	white-space: nowrap;
}

input[type="checkbox"] {
	transform: scale(1.5);
	margin-right: 0.6em;
	vertical-align: middle;
}

p {
	margin-top: 0.5em;
	margin-bottom: 0.5em;
}

.image {
	border: none;
	margin: 1.5em 0;
	padding: 0;
	border-radius: 0;
	text-align: center;
}

.code,
code {
	background: rgba(135, 131, 120, 0.15);
	border-radius: 3px;
	padding: 0.2em 0.4em;
	border-radius: 3px;
	font-size: 85%;
	tab-size: 2;
}

code {
	color: #eb5757;
}

.code {
	padding: 1.5em 1em;
}

.code-wrap {
	white-space: pre-wrap;
	word-break: break-all;
}

.code > code {
	background: none;
	padding: 0;
	font-size: 100%;
	color: inherit;
}

blockquote {
	font-size: 1.25em;
	margin: 1em 0;
	padding-left: 1em;
	border-left: 3px solid rgb(55, 53, 47);
}

.bookmark {
	text-decoration: none;
	max-height: 8em;
	padding: 0;
	display: flex;
	width: 100%;
	align-items: stretch;
}

.bookmark-title {
	font-size: 0.85em;
	overflow: hidden;
	text-overflow: ellipsis;
	height: 1.75em;
	white-space: nowrap;
}

.bookmark-text {
	display: flex;
	flex-direction: column;
}

.bookmark-info {
	flex: 4 1 180px;
	padding: 12px 14px 14px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
}

.bookmark-image {
	width: 33%;
	flex: 1 1 180px;
	display: block;
	position: relative;
	object-fit: cover;
	border-radius: 1px;
}

.bookmark-description {
	color: rgba(55, 53, 47, 0.6);
	font-size: 0.75em;
	overflow: hidden;
	max-height: 4.5em;
	word-break: break-word;
}

.bookmark-href {
	font-size: 0.75em;
	margin-top: 0.25em;
}

.sans { font-family: ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol"; }
.code { font-family: "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace; }
.serif { font-family: Lyon-Text, Georgia, ui-serif, serif; }
.mono { font-family: iawriter-mono, Nitti, Menlo, Courier, monospace; }
.pdf .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK JP'; }
.pdf:lang(zh-CN) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK SC'; }
.pdf:lang(zh-TW) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK TC'; }
.pdf:lang(ko-KR) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK KR'; }
.pdf .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK JP'; }
.pdf:lang(zh-CN) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK SC'; }
.pdf:lang(zh-TW) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK TC'; }
.pdf:lang(ko-KR) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK KR'; }
.pdf .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK JP'; }
.pdf:lang(zh-CN) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK SC'; }
.pdf:lang(zh-TW) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK TC'; }
.pdf:lang(ko-KR) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK KR'; }
.pdf .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK JP'; }
.pdf:lang(zh-CN) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK SC'; }
.pdf:lang(zh-TW) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK TC'; }
.pdf:lang(ko-KR) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK KR'; }
.highlight-default {
	color: rgba(55, 53, 47, 1);
}
.highlight-gray {
	color: rgba(120, 119, 116, 1);
	fill: rgba(120, 119, 116, 1);
}
.highlight-brown {
	color: rgba(159, 107, 83, 1);
	fill: rgba(159, 107, 83, 1);
}
.highlight-orange {
	color: rgba(217, 115, 13, 1);
	fill: rgba(217, 115, 13, 1);
}
.highlight-yellow {
	color: rgba(203, 145, 47, 1);
	fill: rgba(203, 145, 47, 1);
}
.highlight-teal {
	color: rgba(68, 131, 97, 1);
	fill: rgba(68, 131, 97, 1);
}
.highlight-blue {
	color: rgba(51, 126, 169, 1);
	fill: rgba(51, 126, 169, 1);
}
.highlight-purple {
	color: rgba(144, 101, 176, 1);
	fill: rgba(144, 101, 176, 1);
}
.highlight-pink {
	color: rgba(193, 76, 138, 1);
	fill: rgba(193, 76, 138, 1);
}
.highlight-red {
	color: rgba(212, 76, 71, 1);
	fill: rgba(212, 76, 71, 1);
}
.highlight-gray_background {
	background: rgba(241, 241, 239, 1);
}
.highlight-brown_background {
	background: rgba(244, 238, 238, 1);
}
.highlight-orange_background {
	background: rgba(251, 236, 221, 1);
}
.highlight-yellow_background {
	background: rgba(251, 243, 219, 1);
}
.highlight-teal_background {
	background: rgba(237, 243, 236, 1);
}
.highlight-blue_background {
	background: rgba(231, 243, 248, 1);
}
.highlight-purple_background {
	background: rgba(244, 240, 247, 0.8);
}
.highlight-pink_background {
	background: rgba(249, 238, 243, 0.8);
}
.highlight-red_background {
	background: rgba(253, 235, 236, 1);
}
.block-color-default {
	color: inherit;
	fill: inherit;
}
.block-color-gray {
	color: rgba(120, 119, 116, 1);
	fill: rgba(120, 119, 116, 1);
}
.block-color-brown {
	color: rgba(159, 107, 83, 1);
	fill: rgba(159, 107, 83, 1);
}
.block-color-orange {
	color: rgba(217, 115, 13, 1);
	fill: rgba(217, 115, 13, 1);
}
.block-color-yellow {
	color: rgba(203, 145, 47, 1);
	fill: rgba(203, 145, 47, 1);
}
.block-color-teal {
	color: rgba(68, 131, 97, 1);
	fill: rgba(68, 131, 97, 1);
}
.block-color-blue {
	color: rgba(51, 126, 169, 1);
	fill: rgba(51, 126, 169, 1);
}
.block-color-purple {
	color: rgba(144, 101, 176, 1);
	fill: rgba(144, 101, 176, 1);
}
.block-color-pink {
	color: rgba(193, 76, 138, 1);
	fill: rgba(193, 76, 138, 1);
}
.block-color-red {
	color: rgba(212, 76, 71, 1);
	fill: rgba(212, 76, 71, 1);
}
.block-color-gray_background {
	background: rgba(241, 241, 239, 1);
}
.block-color-brown_background {
	background: rgba(244, 238, 238, 1);
}
.block-color-orange_background {
	background: rgba(251, 236, 221, 1);
}
.block-color-yellow_background {
	background: rgba(251, 243, 219, 1);
}
.block-color-teal_background {
	background: rgba(237, 243, 236, 1);
}
.block-color-blue_background {
	background: rgba(231, 243, 248, 1);
}
.block-color-purple_background {
	background: rgba(244, 240, 247, 0.8);
}
.block-color-pink_background {
	background: rgba(249, 238, 243, 0.8);
}
.block-color-red_background {
	background: rgba(253, 235, 236, 1);
}
.select-value-color-pink { background-color: rgba(245, 224, 233, 1); }
.select-value-color-purple { background-color: rgba(232, 222, 238, 1); }
.select-value-color-green { background-color: rgba(219, 237, 219, 1); }
.select-value-color-gray { background-color: rgba(227, 226, 224, 1); }
.select-value-color-opaquegray { background-color: rgba(255, 255, 255, 0.0375); }
.select-value-color-orange { background-color: rgba(250, 222, 201, 1); }
.select-value-color-brown { background-color: rgba(238, 224, 218, 1); }
.select-value-color-red { background-color: rgba(255, 226, 221, 1); }
.select-value-color-yellow { background-color: rgba(253, 236, 200, 1); }
.select-value-color-blue { background-color: rgba(211, 229, 239, 1); }

.checkbox {
	display: inline-flex;
	vertical-align: text-bottom;
	width: 16;
	height: 16;
	background-size: 16px;
	margin-left: 2px;
	margin-right: 5px;
}

.checkbox-on {
	background-image: url("data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2216%22%20height%3D%2216%22%20viewBox%3D%220%200%2016%2016%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%3Crect%20width%3D%2216%22%20height%3D%2216%22%20fill%3D%22%2358A9D7%22%2F%3E%0A%3Cpath%20d%3D%22M6.71429%2012.2852L14%204.9995L12.7143%203.71436L6.71429%209.71378L3.28571%206.2831L2%207.57092L6.71429%2012.2852Z%22%20fill%3D%22white%22%2F%3E%0A%3C%2Fsvg%3E");
}

.checkbox-off {
	background-image: url("data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2216%22%20height%3D%2216%22%20viewBox%3D%220%200%2016%2016%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%3Crect%20x%3D%220.75%22%20y%3D%220.75%22%20width%3D%2214.5%22%20height%3D%2214.5%22%20fill%3D%22white%22%20stroke%3D%22%2336352F%22%20stroke-width%3D%221.5%22%2F%3E%0A%3C%2Fsvg%3E");
}
	
</style></head><body><article id="8a0c985a-5a91-4a26-9ffa-8a06f5d751c9" class="page sans"><header><h1 class="page-title">动态粒子群优化</h1></header><div class="page-body"><h2 id="e849ab26-60b4-4267-ba50-d7d94837f5c6" class="">Dynamic particle swarm optimization of biomolecular simulation parameters with flexible objective functions</h2><blockquote id="3fa96e49-7641-41f1-8a32-b014895fbd15" class="">基于动态粒子群优化的具有灵活目标函数的生物分子的参数模拟</blockquote><p id="6bc3a33e-2b9f-4a07-ba84-b3739d1e58c3" class="">
</p><ul id="366ff7f1-9db6-481c-950d-d33cf273f842" class="bulleted-list"><li style="list-style-type:disc">M. Weiel, M. Gtz, K. André, D. Coquelin, and A. Schug, “Dynamic particle swarm optimization of biomolecular simulation parameters with flexible objective functions,” Nature Machine Intelligence.</li></ul><p id="a193c2b6-848d-4168-8e65-1f7c779a7201" class="">
</p><figure class="block-color-gray_background callout" style="white-space:pre-wrap;display:flex" id="1d1214b2-545e-4cf5-a7b0-4f26a0f08a63"><div style="font-size:1.5em"><span class="icon">💡</span></div><div style="width:100%">Molecular simulations are a powerful tool to complement and interpret ambiguous experimental data on biomolecules to obtain structural models. Such data-assisted simulations often rely on parameters, the choice of which is highly non-trivial and crucial to performance. The key challenge is weighting experimental information with respect to the underlying physical model. We introduce FLAPS, a self-adapting variant of dynamic particle swarm optimization, to overcome this parameter selection problem. FLAPS is suited for the optimization of composite objective functions that depend on both the optimization parameters and additional, a priori unknown weighting parameters, which substantially influence the search-space topology. These weighting parameters are learned at runtime, yielding a dynamically evolving and iteratively refined search-space topology. As a practical example, we show how FLAPS can be used to find functional parameters for small-angle X-ray scattering-guided protein simulations. <blockquote id="9aff5bee-8b6a-47d9-a245-87ffb937768b" class=""><a href="https://www.notion.so/5c90a76869334658b0f5078d15abb89b">分子模拟</a>是补充和解释生物分子模棱两可的实验数据以获得结构模型的有力工具。这种数据辅助模拟通常依赖于参数，参数的选择对性能至关重要。最大的挑战是根据基础物理模型对实验信息进行加权。我们引入了动态粒子群优化的自适应变体(flexible self-adapting particle swarm optimization,FLAPS)来克服这一参数选择的问题。FLAPS 适用于复合目标函数的优化，该目标函数依赖于优化参数和附加的先验未知加权参数，这些参数对搜索空间拓扑结构有很大影响。这些加权参数是在运行时学习的，从而产生动态演变和迭代进化的搜索空间拓扑。作为一个实际示例，我们展示了如何使用 FLAPS 来查找小角度 X 射线散射引导的蛋白质模拟的功能参数。</blockquote></div></figure><p id="b87021fe-dc9a-4f92-a1c0-8bce9e77516b" class="">Proteins are the molecular workhouses of biological cells, with a myriad of tasks including oxygen transport, cellular communication and energy balance. As a protein’s function is linked to its structure and dynamics, its understanding requires resolving the protein’s three-dimensional shape. Misfolded proteins are associated with several neurodegenerative diseases[1], and deciphering the structure–function paradigm paves the way to developing treatments for Alzheimer’s and Parkinson’s diseases[2], amyloidosis[3], type-2 diabetes[4], Creutzfeldt–Jakob disease[5], among others.</p><blockquote id="881dfb4a-9290-462f-9556-e9efeace3344" class="">蛋白质是生物细胞的分子主力军，承担着氧气运输、细胞通讯和能量平衡等众多工作。由于蛋白质的功能与其结构和动力学有关，因此理解蛋白质需要解析蛋白质的三维结构。错误折叠的蛋白质与几种神经退行性疾病有关，而破译这种结构-功能范式为开发治疗阿尔茨海默病[1]和帕金森病[2]、淀粉样变性病[3]、2型糖尿病[4]、克雅氏病[5]等疾病的治疗方法铺平了前进的道路。</blockquote><p id="64bdd7c1-2dca-4e99-867c-9db4dd34cdf4" class="">Proteins are nanoscale and can only be observed indirectly. Hence, experimental data are often ambiguous, incomplete or of such low resolution that they require interpretation to access their
information content. A practical example is small-angle X-ray scattering (SAXS), where dissolved biomolecules are irradiated by X-rays and the scattering intensity is recorded. However, the desired molecular electron density is the Fourier transform of the experimentally inaccessible complex valued scattering amplitude. Recovering structural models from an intensity, that is, the absolute amplitude squared, is an ill-posed inverse problem. The sparse information in the SAXS data is insufficient to determine all degrees of freedom in a molecular structure. Protein structure determination thus depends on combining experimental results and computational methods, and recent studies highlight the potential of such hybrid approaches[6–9].</p><blockquote id="29a55a5d-5d06-47c4-b99c-95c8196a5332" class="">蛋白质是纳米级的，只能间接观察。因此，实验数据通常是模棱两可的、不完整的或分辨率很低，以至于需要对其进行解释才能访问其内容信息。一个实际的例子是小角度 X 射线散射 (SAXS)，其中溶解的生物分子被 X 射线照射并记录散射强度。然而，所需的分子电子密度是实验上无法获得的复值散射幅度的傅里叶变换。从强度即绝对幅度平方中恢复结构模型是一个不适定的逆问题。 SAXS 数据中的稀疏信息不足以确定分子结构中的所有自由度。因此，蛋白质结构的确定取决于实验结果和计算方法的结合，最近的研究强调了这种混合方法的潜力[6-9]。</blockquote><p id="c76462b7-6fbc-438d-8b2d-3a0b9d226df7" class="">One effective approach is to complement the experimental data with molecular dynamics (MD)[10,11]. MD simulations provide a physics-based description of molecular motion and give in-depth
insight into biomolecular function. Atomistic trajectories are derived by integrating Newton’s equations of motion for a system of interacting particles, and forces are calculated from empirical interatomic potentials (force fields). In data-assisted MD, an energetic restraint on the target data is added to the force field to favour conformations consistent with the data. This bias potential is proportional to the least-squares deviation of theoretical data from simulated structures and the experimental data[7,10,12–14]. Derived forces are assumed to drive the protein towards conformations reproducing the target data. As a molecular system seeks to minimize its free energy, the bias effectively determines a cost for disregarding the data in the simulation. The better the simulated structures align to the data, the smaller is the energetic penalty in the form of the bias[15].</p><blockquote id="8c213ee4-545c-4a6b-9814-e8edc8ce415d" class="">一种有效的方法是用分子动力学 (molecular dynamics,MD)[10,11] 补充实验数据。MD 模拟提供了基于物理的分子运动描述，并深入了解生物分子功能。原子轨迹是通过对相互作用粒子系统的牛顿运动方程积分得出的，力是根据经验原子间势（力场）计算的。在数据辅助MD中，对目标数据的能量约束被添加到力场中，以支持与数据一致的构象。这种偏置电位与模拟结构的理论数据和实验数据的最小二乘偏差成正比[7,10,12-14]。假定推导\派生的力将蛋白质推向复制目标数据的构象。由于分子系统试图最小化其自由能，因此偏差有效地确定了在模拟中忽略数据的成本。模拟结构与数据的一致性越好，偏差形式的能量损失就越小。</blockquote><p id="2fefdb89-ff8d-4a9c-b08a-713158133525" class="">An inherent issue with data-assisted simulations is their reliance on MD parameters[7–9]. Selecting adequate values is non-trivial and crucial for simulation performance, and determining the bias
potential’s weight is the key challenge. This selection determines how experimental and theoretical information is balanced. The bias weight is an empirical MD parameter expressing the confidence in the experimental data versus the physics-based force field[8]. In Bayesian methods, the right weighting is derived from a statistical treatment[13,14]. However, such sophisticated approaches are practically inapplicable for users with a primarily experimental background. It is still common practice to manually determine an ‘optimal’ bias weight via grid search, that is, an exhaustive search through a fixed subset of the parameter space[7,9,12]. Adopting concepts from computational intelligence, we introduce FLAPS (‘flexible self-adapting particle swarm’ optimization), a self-learning metaheuristic based on particle swarms, to resolve this parameter selection problem. Our contributions include the following:</p><blockquote id="1ca3e4df-cefa-4191-993d-f237dce54ae5" class="">数据辅助模拟的一个固有问题是它们对 MD 参数的依赖 [7-9]。选择合适的值对于仿真性能而言至关重要，而确定偏置电位的权重是关键的挑战。这种选择决定了如何平衡实验和理论信息。偏差权重是一个经验 MD 参数，表示实验数据与基于物理的力场的置信度[8]。在贝叶斯方法中，正确的权重来自统计处理 [13,14]。然而，这种复杂的方法实际上不适用于具有主要实验背景的用户。通过网格搜索即通过参数空间的固定子集进行穷举搜索[7,9,12]来手动确定“最佳”偏差权重仍然是常见的做法。采用计算智能的概念，引入了基于粒子群的自学习元启发式算法 FLAPS（“灵活的自适应粒子群”优化），来解决参数选择问题。我们的贡献包括：</blockquote><ul id="5acc6037-2d2f-43f0-9463-71b09950cd8a" class="bulleted-list"><li style="list-style-type:disc">A new type of flexible objective function (OF) to assess a data-assisted simulation’s plausibility in terms of simulated structures and thus the suitability of the MD parameters used.</li></ul><ul id="fd29835c-97e5-4edb-a301-a0794d101ec1" class="bulleted-list"><li style="list-style-type:disc">A self-adapting particle swarm optimizer for dynamically evolving environments resulting from multiple quality features of different scales in the flexible OF.</li></ul><ul id="55026b50-a119-47cf-8446-af4d0c8422e3" class="bulleted-list"><li style="list-style-type:disc"> Fully integrated and automated MD parameter selection for data-assisted biomolecular simulations.</li></ul><blockquote id="6f746472-78ca-49b8-b738-c36ee68ca80a" class="">一种新的灵活目标函数(objective function,OF)用于评估数据辅助模拟在模拟结构方面的合理性，从而评估所使用的MD参数的适用性。</blockquote><blockquote id="9983651f-b3ac-4a3b-98d1-6951efe862c0" class="">一种自适应粒子群优化算法，用于动态演化环境，该环境由灵活 OF 中不同尺度的多个质量特征产生。</blockquote><blockquote id="ab50640c-0fba-4e88-b29e-8ea724772625" class="">用于数据辅助生物分子模拟的完全集成和自动化的MD参数选择。</blockquote><p id="e0f8495a-3afd-4f27-91b8-9935c1398060" class="">As a proof of concept, we apply FLAPS to the selection of relevant MD parameters in SAXS-guided structure-based protein simulations[7].</p><blockquote id="51092a17-742e-4453-9e00-a1ffcd97939e" class="">作为概念证明，我们将 FLAPS 应用于 SAXS 引导的基于结构的蛋白质模拟中相关 MD 参数的选择[7]。</blockquote><h2 id="a74e9b47-6aa3-4547-a75b-33e7e642e28c" class="">Particle swarm optimization</h2><p id="f7e702d5-6650-492b-be35-a4ec62f43554" class="">Particle swarm optimization (PSO) is a bio-inspired computational-intelligence technique to handle computationally hard problems based on the emergent behaviour of swarms[16]. Swarm intelligence is the collective behaviour of decentralized, self-organized systems. Such systems comprise a population of cooperating particles. Although there is no supervising control, this leads to ‘intelligent’ global behaviour that is hidden to the individual particles.</p><blockquote id="907326f6-3395-4f11-9767-6eeb1a0dcceb" class="">粒子群优化（PSO）是一种受到生物启发的计算智能技术。用于根据群体的涌现行为来处理计算困难问题。群体智能是分散的、自组织的系统的集体行为。 这样的系统由一群协作的粒子组成。 虽然没有监督控制，但这导致了对单个粒子隐藏的“智能”全局行为。</blockquote><p id="307a8613-ab8c-4675-9f21-fc177a93615e" class="">PSO iteratively improves a candidate solution with respect to a quality-gauging OF. The optimization problem is approached by considering a swarm of particles, each corresponding to a particular position in parameter space. Across multiple search rounds (generations), the particles ‘jump’ around in the search space according to their positions and velocities. Particles ‘remember’ their personal best location and the global best location in their social network, which act as attractors in the search space. To propagate a particle to the next generation, a velocity is added to its current position. The velocity has a cognitive component towards the particle’s current personal best and a social component towards the current best in its communication network. Each component is weighted by a random number in the range [<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>0</mn></mrow><annotation encoding="application/x-tex">0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">0</span></span></span></span></span><span>﻿</span></span>, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ϕ</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\phi_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">ϕ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>], where the acceleration coefficients <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ϕ</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\phi_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">ϕ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> balance exploration versus exploitation in the optimization. Eventual convergence is achieved through progressive particle flight contraction by mechanisms such as velocity clamping[17], inertia weight[18] or constriction[19]. Altogether, this is assumed to move the swarm towards the best parameter combination.</p><blockquote id="5de3b26f-03db-4ac1-945f-2c9d03bfdb5c" class="">PSO 迭代地改进候选解的质量测量 OF。优化问题是通过考虑一群粒子来解决的，每个粒子对应于参数空间中的一个特定位置。在多次迭代搜索中，粒子根据它们的位置和速度在搜索空间中“跳跃”。粒子“记忆”它们在搜索空间中的个体最佳位置和全局最佳位置，它们充当搜索空间中的吸引项。为了将粒子传播到下一代，需要将速度添加到其当前位置。速度具有指向粒子当前个体最佳位置的认知成分和指向其搜索空间中当前全局最佳位置的社会成分。每个分量由 [<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>0</mn></mrow><annotation encoding="application/x-tex">0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">0</span></span></span></span></span><span>﻿</span></span>, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ϕ</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\phi_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">ϕ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>] 范围内的随机数加权，其中加速度系数<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ϕ</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\phi_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">ϕ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>在优化中平衡探索与开发。最终收敛是通过速度限制 [17]、惯性重量[18] 或收缩[19]等机制的渐进式粒子飞行收缩来实现的。总而言之，这被假定为将群体移向最佳参数组合。</blockquote><p id="2def64fa-f3af-423c-91e1-6079dfdca419" class="">PSO has been demonstrated to work well for various static problems[20]. However, realistic problems are often dynamic, and the global optimum can shift with time. Application to such problems has been explored extensively, revealing that the algorithm needs to be enhanced by concepts such as repulsion, dynamic network topologies or multi-swarms[21,22].</p><blockquote id="9776267d-096f-422d-bd59-4f232d6c66a0" class="">PSO 已被证明可以很好地解决各种静态问题[20]。然而，现实问题通常是动态的，全局最优解会随时间变化。因此对此类问题的应用进行了广泛探索，表明该算法需要通过排斥、动态网络拓扑或多群组等概念来增强[21,22]。</blockquote><p id="654a4b7e-4a60-4c14-b418-5ca3f97dec82" class="">Maintaining a population of diverse solutions enables enormous exploration and paves the way to large-scale parallelization. Such algorithms can be scaled easily to exploit the full potential of modern supercomputers23 . PSO has several hyperparameters affecting its behaviour and efficiency, and selecting them has been researched extensively[24,25]. Strategies include using meta-optimizers[26,27] or refining them during the optimization[28]. Reference[29]provides an overview of practical PSO applications, while refs.[22] and [30] give comprehensive reviews of PSO with a focus on dynamic environments and hybridization perspectives, respectively.</p><blockquote id="96bcb595-256b-486b-8136-0f0be604136e" class="">维持大量不同的解可以进行大量探索，并为大规模并行化铺平道路。此类算法可以轻松扩展，以充分利用现代超级计算机的潜力[23]。PSO 有几个影响其行为和效率的超参数，并且对它们的选择已被广泛的研究[24,25]。策略包括在优化过程中使用元优化器[26,27] 对其进行改进[28]。文献[29] 概述了PSO 的实际应用。文献[22]和文献[30]分别对PSO的动态环境和混合的角度进行了全面的描述。</blockquote><h2 id="c4bab996-262e-4a5f-aaa4-66e615bd7102" class="">Multi-response problems</h2><p id="ab41ecd0-aed5-4955-9bdd-ac56093e0e44" class="">Real-life optimization problems are often determined by multiple incomparable or conflicting quality features (responses). To obtain compatible solutions, these responses must be taken into account simultaneously. This is often accomplished by combining individual contributions within one composite OF. PSO has been applied to multi-response optimization in many fields[31–34]. Commonly, the set of responses is reduced via multiplication by manually chosen weights, henceforth referred to as OF parameters. Choosing OF parameters is non-trivial, yet strongly impacts global optimization performance by skewing the OF. FLAPS builds on a flexible OF that
automatically and interdependently balances different responses. OF parameters are learned at runtime through iterative refinement, yielding a dynamically evolving OF landscape. In this way, FLAPS can cope with various responses of different scales.</p><blockquote id="2722826e-ed6b-4735-a5df-113962509076" class="">现实生活中的优化问题通常由多个无法比较或相互冲突的质量特征（响应）决定。为了获得兼容的解，必须同时考虑这些响应。这通常是通过在一个复合 OF 中组合个体贡献来实现的。PSO 已被应用于许多领域的多响应优化[31-34]。通常，响应集通过乘以人工手动选择的权重来减少，以下称为 OF 参数。参数选择是一项重要工作，但由于 OF 参数的倾斜会严重影响全局优化性能。FLAPS 建立在灵活的 OF 之上，该 OF 自动且相互依赖地平衡不同的响应。OF 参数在运行时通过迭代细化来学习，从而产生动态演变的 OF 格局。这样，FLAPS 就可以应对不同尺度的各种响应。</blockquote><h2 id="7c2212df-13da-4bdb-91e8-fb63ca4d47ad" class="">A flexible self-adapting objective function</h2><p id="7475c184-1f53-4976-87b4-c7813fc8a201" class="">Typically, the set of responses is mapped to a scalar score by calculating the scalar product with fixed weights. These OF parameters supposedly reflect relative importance, implicitly encoding arbitrary prior beliefs. Instead, we set up a ‘maximum-entropy’ OF with the fewest possible assumptions:</p><blockquote id="b6bafd1a-4ac7-4b1c-a223-52f74ab89f85" class="">通常，通过计算具有固定权重的标量积，将响应集映射到标量分数。这些 OF 参数被认为反映了相对重要性，隐含地编码了任意的先验信念。相反，我们用最小假设建立了一个“最大熵”：</blockquote><p id="02f56864-ccd8-4a44-bc3a-c4382fd7bf64" class="">
</p><figure id="3bd3285b-2f23-45bf-b1b6-5167480f412a" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled.png"><img style="width:288px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled.png"/></a></figure><p id="7b3999f8-c0cd-485a-9074-d1712842330d" class=""><mark class="highlight-gray">Eq.(1)</mark></p><p id="368e4c5c-87f9-4fba-87dd-6258670afd7d" class="">          </p><p id="72d10ca4-0708-4ecb-ad6e-fda42242fec3" class="">                                    </p><p id="39a143a0-e2e8-49a9-91cc-1aec01bd2aa6" class="">where z is the set of OF parameters, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>μ</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">μ_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal">μ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> deviation of response <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>R</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">R_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.969438em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.00773em;">R</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.00773em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> is the mean and <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>σ</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">σ_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">σ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> is the standard for a particle at position x. All responses are considered equally important but can have different ranges and units. To make them comparable on a shared scale, we standardize each response’s set of values gathered over previous generations. This strategy imitates the concept of rolling batch normalization[35]. Each layer’s inputs are recentred and rescaled with the aim to improve a neural network’s speed, performance and stability. Initially proposed to mitigate internal covariate shift, batch normalization is believed to introduce a regularizing and smoothing effect and promote robustness with respect to different initialization schemes. The OF in equation (1) depends not only on the parameters of interest, x, in our case the MD parameters, but also on a priori unknown, context-providing OF parameters, z =<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mrow><mi>μ</mi><mo separator="true">,</mo><mi>σ</mi></mrow><msub><mo stretchy="false">)</mo><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">({μ, σ})_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.036108em;vertical-align:-0.286108em;"></span><span class="mopen">(</span><span class="mord"><span class="mord mathnormal">μ</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">σ</span></span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>, from the standardization. Their values cannot be deduced from individual OF evaluations, yet fundamentally control OF performance and hence the optimization process.</p><p id="03c9f3bb-f0cb-41a6-9ac6-35d843e92195" class="">其中 z 是 OF 参数集，<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>μ</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">μ_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal">μ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> 是平均值，<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>σ</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">σ_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">σ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> 是位置 x 处粒子的响应 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>R</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">R_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.969438em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.00773em;">R</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.00773em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> 的标准偏差。所有响应都被认为同样重要，但可以有不同的范围和单位。为了使它们在一个共享尺度上具有可比性，我们将前几代收集的每个响应的值集标准化。该策略模仿滚动批量标准化的概念[35]。为了提高神经网络的速度、性能和稳定性，对每一层的输入进行重新编码和重新标定。批量归一化最初是为了减轻内部协变量偏移而提出的，它被认为引入了正则化和平滑效应，并提高了不同初始化方案的鲁棒性。等式 (1) 中的 OF 不仅取决于参数 x，在我们的例子中是 MD 参数，还取决于来自标准化的先验未知的提供上下文的 OF 参数 z =<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mrow><mi>μ</mi><mo separator="true">,</mo><mi>σ</mi></mrow><msub><mo stretchy="false">)</mo><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">({μ, σ})_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.036108em;vertical-align:-0.286108em;"></span><span class="mopen">(</span><span class="mord"><span class="mord mathnormal">μ</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">σ</span></span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>。它们的值不能从单独的 OF 评估中推断出来，但从根本上控制了 OF 的性能，因此也控制了优化过程。</p><p id="a9b97a0e-1629-4226-b657-dafe635cfe95" class="">Our self-adapting PSO variant, FLAPS, solves this problem. Provided with a comprehensive history of all previous particles and their responses, OF parameters are learned on the fly. They are continuously refined according to the current state of the optimization, yielding a dynamically evolving and increasingly distinct OF topology. This environmental dynamism may cause convergence problems if the OF fails to approach a stable topology. As more particles are evaluated, the ranges and distributions of individual responses become better understood. Therefore, OF parameters become more accurate, improving OF performance in assessing the suitability of the actual parameters of interest, x. After each generation, the values z =<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mrow><mi>μ</mi><mo separator="true">,</mo><mi>σ</mi></mrow><msub><mo stretchy="false">)</mo><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">({μ, σ})_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.036108em;vertical-align:-0.286108em;"></span><span class="mopen">(</span><span class="mord"><span class="mord mathnormal">μ</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">σ</span></span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> are used to reevaluate the OF for all particles in the history. Personal best positions, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">p_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">p</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>, and the swarm’s global best position, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>, are updated accordingly for propagating particles to the next generation. FLAPS uses a traditional PSO velocity formulation[16]. Strategies to prevent diverging velocities include introducing an inertia weight[18]or a constriction factor[19]. We regulate the velocities by means of a maximum value at each particle update[17]. FLAPS’s pseudo code is shown in Algorithm 1. Inspired by the ‘simplifying PSO’ paradigm, it builds on a slim standard PSO core and can easily be complemented by concepts such as inertia weight[18]and swarm constriction[19] or diversity increasing mechanisms. Its time complexity is similar to that of a standard PSO with
<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>O</mi><mo stretchy="false">(</mo><mi>s</mi><mi mathvariant="normal">/</mi><mi>p</mi><mo separator="true">⋅</mo><mi>G</mi><mo separator="true">⋅</mo><mi>S</mi><mi>i</mi><mi>m</mi><mo>+</mo><mi>S</mi><mo separator="true">⋅</mo><mi>O</mi><mi>p</mi><mi>t</mi><mo>=</mo><mi>O</mi><mo stretchy="false">(</mo><mi>s</mi><mi mathvariant="normal">/</mi><mi>p</mi><mo separator="true">⋅</mo><mi>G</mi><mo separator="true">⋅</mo><mi>S</mi><mi>i</mi><mi>m</mi><mo>+</mo><msup><mi>S</mi><mn>2</mn></msup><mo separator="true">⋅</mo><mi>G</mi><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">O(s/p·G·Sim+S·Opt=O(s/p·G·Sim+S^2·G)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.02778em;">O</span><span class="mopen">(</span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mord mathnormal">p</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mord mathnormal">im</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:0.8777699999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">Opt</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.02778em;">O</span><span class="mopen">(</span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mord mathnormal">p</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mord mathnormal">im</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:1.064108em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mclose">)</span></span></span></span></span><span>﻿</span></span> in Landau notation, where <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi></mrow><annotation encoding="application/x-tex">P</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">P</span></span></span></span></span><span>﻿</span></span> is the number of simulation processors, ‘Sim’ the maximal simulation time, and all other variables as defined in Algorithm 1.</p><blockquote id="9609c1e2-e03c-479c-b9dd-9d2175356639" class="">我们的自适应PSO变体FLAPS解决了这个问题。 提供了所有先前粒子及其响应的全面历史记录，可以动态学习OF参数。它们根据当前的优化状态不断改进，产生动态演化和日益不同的拓扑结构。如果OF无法接近一个稳定的拓扑，这种动态环境可能导致收敛问题。随着更多的粒子被评估，个体响应的范围和分布变得更好理解。因此，对OF参数的评估变得更加准确，从而提高了OF在评估实际参数x的适用性方面的性能。在每一代之后，值z=<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mrow><mi>μ</mi><mtext>，</mtext><mi>σ</mi></mrow><msub><mo stretchy="false">)</mo><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">({μ，σ})_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.036108em;vertical-align:-0.286108em;"></span><span class="mopen">(</span><span class="mord"><span class="mord mathnormal">μ</span><span class="mord cjk_fallback">，</span><span class="mord mathnormal" style="margin-right:0.03588em;">σ</span></span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>用于重新计算历史上所有粒子的OF。个体最佳位置<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">p_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">p</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>和种群全局最佳位置<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>B</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{Best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.32833099999999993em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.05017em;">B</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>相应地更新，以便将粒子传播到下一代。FLAPS使用传统的PSO速度公式[16]。防止速度发散的策略包括引入惯性重量[18]和收缩因子[19]。我们通过每一个粒子更新时的最大值来调节速度[17]。FLAPS的伪代码如算法1所示。受“简化粒子群算法”范式的启发，它建立在一个标准粒子群算法核心之上，并可以很容易地通过惯性权重[18]和群体收缩[19]和多样性增加机制等概念进行补充。其时间复杂度与标准粒子群算法相似，其Landau表示法为<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>O</mi><mo stretchy="false">(</mo><mi>s</mi><mi mathvariant="normal">/</mi><mi>p</mi><mo separator="true">⋅</mo><mi>G</mi><mo separator="true">⋅</mo><mi>S</mi><mi>i</mi><mi>m</mi><mo>+</mo><mi>S</mi><mo separator="true">⋅</mo><mi>O</mi><mi>p</mi><mi>t</mi><mo>=</mo><mi>O</mi><mo stretchy="false">(</mo><mi>s</mi><mi mathvariant="normal">/</mi><mi>p</mi><mo separator="true">⋅</mo><mi>G</mi><mo separator="true">⋅</mo><mi>S</mi><mi>i</mi><mi>m</mi><mo>+</mo><msup><mi>S</mi><mn>2</mn></msup><mo separator="true">⋅</mo><mi>G</mi><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">O(s/p·G·Sim+S·Opt=O(s/p·G·Sim+S^2·G)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.02778em;">O</span><span class="mopen">(</span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mord mathnormal">p</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mord mathnormal">im</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:0.8777699999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">Opt</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.02778em;">O</span><span class="mopen">(</span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mord mathnormal">p</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mord mathnormal">im</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:1.064108em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mpunct">⋅</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mclose">)</span></span></span></span></span><span>﻿</span></span>,其中<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi></mrow><annotation encoding="application/x-tex">P</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">P</span></span></span></span></span><span>﻿</span></span>为仿真处理器数，&#x27;Sim&#x27;为最大仿真时间，以及算法1中定义的所有变量。</blockquote><pre id="424a64ea-ddb5-4510-9b38-aad295fa762d" class="code"><code>算法 1 FLAPS 算法
分别在搜索空间的上限 b_{up} 和 下限b_{lo} 的之间的随机位置 x_p (p = 1, ..., S) 处使用种群大小为 S 的初始种群 pop。
for g ← 1 to maximum generations G do
  for particle in pop do 
  评估particle在位置 x_p处的响应：
  particle.fargs = [response_j (x_p )]_j
  end 
  将当前一代 pop 附加到历史 histp: histp.append(pop) 
  根据 histp 中响应的当前知识状态更新参数 z_g：z_g = updateParams(histp)
  for particle in histp do
  使用最近的 z_g 重新评估目标函数：particle.fitness = f(x_p ; z_g )
  end 
  for generation in histp do 
    for particle in generation do 
    确定个体最佳p_{best} 并相应更新全局最佳 g_{best}. 
    end
  end
  for particle in pop do
    更新粒子速度和位置：particle.speed+= rand (0, ϕ_1) (p_{best} −particle.position )+ rand (0, ϕ_2)(g_{best} −particle.position) 
    通过 s_max = 0.7 G_{-1} (b_{up}-b_{lo})调节速度:
    if particle.speed &gt; s_{max} then
       particle.speed = s_{max}
    end 
    if particle.speed &lt; -s_{max} then
       particle.speed = -s_{max}
    end
    particle.position+ =particle.speed
  end
end 
结果：g_{best}</code></pre><h2 id="2988e98c-c18a-44a1-a810-9718f0a2c0d2" class="">Application to data-assisted protein simulations</h2><p id="299c65ab-1129-4e86-a471-8e4d6be58e34" class="">We applied FLAPS to the optimization of MD parameters in SAXS-guided protein simulations[7]. SAXS data are integrated into computationally efficient structure-based models, which probe
dynamics arising from a protein’s native geometry[36,37]. To assess the utility of different MD parameter sets, we need a metric for simulation quality in terms of physically reasonable structures matching the data. Designing such an OF in advance is non-trivial and has two major aspects: (1) physical plausibility of a simulated ensemble of protein structures and (2) its agreement with the target data, that is, how well the data are reproduced by simulated structures. To represent these aspects, we use the Rosetta energy function 2015 (REF15)[38,39] and the least-squares deviation χ2 of simulated data from the target data[7].</p><blockquote id="9b64fee8-a2f0-4e73-88ef-4446f494dbac" class="">我们将 FLAPS 应用于 SAXS 引导的蛋白质模拟中 MD 参数的优化[7]。SAXS 数据被集成到计算效率高的基于结构的模型中，该模型探测了蛋白质的天然几何结构所产生的动态变化[36,37]。为了评估不同 MD 参数集的效果，我们需要一个模拟质量的衡量标准，即物理上合理的结构与数据相匹配。提前设计这样的 OF 并非易事，主要有两个方面：（1）模拟的蛋白质结构集合的物理合理性以及（2）它与目标数据的一致性，即模拟结构对数据的再现程度。为了表示这些方面，我们使用罗塞塔能量函数2015 (REF15)[38,39]和模拟数据与目标数据[7]的最小二乘偏差<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>χ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">χ^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">χ</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>。</blockquote><p id="9064275c-96ed-4ac4-8f4b-f64d05e199da" class="">Protein structure determination relies on quick and reliable scoring of many models to select those closest to the native state. Structures are rated by energetic scores associated with their conformational state. REF15 is a weighted sum of energy terms efficiently approximating the energy of a biomolecular conformation as a function of geometric degrees of freedom and chemical identities[38]. With a protein’s native fold corresponding to the state with minimal free energy, a lower-scoring structure is expected to be more native-like. Because the scores do not have a direct conversion to physical energies, REF15 and structural stability are not correlated across different proteins. Similarly, χ2 values without context are inconclusive and must be compared for each protein system. Both REF15 and χ2 are available from a simulation, yielding a molecular system’s atom positions over time.</p><blockquote id="34c1aefe-4b3e-4770-8f62-186698ebb210" class="">蛋白质结构的测定依赖于对许多模型进行快速可靠的评分，以选择最接近天然状态的模型。结构通过与其构象状态相关的能量分数来评定。REF15 是能量项的加权总和，有效地近似作为几何自由度和化学特性的生物分子构象的能量。由于蛋白质的天然折叠对应于具有最小自由能的状态，因此预计得分较低的结构更像天然结构。因为分数没有直接转换为物理能量，所以 REF15 和结构稳定性在不同蛋白质之间不相关。同样，没有上下文的 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>χ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">χ^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">χ</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> 值是不确定的，必须针对每个蛋白质系统进行比较。REF15 和<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>χ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex"> χ^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">χ</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> 都可以从模拟中得到，得出分子系统的原子位置随时间的变化。</blockquote><p id="b3fe79fc-0e25-4b1f-827e-54abff3197d3" class="">For SAXS-guided structure-based simulations, two MD parameters are particularly important: bias weight kχ and temperature <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>T</mi></mrow><annotation encoding="application/x-tex">T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>﻿</span></span>. kχ balances information in the SAXS data with the physical model,
and T is a measure of available thermal energy and controls the system’s conformational flexibility. Thus, a particle corresponds to a simulation using a particular MD parameter set, x =(<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>χ</mi><mo separator="true">,</mo><mi>T</mi></mrow><annotation encoding="application/x-tex">kχ,T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mord mathnormal">χ</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>﻿</span></span>). The
OF is set up as</p><blockquote id="7744dbe8-2f54-402a-a7d2-2bfdb1d304a0" class="">对于 SAXS 引导的基于结构的模拟，两个 MD 参数尤为重要：偏置权重<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>χ</mi></mrow><annotation encoding="application/x-tex"> kχ </annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mord mathnormal">χ</span></span></span></span></span><span>﻿</span></span>和温度 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>T</mi></mrow><annotation encoding="application/x-tex">T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>﻿</span></span>。<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>χ</mi></mrow><annotation encoding="application/x-tex">kχ</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mord mathnormal">χ</span></span></span></span></span><span>﻿</span></span> 平衡 SAXS 数据中的信息与物理模型，<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>T</mi></mrow><annotation encoding="application/x-tex">T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>﻿</span></span> 是可用热能的度量并控制系统的构象灵活性。因此，粒子对应于使用特定 MD 参数集 x =(<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>χ</mi><mo separator="true">,</mo><mi>T</mi></mrow><annotation encoding="application/x-tex">kχ,T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mord mathnormal">χ</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>﻿</span></span>) 的模拟。OF设置为</blockquote><p id="88a0ebd4-e12c-4f7b-bac2-ff7a7ded1b92" class="">
</p><figure id="506306e3-b6b3-4c5c-aefb-2bf0842a77cb" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%201.png"><img style="width:288px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%201.png"/></a></figure><p id="714348c6-09b0-4c39-8879-988a69d61772" class=""><mark class="highlight-gray">Eq.(2)</mark></p><p id="90945aec-ba03-4c69-823e-3f3583aeb563" class="">
</p><p id="cdccce32-e780-46d3-aadf-4cbb36cc57b9" class="">                                   </p><p id="09325c7c-6140-40b3-be84-2ee48d0eeb3a" class="">The first response evaluates the average physical stability of simulated structures, the second is the median χ2 deviation of simulated data from the target data. Owing to the ill-posed nature of the SAXS inverse problem, globally distinct protein structures can possess the same scattering intensity. As shown in Fig. 1, this can lead to a pronounced ambiguity in χ2. To resolve the resulting non-injectivity in the OF, we introduce a third response, the inverse average χ2 deviation. This acts as a regularizer, rewarding deviations from the target data and thus preventing possible overfitting. Combining these responses yields a surrogate model of a simulated ensemble’s similarity to the desired target structure. The smaller the OF, the more physical, data-consistent and (likely) similar to the target state the simulated structures are.</p><blockquote id="716503f6-a3a0-485a-ae50-6202ba418822" class="">第一个是响应评估模拟结构的平均物理稳定性，第二个是响应评估模拟数据与目标数据的中值<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>χ</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">χ2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">χ</span><span class="mord">2</span></span></span></span></span><span>﻿</span></span>偏差。 由于SAXS逆问题的不适定性质，全局不同的蛋白质结构可以具有相同的散射强度。 如图1所示 ,这会导致<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>χ</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">χ2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">χ</span><span class="mord">2</span></span></span></span></span><span>﻿</span></span>的明显歧义。 为了解决在OF中产生的非内射性，我们引入了第三个响应，逆平均<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>χ</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">χ2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">χ</span><span class="mord">2</span></span></span></span></span><span>﻿</span></span>偏差。 这起到了正则化的作用，奖励与目标数据的偏差，从而防止可能的过拟合。 组合这些响应产生了一个模拟集合体与期望目标结构相似度的代用模型。 OF越小，模拟结构的物理、数据一致性和（可能）与目标状态越相似。</blockquote><p id="d35f3675-8b6a-4d56-8049-452cf8feab18" class="">In physico-empirical structure-based models, different combinations of bias weight and temperature can equally yield useful results. There is no MD parameter ground truth for this type of simulation, so a purely evidence-based evaluation according to the similarity of simulated protein structures to the target is to be applied.</p><blockquote id="c679af23-0ae1-4dd6-b6b6-143a25ce3a96" class="">在基于物理经验结构的模型中，偏置权重和温度的不同组合同样可以产生有用的结果。这种类型的模拟没有 MD 参数的基础真值，因此将根据模拟蛋白质结构与目标的相似性进行纯粹的基于证据的评估。</blockquote><p id="544dc284-4171-44d3-84d4-a5d27197a8f7" class="">We use the global distance test (GDT)[40] to quantify differences between two conformations of a protein (Methods section ‘Root-mean-square deviation’). To estimate how similar two superimposed structures are, the displacement of each alpha carbon is compared to various distance cutoffs. Percentages, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>P</mi><mi>x</mi></msub></mrow><annotation encoding="application/x-tex">P_x</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.83333em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.151392em;"><span style="top:-2.5500000000000003em;margin-left:-0.13889em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">x</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> , of alpha carbons with displacements below cutoffs of x Å are used to calculate the total score:</p><blockquote id="aa22a34a-0c1b-469d-a412-45a9852762ca" class="">我们使用全局距离测试 (GDT)[40]来量化蛋白质的两种构象之间的差异（方法部分“均方根偏差”）。为了估计两个叠加结构的相似程度，将每个 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>α</mi></mrow><annotation encoding="application/x-tex">α</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.0037em;">α</span></span></span></span></span><span>﻿</span></span> 碳的位移与不同距离截止值进行比较。位移值小于截止值 x Å 的 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>α</mi></mrow><annotation encoding="application/x-tex"> α </annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.0037em;">α</span></span></span></span></span><span>﻿</span></span> 碳的百分比<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>P</mi><mi>x</mi></msub></mrow><annotation encoding="application/x-tex"> P_x </annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.83333em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.151392em;"><span style="top:-2.5500000000000003em;margin-left:-0.13889em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">x</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>用于计算总分：</blockquote><p id="fff38e97-e74c-47c7-827c-5dc0395346e0" class="">
</p><figure id="b8956a06-c147-4e76-91ad-f394a3fae944" class="image" style="text-align:center"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%202.png"><img style="width:419px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%202.png"/></a></figure><p id="09a10690-8b7f-4c6b-a970-3b0691501abf" class=""><mark class="highlight-gray">Eq.(3)</mark></p><p id="c5bea738-be6c-47ac-8503-97529b206a94" class="">                  </p><p id="03bae6ee-5d24-4098-8e31-f1a5f82be98b" class="">Higher GDT values indicate a stronger similarity between two models. Structures with GDT &gt; 50 are considered topologically accurate[41]. The GDT is used to validate the OF as a surrogate model of
an ensemble’s similarity to the target structure.</p><blockquote id="fa7094a8-8d18-4c53-af59-c9d5ebc0c3db" class="">较高的GDT值表明两个模型之间的相似性更强。 GDT&gt;50的结构被认为在拓扑学上是准确的[41]。 GDT用于验证OF作为一个代用模型的集合体与目标结构的相似性。</blockquote><figure id="6262188f-427a-4d80-a4ef-af6c97836338" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%203.png"><img style="width:583px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%203.png"/></a></figure><p id="7d1220cc-1a62-4122-8679-76c9c7a84256" class=""><mark class="highlight-gray">Fig. 1 | Ambiguous χ2. The ill-posedness of the SAXS inverse problem results in a pronounced ambiguity in the χ2 deviation of simulated data from the target data. This manifests in a non-injective ‘two-branch’ behaviour of the second response as a function of similarity of simulated structures to the desired target state, here quantified by the GDT. GDTsys, GDT between initial and target protein structure; kχ, bias weight; </mark><mark class="highlight-gray"><style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ε</mi></mrow><annotation encoding="application/x-tex">ε</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal">ε</span></span></span></span></span><span>﻿</span></span></mark><mark class="highlight-gray">, energy scale of the structure-based model. Data from all lysine-, arginine-, ornithine-binding protein holo-to-apo runs are combined.</mark></p><h2 id="13ac445b-9e1b-46e4-8e32-8ab3b407400b" class="">Results</h2><p id="c13104f7-f3ad-49e6-aca3-76e5a4540c4a" class="">We optimized MD parameters of SAXS-guided structure-based simulations for two well-characterized proteins: lysine-, arginine-, ornithine-binding (LAO) protein and adenylate kinase (ADK).</p><blockquote id="fd2131b5-1446-446c-b41f-ef2d6180b19d" class="">我们优化了 SAXS 引导的基于结构模拟的 MD 参数，包括两种特征明确的蛋白质：赖氨酸、精氨酸、鸟氨酸结合 (LAO) 蛋白和腺苷酸激酶 (ADK)。</blockquote><p id="e952b2e5-2827-49dc-a287-12b307daea61" class="">Small ligands such as sugars and amino acids are actively transported into bacteria across cell membranes[42]. Dedicated transport systems comprise a receptor (that is, the binding protein) and a membrane-bound protein complex. Interactions of the ligated binding protein with the membrane components induce conformational changes in the latter, forming an entry pathway for the ligand. We study LAO protein (Fig. 2a), which undergoes a conformational change from an apo (unligated; Protein Data Bank43 (PDB) code 2LAO44) to a holo (ligated; PDB code 1LST44) state upon ligand binding. The structures have a GDT of 39.39. SAXS-guided simulations started from the holo state and aimed at the apo state.</p><blockquote id="2247bb75-6c33-402c-8e81-d2a9932df5ad" class="">糖和氨基酸等小配体通过细胞膜[42]积极地被转运到细菌中。专用转运系统包括受体（即结合蛋白）和膜结合蛋白复合物。连接的结合蛋白与膜成分的相互作用诱导后者的构象变化，形成配体的进入途径。我们研究 LAO了 蛋白（图 2a），它在配体结合时经历了从 APO（未连接；蛋白质数据库[43]（PDB）代码 2LAO[44]）到holo（连接；PDB 代码 1LST[44]）状态的构象变化。这些结构的 GDT 为 39.39。 SAXS 引导的模拟从holo状态开始以 apo 状态为目标。</blockquote><p id="3a8c5807-3acd-44ec-8970-a7938ca586f5" class="">Adenosine triphosphate (ATP) is the universal energy source in cells and is vital to processes such as muscle contraction and nerve impulse propagation. By continuously checking ATP levels, ADK provides the cell with a mechanism to monitor energetic levels and metabolic processes. The transition between an open (PDB code 4AKE45) and closed (PDB code 1AKE46) state is quintessen-
tial to its catalytic function47. The structures have a GDT of 33.06. SAXS-guided simulations started from the open conformation and aimed at the closed one.</p><blockquote id="2ad7537e-0b53-4838-a7e7-78430422abc9" class="">三磷酸腺苷(ATP)是细胞中普遍存在的能量来源，对肌肉收缩和神经冲动传播等过程至关重要。通过不断检测ATP水平，ADK为细胞提供了一种监测能量水平和代谢过程的机制。打开状态（PDB代码4AKE[45])和关闭状态（PDB代码1AKE[46])之间的转换是其催化功能的典型特征[47]。这些结构的GDT为33.06。SAXS引导的模拟从开放构象开始，以封闭构象为目标。</blockquote><p id="760099ea-fa39-4022-bfed-252103fed0c6" class="">Artificial target data were calculated from known structures with CRYSOL48. Statistical uncertainties were modelled following ref. [49].</p><blockquote id="7ddf5a14-594b-414b-9772-a2168ad23afb" class="">人工目标数据是用CRYSOL[48]的已知结构计算出来的。统计上的不确定性是按照参考文献建立的[49]。</blockquote><figure id="db110c5d-dac6-415f-94df-272fdd1d7578" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%204.png"><img style="width:601px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%204.png"/></a></figure><p id="be89b13e-ad77-473e-a2d2-2c5cc629d142" class=""><mark class="highlight-gray">Fig. 2 | Test systems. a, Lysine-, arginine-, ornithine-binding protein in its holo (coloured) and apo (grey) state, with a GDT of 39.39. b, Adenylate kinase in its open (coloured) and closed (grey) state, with a GDT of 33.06. The colouring indicates the displacement of each alpha carbon in the initial structure with respect to the target state. The average alpha-carbon displacement in each coloured structure with respect to each grey structure is given. Structures are visualized with PyMOL58.</mark></p><p id="fa018a68-4101-4463-a63e-3042771f94dd" class="">
</p><p id="89a7760b-1e0f-4d96-ada6-47f45059fb3b" class="">We performed seven FLAPS runs with different initial conditions for each protein. Swarm-based metaheuristics such as FLAPS have hyperparameters influencing the optimization behaviour, and
their efficacy can only be demonstrated empirically by a finite number of computational experiments. We present results for a swarm of 10 particles and 15 generations as a workable trade-off between optimization performance and compute time for the considered application. This set-up was found to be sufficient for convergence in preceding trial runs. Calculations were performed on 1,000 cores of a supercomputer. One run cost ~40,000 core hours. The results of the three best runs are listed in Table 1 (complete results are provided in Supplementary Tables 1 and 2).</p><blockquote id="8a6f0382-4134-42ea-b528-666fc292b5e6" class="">我们对每种蛋白质使用不同的初始条件进行了七次 FLAPS 运行。基于群体的元启发式算法（例如 FLAPS）具有影响优化行为的超参数，并且它们的功效只能通过有限数量的计算实验来证明。我们展示了一组 由10 个粒子和 15 代组成的结果，作为所考虑应用程序的优化性能和计算时间之间的可行权衡。发现这种设置足以实现收敛。计算是在一台超级计算机的 1,000 个核心上进行的。一次运行成本约为 40,000 核心小时。表1列出了三个最佳运行的结果（完整的结果见补充表 1 和 2）。</blockquote><figure id="5328cb32-fc1c-49ef-8426-bbecf88d0eb8" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%205.png"><img style="width:1198px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%205.png"/></a></figure><p id="28f9f094-6a0f-4d95-8b89-3ca9b0fef4cf" class="">As shown for LAO protein in Fig. 3, the OF consistently converged to a stable topology (Supplementary section ‘Analyzing swarm convergence’).</p><blockquote id="c097d37c-2405-445f-bbc1-d181805ee4f5" class="">LAO 蛋白如图 3 所示，OF 始终收敛到稳定的拓扑结构（补充部分“分析群收敛”）。</blockquote><figure id="1d890102-b703-4397-819f-9ba73ea5e610" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%206.png"><img style="width:1027px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%206.png"/></a></figure><p id="e2d6a06e-d8ce-4dbe-9ee1-33e97a7c819e" class=""><mark class="highlight-gray">Fig. 3 | Dynamically evolving OF topology. Results are shown for LAO protein (seed 1790954). The current global best position is marked by a star. </mark><mark class="highlight-gray"><style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ε</mi></mrow><annotation encoding="application/x-tex">ε</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal">ε</span></span></span></span></span><span>﻿</span></span></mark><mark class="highlight-gray"> is the energy scale of the structure-based model.</mark></p><p id="2b3957cd-e448-43ca-a4bb-82a3ac67e8bc" class="">
</p><p id="38f6eea4-f0e4-4789-bcf6-378d1c31e8c3" class="">For each simulation, we calculated the median GDT with respect to the target state from all structures in the trajectory. To validate the OF, we state its Pearson correlation ρ with the median GDT as a measure of linear correlation. Because minimizing the OF should be equivalent to maximizing the GDT, negative correlations, ideally −1, are expected. The OF’s suitability is confirmed for both LAO protein and ADK with correlations up to −0.94 and −0.85, respectively.</p><blockquote id="131fa4ac-a60a-4b88-b884-4c7108370e59" class="">对于每一次模拟，我们计算了轨迹中所有结构相对于目标状态的 GDT中值。为了验证 OF，我们将其与Pearson 相关度<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ρ</mi></mrow><annotation encoding="application/x-tex"> ρ</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">ρ</span></span></span></span></span><span>﻿</span></span> 作为线性相关的度量。因为最小化 OF 应该等同于最大化 GDT，所以预计会出现负相关（理想情况下为 -1）。OF 对 LAO 蛋白和 ADK 的适用性都得到了证实，相关性分别达到-0.94 和 -0.85。</blockquote><p id="341f0729-fe31-4ad7-9f0e-55b87bc5a67c" class="">As the Pearson correlation only reflects linearity, we also studied the exact relations between the OF and the median GDT (Fig. 4). To identify the best MD parameter combinations, the OF must have
low values for large GDTs, irrespective of the actual relationship’s complexity. This is the case for both test systems.</p><blockquote id="eb6b4ef9-877d-48ca-b531-ad018982be48" class="">由于 Pearson 相关仅反映线性，我们还研究了 OF 和 GDT中值之间的精确关系（图 4）。为了确定最佳的 MD 参数组合，无论实际关系的复杂性如何，对于大型 GDT，OF 必须具有较低的值，这两个测试系统都是这样。</blockquote><figure id="80137984-c829-4836-aff6-786c66cb2af9" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%207.png"><img style="width:1026px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%207.png"/></a></figure><p id="12373b4f-3957-4b1d-a4f1-0f65ad6d0765" class=""><mark class="highlight-gray">Fig. 4 | OF versus median GDT. a,b, OF versus median GDT for LAO protein (a) and ADK (b). GDTsys
, GDT between initial and target structure of each test system; ε, energy scale of the structure-based model.</mark></p><p id="3f6ac5d3-3434-49c8-ab23-880c3c06a008" class="">
</p><p id="f896714f-277c-4391-950b-f060d0b9c157" class="">Global best positions, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>, returned functional MD parameter combinations throughout. For LAO protein, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> median GDTs were consistently of the order of 70 and correspond well to the best values reached. This means that for half of the simulated structures, at least 70% of all alpha carbons lie within a small radius from their positions in the target state. These results indicate the structural accuracy of the simulated ensemble for <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> and thus convergence to the target state and successful refinement against the data. The same is true for ADK, where <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> and the best median GDTs are around 63 and more similar than those of LAO protein. Example structures from <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> simulations, shown in Fig. 5, are in nearly perfect accordance with the target states.</p><blockquote id="8dbe8b81-518f-429f-a073-13085e727bd5" class="">全局最佳位置<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>，返回整个功能MD参数组合。对于LAO蛋白，<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> 的中值GDTs一直在70左右，很好的对应于达到的最佳值。这意味着在一半的模拟结构中，至少有70%的<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>α</mi></mrow><annotation encoding="application/x-tex">α</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.0037em;">α</span></span></span></span></span><span>﻿</span></span>碳原子位于目标状态位置的小半径范围内。这些结果表明了<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>模拟集合的结构准确性，从而收敛到目标状态，并成功地对数据进行细化。ADK的情况也是如此，<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>和最佳中值GDTs在63左右，比LAO蛋白的GDTs更相似。来自<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>模拟的结构实例如图5所示，与目标状态几乎完全一致。</blockquote><p id="35c5a361-a686-432f-8ae0-ecdd96ec776f" class="">Additionally, we considered the reversed conformational transitions, that is, from apo to holo state for LAO protein and closed to open state for ADK (Supplementary Tables 3 and 4, respectively).
With <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ρ</mi></mrow><annotation encoding="application/x-tex">ρ</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">ρ</span></span></span></span></span><span>﻿</span></span> between −0.56 and −0.88, FLAPS could also identify functional MD parameters for SAXS-guided apo-to-holo simulations of LAO protein. <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> and the best median GDTs were consistently slightly below 70, indicating high similarity of the simulated structures and the desired target state for the MD parameter combinations found.</p><blockquote id="14c5118a-6825-4762-b491-4b8eed7c1d5f" class="">此外，我们考虑了反向构象转变，即 LAO 蛋白从 apo 到全息状态，ADK 关闭到开放状态（分别补充表 3 和 4）。<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>ρ</mi></mrow><annotation encoding="application/x-tex">ρ</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">ρ</span></span></span></span></span><span>﻿</span></span>在-0.56和-0.88之间，FLAPS也可以为SAXS引导的LAO蛋白的apo-to-holo模拟确定功能性MD参数。<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>和最佳中值GDTs始终略低于70，表明模拟结构和找到的MD参数组合的期望目标状态高度相似。</blockquote><p id="4909cf97-586d-4b67-9dd9-8d9b6698f637" class="">By comparison, Pearson correlations up to −0.53 and <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> median GDTs of ~45 indicate only average structural accuracy for the closed-to-open transition of ADK. However, overall best median
GDTs were around 50. With only half of all alpha carbons within a small radius from their target positions, the observed behaviour is not a problem of FLAPS, but is due to the limits of the underlying simulation method[7]. The information in the coarse-grained structure-based
model and the low-resolution SAXS data seems to be insufficient to determine the molecular structure with the same accuracy as for the other test cases. However, even under these circumstances, FLAPS was capable of determining acceptable parameters.</p><blockquote id="af74d2b1-195d-465e-93af-4bc5dbaecf11" class="">相比之下，高达 -0.53 的 Pearson 相关性和约 45 的 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> 中值 GDT 仅表明 ADK 的封闭到开放转变的平均结构精度。然而，总体最佳 GDTs 中值约为 50。由于只有一半的 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>α</mi></mrow><annotation encoding="application/x-tex">α</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.0037em;">α</span></span></span></span></span><span>﻿</span></span> 碳在其目标位置的很小的半径范围内，观察到的行为不是 FLAPS 的问题，而是由于基础模拟方法的限制[7]。基于粗粒度结构的模型和低分辨率 SAXS 数据中的信息似乎不足以与其他测试用例相同的精度确定分子结构。然而，即使在这些情况下，FLAPS 也能够确定可接受的参数。</blockquote><p id="200f7a78-b343-4e22-86d7-853521a8b6d9" class="">To evaluate FLAPS’s efficiency, we performed comparative grid-search optimizations where we found superior performance of FLAPS for all considered protein systems (Supplementary section
‘Comparison to grid search’).</p><blockquote id="c948318f-c3c4-4985-a5fe-da11755a3b0f" class="">为了评估 FLAPS 的效率，我们进行了网格搜索比较优化，我们发现 FLAPS 对所有考虑的蛋白质系统都有优越的性能（补充部分“与网格搜索的比较”）。</blockquote><figure id="bf8c89e9-2d44-4396-a10c-b1626ff2998c" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%208.png"><img style="width:585px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%208.png"/></a></figure><p id="e56bc253-c93a-44ee-a109-6cfddafa0f12" class=""><mark class="highlight-gray">Fig. 5 | Representative structures from global best simulations. a,b, LAO protein (seed 1790954) (a) and ADK (seed 1795691) (b). Structures with maximum GDT are shown (coloured) and are almost identical to the respective target states (grey). The colouring indicates the displacement of each alpha carbon in the simulated structure with respect to the target state. The average alpha-carbon displacement in each coloured structure with respect to each grey structure is given. Structures are visualized with PyMOL[58].</mark></p><h2 id="ce042da4-1648-4c51-85e5-32d72dbf198e" class="">Discussion</h2><p id="e0908ac8-631b-4776-a943-6bff1dbfbf1e" class="">The inverse problem of reconstructing molecular structures from low-resolution SAXS data is still unsolved. Biomolecular simulations are among the most powerful tools for eliminating the arising
ambiguity and access the valuable structural information content of such data. However, data-assisted simulations rely on MD parameters, where, most importantly, experimental information must be weighted accurately with respect to the physical model.</p><blockquote id="7cedfb06-2f9d-4304-8052-9c15fc4a3ca8" class="">从低分辨率 SAXS 数据重建分子结构的逆问题仍未解决。生物分子模拟是消除产生的歧义和获取此类数据的有价值的结构信息的最强大工具之一。然而，数据辅助模拟依赖于 MD 参数，其中最重要的是，实验信息必须相对于物理模型准确加权。</blockquote><p id="4ef30628-60b5-43c3-8a23-ccc3e55df2bd" class="">Here, we have shown how computational intelligence can be used to systematically explore MD parameter spaces and optimize the performance of complex physics-based simulation techniques.
We introduced FLAPS, a data-driven solution for a fully automatic and reproducible parameter search based on particle swarms.</p><blockquote id="4a7632c6-5ebb-4f43-bc5c-4ab9b1c1e5c1" class="">在这里，我们展示了如何使用计算智能来系统地探索 MD 参数空间并优化基于物理的复杂模拟技术的性能。我们介绍了 FLAPS，这是一种基于粒子群的全自动且可重复的参数搜索的数据驱动的解决方案。</blockquote><p id="824f8194-ef0e-473b-a080-d4df8448f53e" class="">To identify the best MD parameters for SAXS-guided protein simulations, we designed an OF as an accurate surrogate of simulation quality in terms of physical structures matching the target data. A suitable OF will typically depend on multiple quality features of different scales to equally reflect a data-assisted simulation’s physical plausibility and its agreement with the data. To handle multiple responses in classical PSO, they need to be mapped to a scalar score via multiplication by fixed weights. These additional OF parameters must either be chosen manually (and probably suboptimally) in advance or be absorbed into the search space, resulting in a massive increase in dimensionality. FLAPS solves this problem by intelligently learning OF parameters in the optimization process, avoiding the need to set them as ‘magic numbers’, while reducing the search-space dimensionality to a minimum. Various responses are automatically balanced with respect to each other to enable a meaningful and unbiased comparison on a shared scale. Implemented in FLAPS, our conceptual OF reliably identified useful MD parameters for two different proteins, where we observed convergence of simulated structures to the target state. Owing to the dedicated in situ processing, the algorithm leverages available computing resources and transparently scales from a laptop to current supercomputers.</p><blockquote id="1dbd793a-020b-4b43-8754-acffc3b81901" class="">为了确定 SAXS 引导的蛋白质模拟的最佳 MD 参数，我们根据与目标数据匹配的物理结构设计了一个 OF 作为模拟质量的准确替代物。合适的 OF 通常取决于不同尺度的多个质量特征，以同样反映数据辅助模拟的物理合理性及其与数据的一致性。为了在经典 PSO 中处理多个响应，需要通过乘以固定权重将它们映射到标量分数。这些额外的 OF 参数要么必须提前人工手动选择（可能是次优的），要么被吸收到搜索空间中，从而导致维度的大量增加。FLAPS通过在优化过程中智能地学习OF参数来解决这个问题，避免将它们设置为 &quot;神奇的数字&quot;，同时将搜索空间的维度降到最低。各种反映被自动地相互平衡，以便在一个共同的尺度上进行有意义的和无偏见的比较。在FLAPS实施中，我们的概念OF可靠地确定了两种不同蛋白质的有用的MD参数，我们观察到模拟结构向目标状态的收敛。由于专门的原位处理，该算法充分利用了可用的计算资源，并透明地从笔记本电脑扩展到当前的超级计算机。</blockquote><p id="354f9d3e-3218-44ca-aa22-1d08754381cb" class="">In previous studies, the bias weight is conveniently chosen as the smallest value yielding satisfactory χ2 (refs. [7,8]). This purely data-based selection criterion neglects the physical information provided by the molecular simulations, risking the selection of physically dysfunctional values due to the ill-posedness of the SAXS inverse problem. The flexible composite OF in FLAPS allows us to include multiple selection criteria, yielding a direct surrogate of simulation quality in terms of not only data conformity but also the physical plausibility of simulated structures. In contrast to grid search, the optimization does not rely on a predefined list of manually chosen and a priori fixed candidates. This circumvents either missing an optimum, spanning large search spaces due to fine-grained grids, or both. Instead, a directional search led by the most current best solutions provides guidance in the selection process and a valid context for meaningful interpretation of multi-layer
criteria. Additionally, the compute time to find useful parameters is decreased by reducing the dimensionality.</p><blockquote id="31b9f1cf-f085-41df-a457-bf63727d2861" class="">在以前的研究中，偏差权重被方便地选择为产生令人满意的 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>χ</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">χ2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">χ</span><span class="mord">2</span></span></span></span></span><span>﻿</span></span> 的最小值（参考文献 [7,8]) 。这种纯粹基于数据的选择标准忽略了分子模拟提供的物理信息，由于 SAXS 逆问题的不适定性，冒着选择物理功能失调值的风险。FLAPS 中的灵活组合 OF 允许我们包含多个选择标准，从而在数据一致性和模拟结构的物理合理性方面直接替代模拟质量。与网格搜索相比，优化不依赖于手动选择的预定义列表和先验固定候选者。这避免了由于细粒度网格而错过最佳、跨越大型搜索空间，或两者兼而有之。相反，由最新的最佳解决方案领导的定向搜索为选择过程提供了指导，并为有意义地解释多层标准提供了有效的上下文。此外，通过减少维度来减少寻找有用参数的计算时间。</blockquote><p id="8fdaf7fd-e0d1-4187-85c4-77036a8e9918" class="">FLAPS can be transferred easily to optimize other life-sciences applications, such as simulation methods incorporating data from various experimental sources[11]. In a more general context, it solves the problem of weighting different contributions of composite OFs in multi-response optimization. Such problems often occur in industrial manufacturing, processing and design, where manually chosen weights are commonly used[32,33]. In FLAPS, OF parameters can be learned in a self-improving manner according to any desired scheme, for example, standardization, rescaling, mean normalization or a relative weighting after one of the aforementioned steps. FLAPS shows how computational-intelligence concepts can be harnessed successfully for practical optimization problems at the forefront of life sciences.</p><blockquote id="43507803-2406-447b-b7ff-2c18c4e849ae" class="">FLAPS 可以轻松转换以优化其它生命科学应用，例如结合来自各种实验来源的数据的模拟方法[11]。在更广泛的情况下，它解决了在多响应优化中加权复合 OFs 的不同贡献的问题。此类问题经常出现在工业制造、加工和设计中，通常使用人工手动选择权重[32,33]。在 FLAPS 中，OF 参数可以根据任何所需的方案以自我改进的方式学习，例如，标准化、重新缩放、均值归一化或在上述步骤之一之后的相对加权。FLAPS 展示了如何成功利用计算智能概念解决生命科学前沿的实际优化问题。</blockquote><h2 id="4284660b-3655-4d02-8e55-47952d30926e" class="">Methods</h2><p id="655096c5-54b8-43ee-bb10-d241ce0bff64" class="">SAXS-guided protein simulations. SAXS-guided protein simulations were set up as described in ref. [7] and run with the MD engine GROMACS 5, including the scattering-guided MD extension[12].  Simulation input files are available on GitHub (<a href="http://github.com/FLAPS-NMI/FLAPS-sim_setups">github.com/FLAPS-NMI/FLAPS-sim_setups</a>). We used the popular CRYSOL software[48] in default mode to compute artificial SAXS intensities from known initial and target structures. Each intensity contained 700 equidistant data points up to a momentum transfer of <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>q</mi><mo>=</mo><mn>0.35</mn><mtext>Å−</mtext></mrow><annotation encoding="application/x-tex">q = 0.35 Å−</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">q</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.72777em;vertical-align:-0.08333em;"></span><span class="mord">0.35Å−</span></span></span></span></span><span>﻿</span></span>1. During a simulation, the SAXS intensities of simulated structures are calculated internally in GROMACS[7,12] using the Debye equation[50]with amino-acid scattering factors corrected for displaced solvent[51]. The CRYSOL intensities were rescaled such that the extrapolation of the forward scattering matches the internally calculated GROMACS value.
Uncertainties were modelled as described in ref. [49]. Target SAXS data were calculated as the difference of rescaled CRYSOL intensities, including uncertainties, of known initial and target structures. The initial structure’s rescaled CRYSOL intensity, including uncertainties, served as the absolute reference scattering in the simulations[7]. We included 17 data points selected as the difference curve’s local extrema and interjacent points centred between each two extrema. For both
proteins, this corresponds well to the number of independent Shannon channels giving the number of independent data points in a SAXS curve[10].</p><blockquote id="e35ce56f-06c1-477a-93a9-71bb75ab2c2a" class="">SAXS 引导的蛋白质模拟。如参考文献[7]中所述设置 SAXS 引导的蛋白质模拟。与 MD 引擎 GROMACS 5 一起运行，包括散射引导的 MD 扩展[12]。模拟输入文件可在 GitHub (<a href="http://github.com/FLAPS-NMI/FLAPS-sim_setups">github.com/FLAPS-NMI/FLAPS-sim_setups</a>) 上获得。我们在默认模式下使用流行的 CRYSOL 软件 [48] 从已知的初始结构和目标结构计算人工 SAXS 强度。每个强度包含 700 个等距数据点，直到动量转移 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>q</mi><mo>=</mo><mn>0.35</mn><msup><mover accent="true"><mi>A</mi><mo>˚</mo></mover><mrow><mo>−</mo><mn>1</mn></mrow></msup></mrow><annotation encoding="application/x-tex">q = 0.35 Å^{-1}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">q</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.9467699999999999em;vertical-align:0em;"></span><span class="mord">0.35</span><span class="mord"><span class="mord accent"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.9467699999999999em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="mord mathnormal">A</span></span><span style="top:-3.25233em;"><span class="pstrut" style="height:3em;"></span><span class="accent-body" style="left:-0.23611em;"><span class="mord">˚</span></span></span></span></span></span></span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mtight">1</span></span></span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>。在模拟过程中，模拟结构的SAXS强度在GROMACS[7,12]内部计算使用Debye方程[50]，并对置换溶剂的氨基酸散射系数进行校正[51]。CRYSOL的强度被重新调整，使正向散射的外推值与GROMACS的内部计算值一致。不确定因素的建模如参考文献[49]所述。目标 SAXS 数据计算是已知初始结构和目标结构的重新调整 CRYSOL 强度（包括不确定性）的差异。初始结构重新调整后的 CRYSOL 强度（包括不确定性）在模拟中用作绝对参考散射[7]。我们将17 个数据点包括选择作为差异曲线的局部极值和以每两个极值为中心的中间点。对于这两种蛋白质，这很好地对应于独立香农通道的数量，给出了 SAXS 曲线中独立数据点的数量[10]。</blockquote><p id="d7eca8c7-9d00-4fe7-9e5a-389d341d0386" class="">Root-mean-square deviation. In addition to the GDT analysis presented in the main text, we considered the more common root-mean-square deviation (RMSD) as a structural similarity measure. The results are presented in Supplementary Tables 1–4.</p><blockquote id="9353ee58-a4ec-4961-a74c-dd0dd281001e" class="">均方根偏差。除了正文中介绍的 GDT 分析之外，我们还考虑将更常见的均方根偏差 (RMSD) 作为结构相似性度量。结果列于补充表 1-4 中。 </blockquote><p id="cf0ad769-5269-4dfb-85e1-f14488f85b9f" class="">
</p><figure id="514bf761-5a38-4231-adce-461c4ac04e36" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%209.png"><img style="width:324px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%209.png"/></a></figure><p id="31d582c2-b0f7-46e0-bb8d-171909802047" class=""><mark class="highlight-gray">Eq.(4)</mark></p><p id="b7e3e759-d4ff-438f-b5cb-e55fb1834b6e" class="">                              </p><p id="8b513387-bd1b-4268-96ce-322f7df7de82" class="">The RMSD is the minimal mass-weighted average distance between N atoms (usually backbone or alpha carbon) of two superimposed structures over all possible spatial translations and rotations,</p><blockquote id="5731b174-8cdd-4934-82e6-335765711d4d" class="">RMSD 是在所有可能的空间平移和旋转中，两个叠加结构的 N 原子（通常是骨架或 α 碳）之间的最小质量加权平均距离。</blockquote><p id="9583855f-251c-45c4-81c6-25cab0894f0b" class="">where M = ΣNi=1 mi and mi is the mass of atom i. ri and ri,0 are the positions of atom i in the mobile and reference structure, respectively. Holo/apo LAO protein and open/closed ADK have alpha-carbon RMSD values of 4.7 Å and 7.1 Å, respectively.</p><blockquote id="e5300bb2-9b58-4318-b7ce-eab1d98055af" class="">其中，<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>=</mo><mrow><msubsup><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>N</mi></msubsup><msub><mi>m</mi><mi>i</mi></msub></mrow></mrow><annotation encoding="application/x-tex">m={\sum_{i=1}^Nm_{i}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal">m</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:1.2809409999999999em;vertical-align:-0.29971000000000003em;"></span><span class="mord"><span class="mop"><span class="mop op-symbol small-op" style="position:relative;top:-0.0000050000000000050004em;">∑</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.981231em;"><span style="top:-2.40029em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span><span class="mrel mtight">=</span><span class="mord mtight">1</span></span></span></span><span style="top:-3.2029em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.10903em;">N</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.29971000000000003em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord"><span class="mord mathnormal">m</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span> 和<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>m</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">m_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.58056em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal">m</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>是原子<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi></mrow><annotation encoding="application/x-tex">i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.65952em;vertical-align:0em;"></span><span class="mord mathnormal">i</span></span></span></span></span><span>﻿</span></span>的质量。<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>r</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex"> r_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.58056em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.02778em;">r</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:-0.02778em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>和<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>r</mi><mrow><mi>i</mi><mo separator="true">,</mo><mn>0</mn></mrow></msub></mrow><annotation encoding="application/x-tex">r_{i,0}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.02778em;">r</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.02778em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span><span class="mpunct mtight">,</span><span class="mord mtight">0</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>﻿</span></span>分别是原子<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi></mrow><annotation encoding="application/x-tex">i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.65952em;vertical-align:0em;"></span><span class="mord mathnormal">i</span></span></span></span></span><span>﻿</span></span>在移动结构和参考结构中的位置。 Holo/apo LAO蛋白和开放/封闭ADK的α-碳RMSD值分别为 4.7 Å 和 7.1 Å。</blockquote><p id="cf8e504a-ea66-45e9-83e4-6c5f90f85ab4" class="">A disadvantage is that RMSD correlates strongly with the largest displacement between two structures, and small numbers of displaced atoms induce large changes. We use GDT[40,52,53]
as the main target metric as it more accurately accounts for local misalignments.</p><blockquote id="3d480a89-d41c-4f7d-ac25-605ccc276cce" class="">缺点是RMSD与两个结构之间的最大位移密切相关，少量的位移原子会引起很大的变化。 我们使用GDT[40,52,53]作为主要的目标指标，因为它更准确地说明了局部错位。</blockquote><p id="a5c0d707-29c5-4ae4-b4ce-02116f6df851" class="">Implementation. FLAPS is implemented as a stand-alone solver in Hyppopy, a Python-based hyperparameter optimization package available at <a href="https://github/">https://github</a>.com/MIC-DKFZ/Hyppopy. Hyppopy provides tools for blackbox optimization. It has a simple, unified application programming interface (API) that can be used to access a collection of solver libraries. Our implementation of FLAPS is available on GitHub[54,55]. We implemented a Message-Passing-Interface (MPI)-parallel
version of the code using a sophisticated parallelization architecture as described in Supplementary Fig. 13. Available compute nodes comprising a given number of processors are divided into blocks, each of which corresponds to one particle in the swarm. Within one block, the simulation itself runs on a single core, while all the other cores process the generated frames in the trajectory on the fly. This results in a massive reduction in runtime.</p><blockquote id="5844ee1b-5472-4d47-b71c-fcc647fb398f" class="">执行。 FLAPS 在 Hyppopy 中作为独立求解器实现，Hyppopy 是一个基于 Python 的超参数优化包，可在 <a href="https://github/">https://github</a>.com/MIC-DKFZ/Hyppopy获得。Hyppopy 提供了黑盒优化工具。它有一个简单、统一的应用程序编程接口 (API)，可用于访问一组求解器库。我们的 FLAPS 实现可在 GitHub[54,55] 上找到。如补充图 13 所述，我们使用复杂的并行化架构实现了代码的消息传递接口（MPI）并行版本。包含给定数量处理器的可用计算节点被划分为块，每个块对应一个群体中的粒子。在一个块内，模拟本身在单个核上运行，而所有其它核在运行中处理轨迹中生成的帧。这导致运行时间大幅减少。</blockquote><p id="946afb80-caa2-4678-ae13-9afe6b2f83ef" class="">The experiments were run on the ForHLR II cluster system located at the Steinbuch Centre for Computing at Karlsruhe Institute of Technology. The system comprises 1,152 thin, that is, solely central processing unit (CPU)-based, compute nodes. Each node is equipped with two 10-core Intel Xeon E5-2660 v3 Haswell CPUs at 3.3 GHz, 64 GB of DDR3 main memory and 4x Mellanox 100-Gbit EDR InfiniBand links. The software packages used were a RHEL Linux with kernel version 4.18.0 and Python 3.6.8.</p><blockquote id="37d92224-9c97-4d36-b84d-fbd261b6ce6d" class="">实验在位于卡尔斯鲁厄理工学院 Steinbuch 计算中心的 ForHLR II 集群系统上运行。该系统包括 1,152 个瘦计算节点，即仅基于中央处理单元 (CPU) 的计算节点。每个节点配备两个 3.3 GHz 的 10 核 Intel Xeon E5-2660 v3 Haswell CPU、64 GB DDR3 主内存和 4 个 Mellanox 100-Gbit EDR InfiniBand 链路。使用的软件包是 RHEL Linux的内核版本 4.18.0 和 Python 3.6.8 。</blockquote><p id="bf9316af-2f8d-4902-aa44-1ce2017ee458" class="">Each run used 51 compute nodes (1,020 cores in total). Owing to the magnitude of metadata and I/O operations, we used a private on-demand file system (BeeGFS On-Demand) with a stripe count of 1, where one node was reserved for the metadata server[56]. Each block in the underlying simulator-worker scheme consisted of five nodes, that is, 100 cores (one simulator, 99 workers). Each run cost ~40,000 CPU hours. For the presented application, we used cognitive acceleration coefficient<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ϕ</mi><mn>1</mn></msub><mo>=</mo><mn>2.0</mn></mrow><annotation encoding="application/x-tex"> ϕ_1= 2.0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">ϕ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.30110799999999993em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">1</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">2.0</span></span></span></span></span><span>﻿</span></span> and social acceleration coefficient<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ϕ</mi><mn>2</mn></msub><mo>=</mo><mn>1.5</mn></mrow><annotation encoding="application/x-tex"> ϕ_2= 1.5</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">ϕ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.30110799999999993em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">1.5</span></span></span></span></span><span>﻿</span></span> in the particle update (Algorithm 1). The complete set-up, including all PSO hyperparameters used, is available on GitHub[57].</p><blockquote id="85b6c910-5109-443a-97c9-18c1b7e21f87" class="">每次运行使用 51 个计算节点（总共 1,020 个内核）。由于元数据和 I/O 操作的规模，我们使用了条带计数为 1 的私有按需文件系统 (BeeGFS On-Demand)，其中为元数据服务器保留了一个节点[56]。底层simulator-worker 方案中的每个block 由5 个节点组成，即100 个核心（一个模拟器，99 个worker）。每次运行花费约 40,000 个 CPU 小时。对于所提出的应用程序，我们在粒子更新（算法 1）中使用了认知加速系数 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ϕ</mi><mn>1</mn></msub><mo>=</mo><mn>2.0</mn></mrow><annotation encoding="application/x-tex"> ϕ_1= 2.0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">ϕ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.30110799999999993em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">1</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">2.0</span></span></span></span></span><span>﻿</span></span> 和社会加速系数 <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>ϕ</mi><mn>2</mn></msub><mo>=</mo><mn>1.5</mn></mrow><annotation encoding="application/x-tex"> ϕ_2= 1.5</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">ϕ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.30110799999999993em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">1.5</span></span></span></span></span><span>﻿</span></span>。完整的设置，包括使用的所有 PSO 超参数，可在 GitHub[57] 上找到。</blockquote><h2 id="0975005c-8c36-4471-bf4a-1620c3cb7394" class="">Data availability</h2><p id="b6ee4a79-adef-46f8-8fb1-976808f790e2" class="">The software for the metaheuristic molecular dynamics parameter optimization of SAXS-guided structure-based protein simulations used in this work54,55 is publicly available at <a href="https://github.com/FLAPS-NMI?tab=repositories">https://github.com/FLAPS-NMI?tab=repositories</a>. A minimal dataset to reproduce the presented results[57] is publicly available at <a href="https://github/">https://github</a>.com/FLAPS-NMI/FLAPS sim_setups/releases/tag/v1.0 and published under the Creative Commons Attribution 4.0 International Public License.</p><blockquote id="76d7fcfc-8967-4c4d-b8c0-9ebfd4414600" class="">本工作中使用的 SAXS 引导的基于结构的蛋白质模拟的元启发式分子动力学参数优化软件[ 54,55] 可在 <a href="https://github.com/FLAPS-NMI?tab=repositories">https://github.com/FLAPS-NMI?tab=repositories</a> 上获得。重现所呈现结果的最小数据集[57] 可在 <a href="https://github/">https://github</a> com/FLAPS-NMI/FLAPS-sim_setups/releases/tag/v1.0 上公开获得。并根据知识共享署名 4.0 国际公共许可证发布。</blockquote><h2 id="abc336ef-beff-471d-9560-346a993a582b" class="">Code availability</h2><p id="ee11ab2d-dd69-4cbe-b6b4-25098f1ff554" class="">All code used in this work is publicly available at <a href="https://github.com/">https://github.com/</a>FLAPS-NMI?tab=repositories and published under the New BSD Licence. The MPI-parallelized FLAPS solver is implemented in Optunity, a hyperparameter tuning package for Python. Our extended version of Optunity55
is available at <a href="https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0">https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0</a> and integrated into Hyppopy, a Python-based toolbox for blackbox optimization. Our extended version of Hyppopy[54]
is available at <a href="https://github.com/FLAPS-NMI/">https://github.com/FLAPS-NMI/</a>FLAPS-Hyppopy/releases/tag/v1.0.</p><blockquote id="ef54cd68-12e4-413e-af0b-e7843c4dfd0d" class="">这项工作中使用的所有代码都可以在 <a href="https://github.com/FLAPS-NMI?tab=repositories">https://github.com/FLAPS-NMI?tab=repositories</a> 上公开获得，并在新 BSD 许可证下发布。 MPI 并行化 FLAPS 求解器在 Optunity 中实现，Optunity 是 Python 的超参数调整包。我们的 Optunity[55] 扩展版本可在 <a href="https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0">https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0</a> 获得，并集成到基于 Python 的黑盒优化工具箱 Hyppopy 中。我们的 Hyppopy[54] 扩展版可在 <a href="https://github.com/FLAPS-NMI/FLAPS-Hyppopy/releases/tag/v1.0">https://github.com/FLAPS-NMI/FLAPS Hyppopy/releases/tag/v1.0</a> 获得。</blockquote><h2 id="6563f67b-19bb-4464-a444-6800eaf7e045" class="">References</h2><ol type="1" id="eb6535e4-6a26-4fbf-9a77-16795bc5f4ab" class="numbered-list" start="1"><li>Selkoe, D. Folding proteins in fatal ways. Nature 426, 900–904 (2003).</li></ol><ol type="1" id="60417bc1-69f2-431e-9c0f-b3086b920717" class="numbered-list" start="2"><li>Selkoe, D. J. Cell biology of protein misfolding: the examples of Alzheimer’s and Parkinson’s diseases. Nat. Cell Biol. 6, 1054–1061 (2004).</li></ol><ol type="1" id="37e725b7-d4a2-43bc-91b8-07cca2184503" class="numbered-list" start="3"><li>Soto, C. Unfolding the role of protein misfolding in neurodegenerative diseases. Nat. Rev. Neurosci. 4, 49–60 (2003).</li></ol><ol type="1" id="46205c3d-6fc0-4c9f-ae44-ea9e9187f139" class="numbered-list" start="4"><li>Mukherjee, A., Morales-Scheihing, D., Butler, P. C. &amp; Soto, C. Type 2 diabetes as a protein misfolding disease. Trends Mol. Med. 21, 439–449 (2015).</li></ol><ol type="1" id="6375c8b3-82c9-4f5d-a656-3f8dc45d72e0" class="numbered-list" start="5"><li>Dobson, C. M. Protein-misfolding diseases: getting out of shape. Nature 418,729–730 (2002).</li></ol><ol type="1" id="9c7ca279-2276-47d9-b5c1-f4ecdd393511" class="numbered-list" start="6"><li>Karaca, E., Rodrigues, J. P., Graziadei, A., Bonvin, A. M. &amp; Carlomagno, T.M3: an integrative framework for structure determination of molecular machines. Nat. Methods 14, 897–902 (2017).</li></ol><ol type="1" id="9f2e1fd8-78bd-4958-90fe-5f3a33885766" class="numbered-list" start="7"><li>Weiel, M., Reinartz, I. &amp; Schug, A. Rapid interpretation of small-angle X-ray scattering data. PLoS Comput. Biol. 15, e1006900 (2019).</li></ol><ol type="1" id="ae811d77-ca67-461b-b877-8c62dd835887" class="numbered-list" start="8"><li>Hermann, M. R. &amp; Hub, J. S. SAXS-restrained ensemble simulations of intrinsically disordered proteins with commitment to the principle of maximum entropy. J. Chem. Theory Comput. 15, 5103–5115 (2019).</li></ol><ol type="1" id="b0be9ab0-a094-4d84-b678-ef378bfef9c0" class="numbered-list" start="9"><li>Chen, P. et al. Combined small-angle X-ray and neutron scattering restraints in molecular dynamics simulations. J. Chem. Theory Comput. 15, 4687–4698 (2019).</li></ol><ol type="1" id="e9565981-4f2c-485d-b803-d886d1bb4e73" class="numbered-list" start="10"><li>Chen, P. &amp; Hub, J. S. Interpretation of solution X-ray scattering by explicit-solvent molecular dynamics. Biophys. J. 108, 2573–2584(2015).</li></ol><ol type="1" id="d52d066a-ad25-453c-85ac-eecc61be8f51" class="numbered-list" start="11"><li>Whitford, P. C. et al. Excited states of ribosome translocation revealed through integrative molecular modeling. Proc. Natl Acad. Sci. USA 108,18943–18948 (2011).</li></ol><ol type="1" id="49f710c1-5ae2-4d2b-b6f8-0e67299a60f1" class="numbered-list" start="12"><li>Björling, A., Niebling, S., Marcellini, M., van der Spoel, D. &amp; Westenhoff, S. Deciphering solution scattering data with experimentally guided molecular dynamics simulations. J. Chem. Theory Comput. 11, 780–787 (2015).</li></ol><ol type="1" id="d82dfaa0-b23a-42ed-b212-050345519e14" class="numbered-list" start="13"><li>Shevchuk, R. &amp; Hub, J. S. Bayesian refinement of protein structures and ensembles against SAXS data using molecular dynamics. PLoS Comput. Biol.13, e1005800 (2017).</li></ol><ol type="1" id="c3425827-d499-445d-b3ab-7b438e8ca716" class="numbered-list" start="14"><li>Hummer, G. &amp; Köfinger, J. Bayesian ensemble refinement by replica simulations and reweighting. J. Chem. Phys. 143, 243150 (2015).</li></ol><ol type="1" id="3d857413-3609-40a0-ae6f-c6a07cae61f4" class="numbered-list" start="15"><li>Hub, J. S. Interpreting solution X-ray scattering data using molecular simulations. Curr. Opin. Struct. Biol. 49, 18–26 (2018).</li></ol><ol type="1" id="adf9c6f8-5328-465e-81d9-520c2f9e3110" class="numbered-list" start="16"><li>Kennedy, J. &amp; Eberhart, R. Particle swarm optimization. In Proc. ICNN’95—International Conference on Neural Networks Vol. 4, 1942–1948 (IEEE, 1995);
<a href="https://doi.org/10.1109/ICNN.1995.488968">https://doi.org/10.1109/ICNN.1995.488968</a></li></ol><ol type="1" id="00b899b4-fb7c-443b-b344-f15f8056f7ac" class="numbered-list" start="17"><li>Kennedy, J. The particle swarm: social adaptation of knowledge. In Proc. 1997 IEEE International Conference on Evolutionary Computation (ICEC’97) 303–308 (IEEE, 1997); <a href="https://doi.org/10.1109/ICEC.1997.592326">https://doi.org/10.1109/ICEC.1997.592326</a></li></ol><ol type="1" id="f3c1ba3e-e4fe-4779-98e9-026120faf913" class="numbered-list" start="18"><li>Shi, Y. &amp; Eberhart, R. A modified particle swarm optimizer. In Proc. 1998 IEEE International Conference on Evolutionary Computation. IEEE World Congress on Computational Intelligence (cat. no. 98TH8360) 69–73 (IEEE,1998); <a href="https://doi.org/10.1109/ICEC.1998.699146">https://doi.org/10.1109/ICEC.1998.699146</a></li></ol><ol type="1" id="a881887f-1638-4cb8-b453-5d9bdd62cccd" class="numbered-list" start="19"><li>Clerc, M. &amp; Kennedy, J. The particle swarm—explosion, stability and convergence in a multidimensional complex space. IEEE Trans. Evol. Comput.6, 58–73 (2002).</li></ol><ol type="1" id="15b0f19a-1651-4b3c-b51e-b58a43c9231e" class="numbered-list" start="20"><li>Parsopoulos, K. E. &amp; Vrahatis, M. N. Recent approaches to global optimization problems through particle swarm optimization. Nat. Comput. 1,235–306 (2002).</li></ol><ol type="1" id="c10f21ad-61de-4bcb-a20a-e83929153e37" class="numbered-list" start="21"><li>Jordehi, A. R. Particle swarm optimisation for dynamic optimisation problems: a review. Neural Comput. Appl. 25, 1507–1516 (2014).</li></ol><ol type="1" id="deb6f2bf-3b00-48e1-aa91-890cf74ee118" class="numbered-list" start="22"><li>Blackwell, T. in Evolutionary Computation in Dynamic and Uncertain Environments (eds Yang, S. et al.) 29–49 (Springer, 2007); <a href="https://doi/">https://doi</a>.org/10.1007/978-3-540-49774-5_2</li></ol><ol type="1" id="fee9faa2-94eb-4731-9150-38e1b19933ea" class="numbered-list" start="23"><li>Stanley, K. O., Clune, J., Lehman, J. &amp; Miikkulainen, R. Designing neural networks through neuroevolution. Nat. Mach. Intell. 1, 24–35 (2019).</li></ol><ol type="1" id="17fed1f4-3e67-4b36-b9b6-6d145a873e60" class="numbered-list" start="24"><li>Taherkhani, M. &amp; Safabakhsh, R. A novel stability-based adaptive inertia weight for particle swarm optimization. Appl. Soft Comput. 38,281–295 (2016).</li></ol><ol type="1" id="11ea2433-ebc1-423a-a0f3-ab90b6c40800" class="numbered-list" start="25"><li>Eberhart, R. C. &amp; Shi, Y. Comparing inertia weights and constriction factors in particle swarm optimization. In Proc. 2000 Congress on Evolutionary Computation. CEC00 (cat. no. 00TH8512) Vol. 1, 84–88 (IEEE, 2000);<a href="https://doi.org/10.1109/CEC.2000.870279">https://doi.org/10.1109/CEC.2000.870279</a></li></ol><ol type="1" id="0d266271-c70a-4bec-b8e7-d622bb7aa7eb" class="numbered-list" start="26"><li>Pedersen, M. E. H. &amp; Chipperfield, A. J. Simplifying particle swarm optimization. Appl. Soft Comput. 10, 618–628 (2010).</li></ol><ol type="1" id="7cac1ba4-d695-4d2a-9666-22b51402131a" class="numbered-list" start="27"><li>Meissner, M., Schmuker, M. &amp; Schneider, G. Optimized particle swarm optimization (OPSO) and its application to artificial neural network training.BMC Bioinformatics 7, 125 (2006).</li></ol><ol type="1" id="2f6d5a57-d86d-47b9-a47f-87aa955b7448" class="numbered-list" start="28"><li>Nobile, M. S. et al. Fuzzy self-tuning PSO: a settings-free algorithm for global optimization. Swarm Evol. Comput. 39, 70–85 (2018).</li></ol><ol type="1" id="a74bb90f-6ad4-45f6-8c98-9ac1fdd619e1" class="numbered-list" start="29"><li>Poli, R. Analysis of the publications on the applications of particle swarm optimisation. J. Artif. Evol. Appl. 2008, 685175 (2008).</li></ol><ol type="1" id="8ceb6d77-f50a-4efa-96ef-d93365c86429" class="numbered-list" start="30"><li>Sengupta, S., Basak, S. &amp; Peters, R. A. Particle swarm optimization: a survey of historical and recent developments with hybridization perspectives. Mach.Learn. Knowledge Extraction 1, 157–191 (2019).</li></ol><ol type="1" id="0215636b-adc7-4e49-8d72-ff5342dc4d64" class="numbered-list" start="31"><li>Navalertporn, T. &amp; Afzulpurkar, N. V. Optimization of tile manufacturing process using particle swarm optimization. Swarm Evol. Comput. 1,97–109 (2011).</li></ol><ol type="1" id="6d35edb4-ca51-4b74-b89f-7f8da52c8914" class="numbered-list" start="32"><li>Pawar, P., Rao, R. &amp; Davim, J. Multiobjective optimization of grinding process parameters using particle swarm optimization algorithm. Mater. Manuf. Process. 25, 424–431 (2010).</li></ol><ol type="1" id="b4c49e42-1ff6-4f04-9e9e-6b253a273fc3" class="numbered-list" start="33"><li>Ma, C. &amp; Qu, L. Multiobjective optimization of switched reluctance motors based on design of experiments and particle swarm optimization. IEEE Trans. Energy Conver. 30, 1144–1153 (2015).</li></ol><ol type="1" id="882fed13-5dc4-4826-9431-e896f382df29" class="numbered-list" start="34"><li>Zhang, C., Chen, Z., Mei, Q. &amp; Duan, J. Application of particle swarm optimization combined with response surface methodology to transverse flux permanent magnet motor optimization. IEEE Trans. Magn. 53,8113107 (2017).</li></ol><ol type="1" id="d15b2d42-452d-4c82-9480-78b693af5c22" class="numbered-list" start="35"><li>Ioffe, S. &amp; Szegedy, C. Batch normalization: accelerating deep network training by reducing internal covariate shift. In Proc. 32nd International Conference on Machine Learning Vol. 37 (eds. Bach, F. &amp; Blei, D.) 448–456 (PMLR, 2015).</li></ol><ol type="1" id="c7989c8f-4ba8-4dd0-862d-bfef55d790b4" class="numbered-list" start="36"><li>Onuchic, J. N. &amp; Wolynes, P. G. Theory of protein folding. Curr. Opin. Struct. Biol. 14, 70–75 (2004).</li></ol><ol type="1" id="cc252d2c-0be1-49db-b9ca-cb5654c9ec41" class="numbered-list" start="37"><li>Schug, A. &amp; Onuchic, J. N. From protein folding to protein function and biomolecular binding by energy landscape theory. Curr. Opin. Pharmacol. 10,709–714 (2010).</li></ol><ol type="1" id="ad25bf71-9194-4b36-9a4e-12c2d04811a3" class="numbered-list" start="38"><li>Alford, R. F. et al. The Rosetta all-atom energy function for macromolecular modeling and design. J. Chem. Theory Comput. 13,3031–3048 (2017).</li></ol><ol type="1" id="1df558e5-f6ef-4b95-a93a-7634a95004bf" class="numbered-list" start="39"><li>Leman, J. K. et al. Macromolecular modeling and design in Rosetta: recent methods and frameworks. Nat. Methods 17, 665–680 (2020).</li></ol><ol type="1" id="4d355e7f-40e3-4ffd-ad18-9c714a54c753" class="numbered-list" start="40"><li>Zemla, A. LGA: a method for finding 3D similarities in protein structures. Nucleic Acids Res. 31, 3370–3374 (2003).</li></ol><ol type="1" id="1d9658a1-e1f0-4cec-b255-d4e773dfe605" class="numbered-list" start="41"><li>Moult, J., Fidelis, K., Kryshtafovych, A., Schwede, T. &amp; Tramontano, A.Critical assessment of methods of protein structure prediction (CASP)–round XII. Proteins 86, 7–15 (2018).</li></ol><ol type="1" id="f0e9873c-2d6f-4690-889c-4f1e370146d3" class="numbered-list" start="42"><li>Ames, G. F.-L. Bacterial periplasmic transport systems: structure, mechanism and evolution. Annu. Rev. Biochem. 55, 397–425 (1986).</li></ol><ol type="1" id="13741a20-a8c5-4fa3-abc4-a137e4885f19" class="numbered-list" start="43"><li>Berman, H. M. et al. The Protein Data Bank. Nucleic Acids Res. 28,235–242 (2000).</li></ol><ol type="1" id="830dcf4e-019e-4bdd-82cd-bcd984c66558" class="numbered-list" start="44"><li>Oh, B.-H. et al. Three-dimensional structures of the periplasmic lysine/arginine/ornithine-binding protein with and without a ligand. J. Biol. Chem.268, 11348–11355 (1993).</li></ol><ol type="1" id="05a179d3-407a-4454-bdac-8dfa1f956a1d" class="numbered-list" start="45"><li>Müller, C., Schlauderer, G., Reinstein, J. &amp; Schulz, G. E. Adenylate kinase motions during catalysis: an energetic counterweight balancing substrate binding. Structure 4, 147–156 (1996).</li></ol><ol type="1" id="10f68c24-5384-4b52-bbe6-180b669d1374" class="numbered-list" start="46"><li>Müller, C. W. &amp; Schulz, G. E. Structure of the complex between adenylate kinase from Escherichia coli and the inhibitor Ap5 A refined at 1.9-Å resolution: a model for a catalytic transition state. J. Mol. Biol. 224,159–177 (1992).</li></ol><ol type="1" id="eb6fb9bd-46fa-4020-8e87-3e12aedc5dc9" class="numbered-list" start="47"><li>Whitford, P. C., Miyashita, O., Levy, Y. &amp; Onuchic, J. N. Conformational transitions of adenylate kinase: switching by cracking. J. Mol. Biol. 366, 1661–1671 (2007).</li></ol><ol type="1" id="e30cb62d-f558-4706-b724-77404d163f8b" class="numbered-list" start="48"><li>Svergun, D., Barberato, C. &amp; Koch, M. H. CRYSOL—a program to evaluate X-ray solution scattering of biological macromolecules from atomic coordinates. J. Appl. Crystallogr. 28, 768–773 (1995).</li></ol><ol type="1" id="9525083a-4276-49da-80a4-13be24324644" class="numbered-list" start="49"><li>Sedlak, S. M., Bruetzel, L. K. &amp; Lipfert, J. Quantitative evaluation of statistical errors in small-angle X-ray scattering measurements. J. Appl. Crystallogr. 50, 621–630 (2017).</li></ol><ol type="1" id="960a1a6c-638d-4335-ad4e-62dce6905535" class="numbered-list" start="50"><li>Debye, P. Zerstreuung von Röntgenstrahlen. Annal. Phys. 351,809–823 (1915).</li></ol><ol type="1" id="2a157ddb-49c1-4da1-946a-38d95d6a2cc0" class="numbered-list" start="51"><li>Yang, S., Park, S., Makowski, L. &amp; Roux, B. A rapid coarse residue-based computational method for X-ray solution scattering characterization of protein folds and multiple conformational states of large protein complexes.Biophys. J. 96, 4449–4463 (2009).</li></ol><ol type="1" id="676fff88-d1f2-48c6-a487-f7ac083530d3" class="numbered-list" start="52"><li>Kufareva, I. &amp; Abagyan, R. in Homology Modeling (eds Orry, A. J. W. &amp; Abagyan, R.) 231–257 (Springer, 2011).</li></ol><ol type="1" id="4ec26413-2f59-43b6-acab-f334a1db9dfb" class="numbered-list" start="53"><li>Modi, V., Xu, Q., Adhikari, S. &amp; Dunbrack Jr, R. L. Assessment of template-based modeling of protein structure in CASP11. Proteins 84,200–220 (2016).</li></ol><ol type="1" id="4037dad5-73b5-4999-bfbf-861c47971d0d" class="numbered-list" start="54"><li>Weiel, M. et al. FLAPS Optunity code repository (FLAPS-NMI@Github/
FLAPS-optunity, 2021); <a href="https://doi.org/10.5281/zenodo.4773992">https://doi.org/10.5281/zenodo.4773992</a>,<a href="https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0">https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0</a></li></ol><ol type="1" id="2d967381-00bb-4023-bda5-602152897d42" class="numbered-list" start="55"><li>Soysal, M. et al. Using on-demand file systems in HPC environments. In 2019 International Conference on High Performance Computing &amp; Simulation (HPCS) 390–398 (IEEE, 2019); <a href="https://doi.org/10.1109/">https://doi.org/10.1109/</a>HPCS48598.2019.9188216</li></ol><ol type="1" id="a21fb0be-45da-43cd-8d95-90970915a7f5" class="numbered-list" start="56"><li>Weiel, M. et al. Minimal dataset repository for reproduction of presented results (FLAPS-NMI@Github/FLAPS-sim_setups, 2021); <a href="https://doi.org/">https://doi.org/</a>10.5281/zenodo.4773999, <a href="https://github.com/FLAPS-NMI/FLAPS-sim_setups/">https://github.com/FLAPS-NMI/FLAPS-sim_setups/</a>releases/tag/v1.0</li></ol><ol type="1" id="5cb38a41-618f-4245-b846-427c50734d0c" class="numbered-list" start="57"><li>The PyMOL Molecular Graphics System, Version 1.8 (Schrödinger, 2015).</li></ol><p id="aceb260a-2c23-4196-96b6-72766b8e3ecb" class="">
</p></div></article></body></html>