<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/><title>åŠ¨æ€ç²’å­ç¾¤ä¼˜åŒ–</title><style>
/* cspell:disable-file */
/* webkit printing magic: print all background colors */
html {
	-webkit-print-color-adjust: exact;
}
* {
	box-sizing: border-box;
	-webkit-print-color-adjust: exact;
}

html,
body {
	margin: 0;
	padding: 0;
}
@media only screen {
	body {
		margin: 2em auto;
		max-width: 900px;
		color: rgb(55, 53, 47);
	}
}

body {
	line-height: 1.5;
	white-space: pre-wrap;
}

a,
a.visited {
	color: inherit;
	text-decoration: underline;
}

.pdf-relative-link-path {
	font-size: 80%;
	color: #444;
}

h1,
h2,
h3 {
	letter-spacing: -0.01em;
	line-height: 1.2;
	font-weight: 600;
	margin-bottom: 0;
}

.page-title {
	font-size: 2.5rem;
	font-weight: 700;
	margin-top: 0;
	margin-bottom: 0.75em;
}

h1 {
	font-size: 1.875rem;
	margin-top: 1.875rem;
}

h2 {
	font-size: 1.5rem;
	margin-top: 1.5rem;
}

h3 {
	font-size: 1.25rem;
	margin-top: 1.25rem;
}

.source {
	border: 1px solid #ddd;
	border-radius: 3px;
	padding: 1.5em;
	word-break: break-all;
}

.callout {
	border-radius: 3px;
	padding: 1rem;
}

figure {
	margin: 1.25em 0;
	page-break-inside: avoid;
}

figcaption {
	opacity: 0.5;
	font-size: 85%;
	margin-top: 0.5em;
}

mark {
	background-color: transparent;
}

.indented {
	padding-left: 1.5em;
}

hr {
	background: transparent;
	display: block;
	width: 100%;
	height: 1px;
	visibility: visible;
	border: none;
	border-bottom: 1px solid rgba(55, 53, 47, 0.09);
}

img {
	max-width: 100%;
}

@media only print {
	img {
		max-height: 100vh;
		object-fit: contain;
	}
}

@page {
	margin: 1in;
}

.collection-content {
	font-size: 0.875rem;
}

.column-list {
	display: flex;
	justify-content: space-between;
}

.column {
	padding: 0 1em;
}

.column:first-child {
	padding-left: 0;
}

.column:last-child {
	padding-right: 0;
}

.table_of_contents-item {
	display: block;
	font-size: 0.875rem;
	line-height: 1.3;
	padding: 0.125rem;
}

.table_of_contents-indent-1 {
	margin-left: 1.5rem;
}

.table_of_contents-indent-2 {
	margin-left: 3rem;
}

.table_of_contents-indent-3 {
	margin-left: 4.5rem;
}

.table_of_contents-link {
	text-decoration: none;
	opacity: 0.7;
	border-bottom: 1px solid rgba(55, 53, 47, 0.18);
}

table,
th,
td {
	border: 1px solid rgba(55, 53, 47, 0.09);
	border-collapse: collapse;
}

table {
	border-left: none;
	border-right: none;
}

th,
td {
	font-weight: normal;
	padding: 0.25em 0.5em;
	line-height: 1.5;
	min-height: 1.5em;
	text-align: left;
}

th {
	color: rgba(55, 53, 47, 0.6);
}

ol,
ul {
	margin: 0;
	margin-block-start: 0.6em;
	margin-block-end: 0.6em;
}

li > ol:first-child,
li > ul:first-child {
	margin-block-start: 0.6em;
}

ul > li {
	list-style: disc;
}

ul.to-do-list {
	text-indent: -1.7em;
}

ul.to-do-list > li {
	list-style: none;
}

.to-do-children-checked {
	text-decoration: line-through;
	opacity: 0.375;
}

ul.toggle > li {
	list-style: none;
}

ul {
	padding-inline-start: 1.7em;
}

ul > li {
	padding-left: 0.1em;
}

ol {
	padding-inline-start: 1.6em;
}

ol > li {
	padding-left: 0.2em;
}

.mono ol {
	padding-inline-start: 2em;
}

.mono ol > li {
	text-indent: -0.4em;
}

.toggle {
	padding-inline-start: 0em;
	list-style-type: none;
}

/* Indent toggle children */
.toggle > li > details {
	padding-left: 1.7em;
}

.toggle > li > details > summary {
	margin-left: -1.1em;
}

.selected-value {
	display: inline-block;
	padding: 0 0.5em;
	background: rgba(206, 205, 202, 0.5);
	border-radius: 3px;
	margin-right: 0.5em;
	margin-top: 0.3em;
	margin-bottom: 0.3em;
	white-space: nowrap;
}

.collection-title {
	display: inline-block;
	margin-right: 1em;
}

.simple-table {
	margin-top: 1em;
	font-size: 0.875rem;
	empty-cells: show;
}
.simple-table td {
	height: 29px;
	min-width: 120px;
}

.simple-table th {
	height: 29px;
	min-width: 120px;
}

.simple-table-header-color {
	background: rgb(247, 246, 243);
	color: black;
}
.simple-table-header {
	font-weight: 500;
}

time {
	opacity: 0.5;
}

.icon {
	display: inline-block;
	max-width: 1.2em;
	max-height: 1.2em;
	text-decoration: none;
	vertical-align: text-bottom;
	margin-right: 0.5em;
}

img.icon {
	border-radius: 3px;
}

.user-icon {
	width: 1.5em;
	height: 1.5em;
	border-radius: 100%;
	margin-right: 0.5rem;
}

.user-icon-inner {
	font-size: 0.8em;
}

.text-icon {
	border: 1px solid #000;
	text-align: center;
}

.page-cover-image {
	display: block;
	object-fit: cover;
	width: 100%;
	max-height: 30vh;
}

.page-header-icon {
	font-size: 3rem;
	margin-bottom: 1rem;
}

.page-header-icon-with-cover {
	margin-top: -0.72em;
	margin-left: 0.07em;
}

.page-header-icon img {
	border-radius: 3px;
}

.link-to-page {
	margin: 1em 0;
	padding: 0;
	border: none;
	font-weight: 500;
}

p > .user {
	opacity: 0.5;
}

td > .user,
td > time {
	white-space: nowrap;
}

input[type="checkbox"] {
	transform: scale(1.5);
	margin-right: 0.6em;
	vertical-align: middle;
}

p {
	margin-top: 0.5em;
	margin-bottom: 0.5em;
}

.image {
	border: none;
	margin: 1.5em 0;
	padding: 0;
	border-radius: 0;
	text-align: center;
}

.code,
code {
	background: rgba(135, 131, 120, 0.15);
	border-radius: 3px;
	padding: 0.2em 0.4em;
	border-radius: 3px;
	font-size: 85%;
	tab-size: 2;
}

code {
	color: #eb5757;
}

.code {
	padding: 1.5em 1em;
}

.code-wrap {
	white-space: pre-wrap;
	word-break: break-all;
}

.code > code {
	background: none;
	padding: 0;
	font-size: 100%;
	color: inherit;
}

blockquote {
	font-size: 1.25em;
	margin: 1em 0;
	padding-left: 1em;
	border-left: 3px solid rgb(55, 53, 47);
}

.bookmark {
	text-decoration: none;
	max-height: 8em;
	padding: 0;
	display: flex;
	width: 100%;
	align-items: stretch;
}

.bookmark-title {
	font-size: 0.85em;
	overflow: hidden;
	text-overflow: ellipsis;
	height: 1.75em;
	white-space: nowrap;
}

.bookmark-text {
	display: flex;
	flex-direction: column;
}

.bookmark-info {
	flex: 4 1 180px;
	padding: 12px 14px 14px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
}

.bookmark-image {
	width: 33%;
	flex: 1 1 180px;
	display: block;
	position: relative;
	object-fit: cover;
	border-radius: 1px;
}

.bookmark-description {
	color: rgba(55, 53, 47, 0.6);
	font-size: 0.75em;
	overflow: hidden;
	max-height: 4.5em;
	word-break: break-word;
}

.bookmark-href {
	font-size: 0.75em;
	margin-top: 0.25em;
}

.sans { font-family: ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol"; }
.code { font-family: "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace; }
.serif { font-family: Lyon-Text, Georgia, ui-serif, serif; }
.mono { font-family: iawriter-mono, Nitti, Menlo, Courier, monospace; }
.pdf .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK JP'; }
.pdf:lang(zh-CN) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK SC'; }
.pdf:lang(zh-TW) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK TC'; }
.pdf:lang(ko-KR) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK KR'; }
.pdf .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK JP'; }
.pdf:lang(zh-CN) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK SC'; }
.pdf:lang(zh-TW) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK TC'; }
.pdf:lang(ko-KR) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK KR'; }
.pdf .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK JP'; }
.pdf:lang(zh-CN) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK SC'; }
.pdf:lang(zh-TW) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK TC'; }
.pdf:lang(ko-KR) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK KR'; }
.pdf .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK JP'; }
.pdf:lang(zh-CN) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK SC'; }
.pdf:lang(zh-TW) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK TC'; }
.pdf:lang(ko-KR) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK KR'; }
.highlight-default {
	color: rgba(55, 53, 47, 1);
}
.highlight-gray {
	color: rgba(120, 119, 116, 1);
	fill: rgba(120, 119, 116, 1);
}
.highlight-brown {
	color: rgba(159, 107, 83, 1);
	fill: rgba(159, 107, 83, 1);
}
.highlight-orange {
	color: rgba(217, 115, 13, 1);
	fill: rgba(217, 115, 13, 1);
}
.highlight-yellow {
	color: rgba(203, 145, 47, 1);
	fill: rgba(203, 145, 47, 1);
}
.highlight-teal {
	color: rgba(68, 131, 97, 1);
	fill: rgba(68, 131, 97, 1);
}
.highlight-blue {
	color: rgba(51, 126, 169, 1);
	fill: rgba(51, 126, 169, 1);
}
.highlight-purple {
	color: rgba(144, 101, 176, 1);
	fill: rgba(144, 101, 176, 1);
}
.highlight-pink {
	color: rgba(193, 76, 138, 1);
	fill: rgba(193, 76, 138, 1);
}
.highlight-red {
	color: rgba(212, 76, 71, 1);
	fill: rgba(212, 76, 71, 1);
}
.highlight-gray_background {
	background: rgba(241, 241, 239, 1);
}
.highlight-brown_background {
	background: rgba(244, 238, 238, 1);
}
.highlight-orange_background {
	background: rgba(251, 236, 221, 1);
}
.highlight-yellow_background {
	background: rgba(251, 243, 219, 1);
}
.highlight-teal_background {
	background: rgba(237, 243, 236, 1);
}
.highlight-blue_background {
	background: rgba(231, 243, 248, 1);
}
.highlight-purple_background {
	background: rgba(244, 240, 247, 0.8);
}
.highlight-pink_background {
	background: rgba(249, 238, 243, 0.8);
}
.highlight-red_background {
	background: rgba(253, 235, 236, 1);
}
.block-color-default {
	color: inherit;
	fill: inherit;
}
.block-color-gray {
	color: rgba(120, 119, 116, 1);
	fill: rgba(120, 119, 116, 1);
}
.block-color-brown {
	color: rgba(159, 107, 83, 1);
	fill: rgba(159, 107, 83, 1);
}
.block-color-orange {
	color: rgba(217, 115, 13, 1);
	fill: rgba(217, 115, 13, 1);
}
.block-color-yellow {
	color: rgba(203, 145, 47, 1);
	fill: rgba(203, 145, 47, 1);
}
.block-color-teal {
	color: rgba(68, 131, 97, 1);
	fill: rgba(68, 131, 97, 1);
}
.block-color-blue {
	color: rgba(51, 126, 169, 1);
	fill: rgba(51, 126, 169, 1);
}
.block-color-purple {
	color: rgba(144, 101, 176, 1);
	fill: rgba(144, 101, 176, 1);
}
.block-color-pink {
	color: rgba(193, 76, 138, 1);
	fill: rgba(193, 76, 138, 1);
}
.block-color-red {
	color: rgba(212, 76, 71, 1);
	fill: rgba(212, 76, 71, 1);
}
.block-color-gray_background {
	background: rgba(241, 241, 239, 1);
}
.block-color-brown_background {
	background: rgba(244, 238, 238, 1);
}
.block-color-orange_background {
	background: rgba(251, 236, 221, 1);
}
.block-color-yellow_background {
	background: rgba(251, 243, 219, 1);
}
.block-color-teal_background {
	background: rgba(237, 243, 236, 1);
}
.block-color-blue_background {
	background: rgba(231, 243, 248, 1);
}
.block-color-purple_background {
	background: rgba(244, 240, 247, 0.8);
}
.block-color-pink_background {
	background: rgba(249, 238, 243, 0.8);
}
.block-color-red_background {
	background: rgba(253, 235, 236, 1);
}
.select-value-color-pink { background-color: rgba(245, 224, 233, 1); }
.select-value-color-purple { background-color: rgba(232, 222, 238, 1); }
.select-value-color-green { background-color: rgba(219, 237, 219, 1); }
.select-value-color-gray { background-color: rgba(227, 226, 224, 1); }
.select-value-color-opaquegray { background-color: rgba(255, 255, 255, 0.0375); }
.select-value-color-orange { background-color: rgba(250, 222, 201, 1); }
.select-value-color-brown { background-color: rgba(238, 224, 218, 1); }
.select-value-color-red { background-color: rgba(255, 226, 221, 1); }
.select-value-color-yellow { background-color: rgba(253, 236, 200, 1); }
.select-value-color-blue { background-color: rgba(211, 229, 239, 1); }

.checkbox {
	display: inline-flex;
	vertical-align: text-bottom;
	width: 16;
	height: 16;
	background-size: 16px;
	margin-left: 2px;
	margin-right: 5px;
}

.checkbox-on {
	background-image: url("data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2216%22%20height%3D%2216%22%20viewBox%3D%220%200%2016%2016%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%3Crect%20width%3D%2216%22%20height%3D%2216%22%20fill%3D%22%2358A9D7%22%2F%3E%0A%3Cpath%20d%3D%22M6.71429%2012.2852L14%204.9995L12.7143%203.71436L6.71429%209.71378L3.28571%206.2831L2%207.57092L6.71429%2012.2852Z%22%20fill%3D%22white%22%2F%3E%0A%3C%2Fsvg%3E");
}

.checkbox-off {
	background-image: url("data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2216%22%20height%3D%2216%22%20viewBox%3D%220%200%2016%2016%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%3Crect%20x%3D%220.75%22%20y%3D%220.75%22%20width%3D%2214.5%22%20height%3D%2214.5%22%20fill%3D%22white%22%20stroke%3D%22%2336352F%22%20stroke-width%3D%221.5%22%2F%3E%0A%3C%2Fsvg%3E");
}
	
</style></head><body><article id="8a0c985a-5a91-4a26-9ffa-8a06f5d751c9" class="page sans"><header><h1 class="page-title">åŠ¨æ€ç²’å­ç¾¤ä¼˜åŒ–</h1></header><div class="page-body"><h2 id="e849ab26-60b4-4267-ba50-d7d94837f5c6" class="">Dynamic particle swarm optimization of biomolecular simulation parameters with flexible objective functions</h2><blockquote id="3fa96e49-7641-41f1-8a32-b014895fbd15" class="">åŸºäºåŠ¨æ€ç²’å­ç¾¤ä¼˜åŒ–çš„å…·æœ‰çµæ´»ç›®æ ‡å‡½æ•°çš„ç”Ÿç‰©åˆ†å­çš„å‚æ•°æ¨¡æ‹Ÿ</blockquote><p id="6bc3a33e-2b9f-4a07-ba84-b3739d1e58c3" class="">
</p><ul id="366ff7f1-9db6-481c-950d-d33cf273f842" class="bulleted-list"><li style="list-style-type:disc">M. Weiel, M. Gtz, K. AndrÃ©, D. Coquelin, and A. Schug, â€œDynamic particle swarm optimization of biomolecular simulation parameters with flexible objective functions,â€ Nature Machine Intelligence.</li></ul><p id="a193c2b6-848d-4168-8e65-1f7c779a7201" class="">
</p><figure class="block-color-gray_background callout" style="white-space:pre-wrap;display:flex" id="1d1214b2-545e-4cf5-a7b0-4f26a0f08a63"><div style="font-size:1.5em"><span class="icon">ğŸ’¡</span></div><div style="width:100%">Molecular simulations are a powerful tool to complement and interpret ambiguous experimental data on biomolecules to obtain structural models. Such data-assisted simulations often rely on parameters, the choice of which is highly non-trivial and crucial to performance. The key challenge is weighting experimental information with respect to the underlying physical model. We introduce FLAPS, a self-adapting variant of dynamic particle swarm optimization, to overcome this parameter selection problem. FLAPS is suited for the optimization of composite objective functions that depend on both the optimization parameters and additional, a priori unknown weighting parameters, which substantially influence the search-space topology. These weighting parameters are learned at runtime, yielding a dynamically evolving and iteratively refined search-space topology. As a practical example, we show how FLAPS can be used to find functional parameters for small-angle X-ray scattering-guided protein simulations. <blockquote id="9aff5bee-8b6a-47d9-a245-87ffb937768b" class=""><a href="https://www.notion.so/5c90a76869334658b0f5078d15abb89b">åˆ†å­æ¨¡æ‹Ÿ</a>æ˜¯è¡¥å……å’Œè§£é‡Šç”Ÿç‰©åˆ†å­æ¨¡æ£±ä¸¤å¯çš„å®éªŒæ•°æ®ä»¥è·å¾—ç»“æ„æ¨¡å‹çš„æœ‰åŠ›å·¥å…·ã€‚è¿™ç§æ•°æ®è¾…åŠ©æ¨¡æ‹Ÿé€šå¸¸ä¾èµ–äºå‚æ•°ï¼Œå‚æ•°çš„é€‰æ‹©å¯¹æ€§èƒ½è‡³å…³é‡è¦ã€‚æœ€å¤§çš„æŒ‘æˆ˜æ˜¯æ ¹æ®åŸºç¡€ç‰©ç†æ¨¡å‹å¯¹å®éªŒä¿¡æ¯è¿›è¡ŒåŠ æƒã€‚æˆ‘ä»¬å¼•å…¥äº†åŠ¨æ€ç²’å­ç¾¤ä¼˜åŒ–çš„è‡ªé€‚åº”å˜ä½“(flexible self-adapting particle swarm optimization,FLAPS)æ¥å…‹æœè¿™ä¸€å‚æ•°é€‰æ‹©çš„é—®é¢˜ã€‚FLAPS é€‚ç”¨äºå¤åˆç›®æ ‡å‡½æ•°çš„ä¼˜åŒ–ï¼Œè¯¥ç›®æ ‡å‡½æ•°ä¾èµ–äºä¼˜åŒ–å‚æ•°å’Œé™„åŠ çš„å…ˆéªŒæœªçŸ¥åŠ æƒå‚æ•°ï¼Œè¿™äº›å‚æ•°å¯¹æœç´¢ç©ºé—´æ‹“æ‰‘ç»“æ„æœ‰å¾ˆå¤§å½±å“ã€‚è¿™äº›åŠ æƒå‚æ•°æ˜¯åœ¨è¿è¡Œæ—¶å­¦ä¹ çš„ï¼Œä»è€Œäº§ç”ŸåŠ¨æ€æ¼”å˜å’Œè¿­ä»£è¿›åŒ–çš„æœç´¢ç©ºé—´æ‹“æ‰‘ã€‚ä½œä¸ºä¸€ä¸ªå®é™…ç¤ºä¾‹ï¼Œæˆ‘ä»¬å±•ç¤ºäº†å¦‚ä½•ä½¿ç”¨ FLAPS æ¥æŸ¥æ‰¾å°è§’åº¦ X å°„çº¿æ•£å°„å¼•å¯¼çš„è›‹ç™½è´¨æ¨¡æ‹Ÿçš„åŠŸèƒ½å‚æ•°ã€‚</blockquote></div></figure><p id="b87021fe-dc9a-4f92-a1c0-8bce9e77516b" class="">Proteins are the molecular workhouses of biological cells, with a myriad of tasks including oxygen transport, cellular communication and energy balance. As a proteinâ€™s function is linked to its structure and dynamics, its understanding requires resolving the proteinâ€™s three-dimensional shape. Misfolded proteins are associated with several neurodegenerative diseases[1], and deciphering the structureâ€“function paradigm paves the way to developing treatments for Alzheimerâ€™s and Parkinsonâ€™s diseases[2], amyloidosis[3], type-2 diabetes[4], Creutzfeldtâ€“Jakob disease[5], among others.</p><blockquote id="881dfb4a-9290-462f-9556-e9efeace3344" class="">è›‹ç™½è´¨æ˜¯ç”Ÿç‰©ç»†èƒçš„åˆ†å­ä¸»åŠ›å†›ï¼Œæ‰¿æ‹…ç€æ°§æ°”è¿è¾“ã€ç»†èƒé€šè®¯å’Œèƒ½é‡å¹³è¡¡ç­‰ä¼—å¤šå·¥ä½œã€‚ç”±äºè›‹ç™½è´¨çš„åŠŸèƒ½ä¸å…¶ç»“æ„å’ŒåŠ¨åŠ›å­¦æœ‰å…³ï¼Œå› æ­¤ç†è§£è›‹ç™½è´¨éœ€è¦è§£æè›‹ç™½è´¨çš„ä¸‰ç»´ç»“æ„ã€‚é”™è¯¯æŠ˜å çš„è›‹ç™½è´¨ä¸å‡ ç§ç¥ç»é€€è¡Œæ€§ç–¾ç—…æœ‰å…³ï¼Œè€Œç ´è¯‘è¿™ç§ç»“æ„-åŠŸèƒ½èŒƒå¼ä¸ºå¼€å‘æ²»ç–—é˜¿å°”èŒ¨æµ·é»˜ç—…[1]å’Œå¸•é‡‘æ£®ç—…[2]ã€æ·€ç²‰æ ·å˜æ€§ç—…[3]ã€2å‹ç³–å°¿ç—…[4]ã€å…‹é›…æ°ç—…[5]ç­‰ç–¾ç—…çš„æ²»ç–—æ–¹æ³•é“ºå¹³äº†å‰è¿›çš„é“è·¯ã€‚</blockquote><p id="64bdd7c1-2dca-4e99-867c-9db4dd34cdf4" class="">Proteins are nanoscale and can only be observed indirectly. Hence, experimental data are often ambiguous, incomplete or of such low resolution that they require interpretation to access their
information content. A practical example is small-angle X-ray scattering (SAXS), where dissolved biomolecules are irradiated by X-rays and the scattering intensity is recorded. However, the desired molecular electron density is the Fourier transform of the experimentally inaccessible complex valued scattering amplitude. Recovering structural models from an intensity, that is, the absolute amplitude squared, is an ill-posed inverse problem. The sparse information in the SAXS data is insufficient to determine all degrees of freedom in a molecular structure. Protein structure determination thus depends on combining experimental results and computational methods, and recent studies highlight the potential of such hybrid approaches[6â€“9].</p><blockquote id="29a55a5d-5d06-47c4-b99c-95c8196a5332" class="">è›‹ç™½è´¨æ˜¯çº³ç±³çº§çš„ï¼Œåªèƒ½é—´æ¥è§‚å¯Ÿã€‚å› æ­¤ï¼Œå®éªŒæ•°æ®é€šå¸¸æ˜¯æ¨¡æ£±ä¸¤å¯çš„ã€ä¸å®Œæ•´çš„æˆ–åˆ†è¾¨ç‡å¾ˆä½ï¼Œä»¥è‡³äºéœ€è¦å¯¹å…¶è¿›è¡Œè§£é‡Šæ‰èƒ½è®¿é—®å…¶å†…å®¹ä¿¡æ¯ã€‚ä¸€ä¸ªå®é™…çš„ä¾‹å­æ˜¯å°è§’åº¦ X å°„çº¿æ•£å°„ (SAXS)ï¼Œå…¶ä¸­æº¶è§£çš„ç”Ÿç‰©åˆ†å­è¢« X å°„çº¿ç…§å°„å¹¶è®°å½•æ•£å°„å¼ºåº¦ã€‚ç„¶è€Œï¼Œæ‰€éœ€çš„åˆ†å­ç”µå­å¯†åº¦æ˜¯å®éªŒä¸Šæ— æ³•è·å¾—çš„å¤å€¼æ•£å°„å¹…åº¦çš„å‚…é‡Œå¶å˜æ¢ã€‚ä»å¼ºåº¦å³ç»å¯¹å¹…åº¦å¹³æ–¹ä¸­æ¢å¤ç»“æ„æ¨¡å‹æ˜¯ä¸€ä¸ªä¸é€‚å®šçš„é€†é—®é¢˜ã€‚ SAXS æ•°æ®ä¸­çš„ç¨€ç–ä¿¡æ¯ä¸è¶³ä»¥ç¡®å®šåˆ†å­ç»“æ„ä¸­çš„æ‰€æœ‰è‡ªç”±åº¦ã€‚å› æ­¤ï¼Œè›‹ç™½è´¨ç»“æ„çš„ç¡®å®šå–å†³äºå®éªŒç»“æœå’Œè®¡ç®—æ–¹æ³•çš„ç»“åˆï¼Œæœ€è¿‘çš„ç ”ç©¶å¼ºè°ƒäº†è¿™ç§æ··åˆæ–¹æ³•çš„æ½œåŠ›[6-9]ã€‚</blockquote><p id="c76462b7-6fbc-438d-8b2d-3a0b9d226df7" class="">One effective approach is to complement the experimental data with molecular dynamics (MD)[10,11]. MD simulations provide a physics-based description of molecular motion and give in-depth
insight into biomolecular function. Atomistic trajectories are derived by integrating Newtonâ€™s equations of motion for a system of interacting particles, and forces are calculated from empirical interatomic potentials (force fields). In data-assisted MD, an energetic restraint on the target data is added to the force field to favour conformations consistent with the data. This bias potential is proportional to the least-squares deviation of theoretical data from simulated structures and the experimental data[7,10,12â€“14]. Derived forces are assumed to drive the protein towards conformations reproducing the target data. As a molecular system seeks to minimize its free energy, the bias effectively determines a cost for disregarding the data in the simulation. The better the simulated structures align to the data, the smaller is the energetic penalty in the form of the bias[15].</p><blockquote id="8c213ee4-545c-4a6b-9814-e8edc8ce415d" class="">ä¸€ç§æœ‰æ•ˆçš„æ–¹æ³•æ˜¯ç”¨åˆ†å­åŠ¨åŠ›å­¦ (molecular dynamics,MD)[10,11] è¡¥å……å®éªŒæ•°æ®ã€‚MD æ¨¡æ‹Ÿæä¾›äº†åŸºäºç‰©ç†çš„åˆ†å­è¿åŠ¨æè¿°ï¼Œå¹¶æ·±å…¥äº†è§£ç”Ÿç‰©åˆ†å­åŠŸèƒ½ã€‚åŸå­è½¨è¿¹æ˜¯é€šè¿‡å¯¹ç›¸äº’ä½œç”¨ç²’å­ç³»ç»Ÿçš„ç‰›é¡¿è¿åŠ¨æ–¹ç¨‹ç§¯åˆ†å¾—å‡ºçš„ï¼ŒåŠ›æ˜¯æ ¹æ®ç»éªŒåŸå­é—´åŠ¿ï¼ˆåŠ›åœºï¼‰è®¡ç®—çš„ã€‚åœ¨æ•°æ®è¾…åŠ©MDä¸­ï¼Œå¯¹ç›®æ ‡æ•°æ®çš„èƒ½é‡çº¦æŸè¢«æ·»åŠ åˆ°åŠ›åœºä¸­ï¼Œä»¥æ”¯æŒä¸æ•°æ®ä¸€è‡´çš„æ„è±¡ã€‚è¿™ç§åç½®ç”µä½ä¸æ¨¡æ‹Ÿç»“æ„çš„ç†è®ºæ•°æ®å’Œå®éªŒæ•°æ®çš„æœ€å°äºŒä¹˜åå·®æˆæ­£æ¯”[7,10,12-14]ã€‚å‡å®šæ¨å¯¼\æ´¾ç”Ÿçš„åŠ›å°†è›‹ç™½è´¨æ¨å‘å¤åˆ¶ç›®æ ‡æ•°æ®çš„æ„è±¡ã€‚ç”±äºåˆ†å­ç³»ç»Ÿè¯•å›¾æœ€å°åŒ–å…¶è‡ªç”±èƒ½ï¼Œå› æ­¤åå·®æœ‰æ•ˆåœ°ç¡®å®šäº†åœ¨æ¨¡æ‹Ÿä¸­å¿½ç•¥æ•°æ®çš„æˆæœ¬ã€‚æ¨¡æ‹Ÿç»“æ„ä¸æ•°æ®çš„ä¸€è‡´æ€§è¶Šå¥½ï¼Œåå·®å½¢å¼çš„èƒ½é‡æŸå¤±å°±è¶Šå°ã€‚</blockquote><p id="2fefdb89-ff8d-4a9c-b08a-713158133525" class="">An inherent issue with data-assisted simulations is their reliance on MD parameters[7â€“9]. Selecting adequate values is non-trivial and crucial for simulation performance, and determining the bias
potentialâ€™s weight is the key challenge. This selection determines how experimental and theoretical information is balanced. The bias weight is an empirical MD parameter expressing the confidence in the experimental data versus the physics-based force field[8]. In Bayesian methods, the right weighting is derived from a statistical treatment[13,14]. However, such sophisticated approaches are practically inapplicable for users with a primarily experimental background. It is still common practice to manually determine an â€˜optimalâ€™ bias weight via grid search, that is, an exhaustive search through a fixed subset of the parameter space[7,9,12]. Adopting concepts from computational intelligence, we introduce FLAPS (â€˜flexible self-adapting particle swarmâ€™ optimization), a self-learning metaheuristic based on particle swarms, to resolve this parameter selection problem. Our contributions include the following:</p><blockquote id="1ca3e4df-cefa-4191-993d-f237dce54ae5" class="">æ•°æ®è¾…åŠ©æ¨¡æ‹Ÿçš„ä¸€ä¸ªå›ºæœ‰é—®é¢˜æ˜¯å®ƒä»¬å¯¹ MD å‚æ•°çš„ä¾èµ– [7-9]ã€‚é€‰æ‹©åˆé€‚çš„å€¼å¯¹äºä»¿çœŸæ€§èƒ½è€Œè¨€è‡³å…³é‡è¦ï¼Œè€Œç¡®å®šåç½®ç”µä½çš„æƒé‡æ˜¯å…³é”®çš„æŒ‘æˆ˜ã€‚è¿™ç§é€‰æ‹©å†³å®šäº†å¦‚ä½•å¹³è¡¡å®éªŒå’Œç†è®ºä¿¡æ¯ã€‚åå·®æƒé‡æ˜¯ä¸€ä¸ªç»éªŒ MD å‚æ•°ï¼Œè¡¨ç¤ºå®éªŒæ•°æ®ä¸åŸºäºç‰©ç†çš„åŠ›åœºçš„ç½®ä¿¡åº¦[8]ã€‚åœ¨è´å¶æ–¯æ–¹æ³•ä¸­ï¼Œæ­£ç¡®çš„æƒé‡æ¥è‡ªç»Ÿè®¡å¤„ç† [13,14]ã€‚ç„¶è€Œï¼Œè¿™ç§å¤æ‚çš„æ–¹æ³•å®é™…ä¸Šä¸é€‚ç”¨äºå…·æœ‰ä¸»è¦å®éªŒèƒŒæ™¯çš„ç”¨æˆ·ã€‚é€šè¿‡ç½‘æ ¼æœç´¢å³é€šè¿‡å‚æ•°ç©ºé—´çš„å›ºå®šå­é›†è¿›è¡Œç©·ä¸¾æœç´¢[7,9,12]æ¥æ‰‹åŠ¨ç¡®å®šâ€œæœ€ä½³â€åå·®æƒé‡ä»ç„¶æ˜¯å¸¸è§çš„åšæ³•ã€‚é‡‡ç”¨è®¡ç®—æ™ºèƒ½çš„æ¦‚å¿µï¼Œå¼•å…¥äº†åŸºäºç²’å­ç¾¤çš„è‡ªå­¦ä¹ å…ƒå¯å‘å¼ç®—æ³• FLAPSï¼ˆâ€œçµæ´»çš„è‡ªé€‚åº”ç²’å­ç¾¤â€ä¼˜åŒ–ï¼‰ï¼Œæ¥è§£å†³å‚æ•°é€‰æ‹©é—®é¢˜ã€‚æˆ‘ä»¬çš„è´¡çŒ®åŒ…æ‹¬ï¼š</blockquote><ul id="5acc6037-2d2f-43f0-9463-71b09950cd8a" class="bulleted-list"><li style="list-style-type:disc">A new type of flexible objective function (OF) to assess a data-assisted simulationâ€™s plausibility in terms of simulated structures and thus the suitability of the MD parameters used.</li></ul><ul id="fd29835c-97e5-4edb-a301-a0794d101ec1" class="bulleted-list"><li style="list-style-type:disc">A self-adapting particle swarm optimizer for dynamically evolving environments resulting from multiple quality features of different scales in the flexible OF.</li></ul><ul id="55026b50-a119-47cf-8446-af4d0c8422e3" class="bulleted-list"><li style="list-style-type:disc"> Fully integrated and automated MD parameter selection for data-assisted biomolecular simulations.</li></ul><blockquote id="6f746472-78ca-49b8-b738-c36ee68ca80a" class="">ä¸€ç§æ–°çš„çµæ´»ç›®æ ‡å‡½æ•°(objective function,OF)ç”¨äºè¯„ä¼°æ•°æ®è¾…åŠ©æ¨¡æ‹Ÿåœ¨æ¨¡æ‹Ÿç»“æ„æ–¹é¢çš„åˆç†æ€§ï¼Œä»è€Œè¯„ä¼°æ‰€ä½¿ç”¨çš„MDå‚æ•°çš„é€‚ç”¨æ€§ã€‚</blockquote><blockquote id="9983651f-b3ac-4a3b-98d1-6951efe862c0" class="">ä¸€ç§è‡ªé€‚åº”ç²’å­ç¾¤ä¼˜åŒ–ç®—æ³•ï¼Œç”¨äºåŠ¨æ€æ¼”åŒ–ç¯å¢ƒï¼Œè¯¥ç¯å¢ƒç”±çµæ´» OF ä¸­ä¸åŒå°ºåº¦çš„å¤šä¸ªè´¨é‡ç‰¹å¾äº§ç”Ÿã€‚</blockquote><blockquote id="ab50640c-0fba-4e88-b29e-8ea724772625" class="">ç”¨äºæ•°æ®è¾…åŠ©ç”Ÿç‰©åˆ†å­æ¨¡æ‹Ÿçš„å®Œå…¨é›†æˆå’Œè‡ªåŠ¨åŒ–çš„MDå‚æ•°é€‰æ‹©ã€‚</blockquote><p id="e0f8495a-3afd-4f27-91b8-9935c1398060" class="">As a proof of concept, we apply FLAPS to the selection of relevant MD parameters in SAXS-guided structure-based protein simulations[7].</p><blockquote id="51092a17-742e-4453-9e00-a1ffcd97939e" class="">ä½œä¸ºæ¦‚å¿µè¯æ˜ï¼Œæˆ‘ä»¬å°† FLAPS åº”ç”¨äº SAXS å¼•å¯¼çš„åŸºäºç»“æ„çš„è›‹ç™½è´¨æ¨¡æ‹Ÿä¸­ç›¸å…³ MD å‚æ•°çš„é€‰æ‹©[7]ã€‚</blockquote><h2 id="a74e9b47-6aa3-4547-a75b-33e7e642e28c" class="">Particle swarm optimization</h2><p id="f7e702d5-6650-492b-be35-a4ec62f43554" class="">Particle swarm optimization (PSO) is a bio-inspired computational-intelligence technique to handle computationally hard problems based on the emergent behaviour of swarms[16]. Swarm intelligence is the collective behaviour of decentralized, self-organized systems. Such systems comprise a population of cooperating particles. Although there is no supervising control, this leads to â€˜intelligentâ€™ global behaviour that is hidden to the individual particles.</p><blockquote id="907326f6-3395-4f11-9767-6eeb1a0dcceb" class="">ç²’å­ç¾¤ä¼˜åŒ–ï¼ˆPSOï¼‰æ˜¯ä¸€ç§å—åˆ°ç”Ÿç‰©å¯å‘çš„è®¡ç®—æ™ºèƒ½æŠ€æœ¯ã€‚ç”¨äºæ ¹æ®ç¾¤ä½“çš„æ¶Œç°è¡Œä¸ºæ¥å¤„ç†è®¡ç®—å›°éš¾é—®é¢˜ã€‚ç¾¤ä½“æ™ºèƒ½æ˜¯åˆ†æ•£çš„ã€è‡ªç»„ç»‡çš„ç³»ç»Ÿçš„é›†ä½“è¡Œä¸ºã€‚ è¿™æ ·çš„ç³»ç»Ÿç”±ä¸€ç¾¤åä½œçš„ç²’å­ç»„æˆã€‚ è™½ç„¶æ²¡æœ‰ç›‘ç£æ§åˆ¶ï¼Œä½†è¿™å¯¼è‡´äº†å¯¹å•ä¸ªç²’å­éšè—çš„â€œæ™ºèƒ½â€å…¨å±€è¡Œä¸ºã€‚</blockquote><p id="307a8613-ab8c-4675-9f21-fc177a93615e" class="">PSO iteratively improves a candidate solution with respect to a quality-gauging OF. The optimization problem is approached by considering a swarm of particles, each corresponding to a particular position in parameter space. Across multiple search rounds (generations), the particles â€˜jumpâ€™ around in the search space according to their positions and velocities. Particles â€˜rememberâ€™ their personal best location and the global best location in their social network, which act as attractors in the search space. To propagate a particle to the next generation, a velocity is added to its current position. The velocity has a cognitive component towards the particleâ€™s current personal best and a social component towards the current best in its communication network. Each component is weighted by a random number in the range [<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>0</mn></mrow><annotation encoding="application/x-tex">0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">0</span></span></span></span></span><span>ï»¿</span></span>, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï•</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\phi_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï•</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>], where the acceleration coefficients <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï•</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\phi_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï•</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> balance exploration versus exploitation in the optimization. Eventual convergence is achieved through progressive particle flight contraction by mechanisms such as velocity clamping[17], inertia weight[18] or constriction[19]. Altogether, this is assumed to move the swarm towards the best parameter combination.</p><blockquote id="5de3b26f-03db-4ac1-945f-2c9d03bfdb5c" class="">PSO è¿­ä»£åœ°æ”¹è¿›å€™é€‰è§£çš„è´¨é‡æµ‹é‡ OFã€‚ä¼˜åŒ–é—®é¢˜æ˜¯é€šè¿‡è€ƒè™‘ä¸€ç¾¤ç²’å­æ¥è§£å†³çš„ï¼Œæ¯ä¸ªç²’å­å¯¹åº”äºå‚æ•°ç©ºé—´ä¸­çš„ä¸€ä¸ªç‰¹å®šä½ç½®ã€‚åœ¨å¤šæ¬¡è¿­ä»£æœç´¢ä¸­ï¼Œç²’å­æ ¹æ®å®ƒä»¬çš„ä½ç½®å’Œé€Ÿåº¦åœ¨æœç´¢ç©ºé—´ä¸­â€œè·³è·ƒâ€ã€‚ç²’å­â€œè®°å¿†â€å®ƒä»¬åœ¨æœç´¢ç©ºé—´ä¸­çš„ä¸ªä½“æœ€ä½³ä½ç½®å’Œå…¨å±€æœ€ä½³ä½ç½®ï¼Œå®ƒä»¬å……å½“æœç´¢ç©ºé—´ä¸­çš„å¸å¼•é¡¹ã€‚ä¸ºäº†å°†ç²’å­ä¼ æ’­åˆ°ä¸‹ä¸€ä»£ï¼Œéœ€è¦å°†é€Ÿåº¦æ·»åŠ åˆ°å…¶å½“å‰ä½ç½®ã€‚é€Ÿåº¦å…·æœ‰æŒ‡å‘ç²’å­å½“å‰ä¸ªä½“æœ€ä½³ä½ç½®çš„è®¤çŸ¥æˆåˆ†å’ŒæŒ‡å‘å…¶æœç´¢ç©ºé—´ä¸­å½“å‰å…¨å±€æœ€ä½³ä½ç½®çš„ç¤¾ä¼šæˆåˆ†ã€‚æ¯ä¸ªåˆ†é‡ç”± [<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>0</mn></mrow><annotation encoding="application/x-tex">0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">0</span></span></span></span></span><span>ï»¿</span></span>, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï•</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\phi_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï•</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>] èŒƒå›´å†…çš„éšæœºæ•°åŠ æƒï¼Œå…¶ä¸­åŠ é€Ÿåº¦ç³»æ•°<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï•</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\phi_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï•</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>åœ¨ä¼˜åŒ–ä¸­å¹³è¡¡æ¢ç´¢ä¸å¼€å‘ã€‚æœ€ç»ˆæ”¶æ•›æ˜¯é€šè¿‡é€Ÿåº¦é™åˆ¶ [17]ã€æƒ¯æ€§é‡é‡[18] æˆ–æ”¶ç¼©[19]ç­‰æœºåˆ¶çš„æ¸è¿›å¼ç²’å­é£è¡Œæ”¶ç¼©æ¥å®ç°çš„ã€‚æ€»è€Œè¨€ä¹‹ï¼Œè¿™è¢«å‡å®šä¸ºå°†ç¾¤ä½“ç§»å‘æœ€ä½³å‚æ•°ç»„åˆã€‚</blockquote><p id="2def64fa-f3af-423c-91e1-6079dfdca419" class="">PSO has been demonstrated to work well for various static problems[20]. However, realistic problems are often dynamic, and the global optimum can shift with time. Application to such problems has been explored extensively, revealing that the algorithm needs to be enhanced by concepts such as repulsion, dynamic network topologies or multi-swarms[21,22].</p><blockquote id="9776267d-096f-422d-bd59-4f232d6c66a0" class="">PSO å·²è¢«è¯æ˜å¯ä»¥å¾ˆå¥½åœ°è§£å†³å„ç§é™æ€é—®é¢˜[20]ã€‚ç„¶è€Œï¼Œç°å®é—®é¢˜é€šå¸¸æ˜¯åŠ¨æ€çš„ï¼Œå…¨å±€æœ€ä¼˜è§£ä¼šéšæ—¶é—´å˜åŒ–ã€‚å› æ­¤å¯¹æ­¤ç±»é—®é¢˜çš„åº”ç”¨è¿›è¡Œäº†å¹¿æ³›æ¢ç´¢ï¼Œè¡¨æ˜è¯¥ç®—æ³•éœ€è¦é€šè¿‡æ’æ–¥ã€åŠ¨æ€ç½‘ç»œæ‹“æ‰‘æˆ–å¤šç¾¤ç»„ç­‰æ¦‚å¿µæ¥å¢å¼º[21,22]ã€‚</blockquote><p id="654a4b7e-4a60-4c14-b418-5ca3f97dec82" class="">Maintaining a population of diverse solutions enables enormous exploration and paves the way to large-scale parallelization. Such algorithms can be scaled easily to exploit the full potential of modern supercomputers23 . PSO has several hyperparameters affecting its behaviour and efficiency, and selecting them has been researched extensively[24,25]. Strategies include using meta-optimizers[26,27] or refining them during the optimization[28]. Reference[29]provides an overview of practical PSO applications, while refs.[22] and [30] give comprehensive reviews of PSO with a focus on dynamic environments and hybridization perspectives, respectively.</p><blockquote id="96bcb595-256b-486b-8136-0f0be604136e" class="">ç»´æŒå¤§é‡ä¸åŒçš„è§£å¯ä»¥è¿›è¡Œå¤§é‡æ¢ç´¢ï¼Œå¹¶ä¸ºå¤§è§„æ¨¡å¹¶è¡ŒåŒ–é“ºå¹³é“è·¯ã€‚æ­¤ç±»ç®—æ³•å¯ä»¥è½»æ¾æ‰©å±•ï¼Œä»¥å……åˆ†åˆ©ç”¨ç°ä»£è¶…çº§è®¡ç®—æœºçš„æ½œåŠ›[23]ã€‚PSO æœ‰å‡ ä¸ªå½±å“å…¶è¡Œä¸ºå’Œæ•ˆç‡çš„è¶…å‚æ•°ï¼Œå¹¶ä¸”å¯¹å®ƒä»¬çš„é€‰æ‹©å·²è¢«å¹¿æ³›çš„ç ”ç©¶[24,25]ã€‚ç­–ç•¥åŒ…æ‹¬åœ¨ä¼˜åŒ–è¿‡ç¨‹ä¸­ä½¿ç”¨å…ƒä¼˜åŒ–å™¨[26,27] å¯¹å…¶è¿›è¡Œæ”¹è¿›[28]ã€‚æ–‡çŒ®[29] æ¦‚è¿°äº†PSO çš„å®é™…åº”ç”¨ã€‚æ–‡çŒ®[22]å’Œæ–‡çŒ®[30]åˆ†åˆ«å¯¹PSOçš„åŠ¨æ€ç¯å¢ƒå’Œæ··åˆçš„è§’åº¦è¿›è¡Œäº†å…¨é¢çš„æè¿°ã€‚</blockquote><h2 id="c4bab996-262e-4a5f-aaa4-66e615bd7102" class="">Multi-response problems</h2><p id="ab41ecd0-aed5-4955-9bdd-ac56093e0e44" class="">Real-life optimization problems are often determined by multiple incomparable or conflicting quality features (responses). To obtain compatible solutions, these responses must be taken into account simultaneously. This is often accomplished by combining individual contributions within one composite OF. PSO has been applied to multi-response optimization in many fields[31â€“34]. Commonly, the set of responses is reduced via multiplication by manually chosen weights, henceforth referred to as OF parameters. Choosing OF parameters is non-trivial, yet strongly impacts global optimization performance by skewing the OF. FLAPS builds on a flexible OF that
automatically and interdependently balances different responses. OF parameters are learned at runtime through iterative refinement, yielding a dynamically evolving OF landscape. In this way, FLAPS can cope with various responses of different scales.</p><blockquote id="2722826e-ed6b-4735-a5df-113962509076" class="">ç°å®ç”Ÿæ´»ä¸­çš„ä¼˜åŒ–é—®é¢˜é€šå¸¸ç”±å¤šä¸ªæ— æ³•æ¯”è¾ƒæˆ–ç›¸äº’å†²çªçš„è´¨é‡ç‰¹å¾ï¼ˆå“åº”ï¼‰å†³å®šã€‚ä¸ºäº†è·å¾—å…¼å®¹çš„è§£ï¼Œå¿…é¡»åŒæ—¶è€ƒè™‘è¿™äº›å“åº”ã€‚è¿™é€šå¸¸æ˜¯é€šè¿‡åœ¨ä¸€ä¸ªå¤åˆ OF ä¸­ç»„åˆä¸ªä½“è´¡çŒ®æ¥å®ç°çš„ã€‚PSO å·²è¢«åº”ç”¨äºè®¸å¤šé¢†åŸŸçš„å¤šå“åº”ä¼˜åŒ–[31-34]ã€‚é€šå¸¸ï¼Œå“åº”é›†é€šè¿‡ä¹˜ä»¥äººå·¥æ‰‹åŠ¨é€‰æ‹©çš„æƒé‡æ¥å‡å°‘ï¼Œä»¥ä¸‹ç§°ä¸º OF å‚æ•°ã€‚å‚æ•°é€‰æ‹©æ˜¯ä¸€é¡¹é‡è¦å·¥ä½œï¼Œä½†ç”±äº OF å‚æ•°çš„å€¾æ–œä¼šä¸¥é‡å½±å“å…¨å±€ä¼˜åŒ–æ€§èƒ½ã€‚FLAPS å»ºç«‹åœ¨çµæ´»çš„ OF ä¹‹ä¸Šï¼Œè¯¥ OF è‡ªåŠ¨ä¸”ç›¸äº’ä¾èµ–åœ°å¹³è¡¡ä¸åŒçš„å“åº”ã€‚OF å‚æ•°åœ¨è¿è¡Œæ—¶é€šè¿‡è¿­ä»£ç»†åŒ–æ¥å­¦ä¹ ï¼Œä»è€Œäº§ç”ŸåŠ¨æ€æ¼”å˜çš„ OF æ ¼å±€ã€‚è¿™æ ·ï¼ŒFLAPS å°±å¯ä»¥åº”å¯¹ä¸åŒå°ºåº¦çš„å„ç§å“åº”ã€‚</blockquote><h2 id="7c2212df-13da-4bdb-91e8-fb63ca4d47ad" class="">A flexible self-adapting objective function</h2><p id="7475c184-1f53-4976-87b4-c7813fc8a201" class="">Typically, the set of responses is mapped to a scalar score by calculating the scalar product with fixed weights. These OF parameters supposedly reflect relative importance, implicitly encoding arbitrary prior beliefs. Instead, we set up a â€˜maximum-entropyâ€™ OF with the fewest possible assumptions:</p><blockquote id="b6bafd1a-4ac7-4b1c-a223-52f74ab89f85" class="">é€šå¸¸ï¼Œé€šè¿‡è®¡ç®—å…·æœ‰å›ºå®šæƒé‡çš„æ ‡é‡ç§¯ï¼Œå°†å“åº”é›†æ˜ å°„åˆ°æ ‡é‡åˆ†æ•°ã€‚è¿™äº› OF å‚æ•°è¢«è®¤ä¸ºåæ˜ äº†ç›¸å¯¹é‡è¦æ€§ï¼Œéšå«åœ°ç¼–ç äº†ä»»æ„çš„å…ˆéªŒä¿¡å¿µã€‚ç›¸åï¼Œæˆ‘ä»¬ç”¨æœ€å°å‡è®¾å»ºç«‹äº†ä¸€ä¸ªâ€œæœ€å¤§ç†µâ€ï¼š</blockquote><p id="02f56864-ccd8-4a44-bc3a-c4382fd7bf64" class="">
</p><figure id="3bd3285b-2f23-45bf-b1b6-5167480f412a" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled.png"><img style="width:288px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled.png"/></a></figure><p id="7b3999f8-c0cd-485a-9074-d1712842330d" class=""><mark class="highlight-gray">Eq.(1)</mark></p><p id="368e4c5c-87f9-4fba-87dd-6258670afd7d" class="">          </p><p id="72d10ca4-0708-4ecb-ad6e-fda42242fec3" class="">                                    </p><p id="39a143a0-e2e8-49a9-91cc-1aec01bd2aa6" class="">where z is the set of OF parameters, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Î¼</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">Î¼_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal">Î¼</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> deviation of response <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>R</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">R_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.969438em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.00773em;">R</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.00773em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> is the mean and <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ïƒ</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">Ïƒ_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">Ïƒ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> is the standard for a particle at position x. All responses are considered equally important but can have different ranges and units. To make them comparable on a shared scale, we standardize each responseâ€™s set of values gathered over previous generations. This strategy imitates the concept of rolling batch normalization[35]. Each layerâ€™s inputs are recentred and rescaled with the aim to improve a neural networkâ€™s speed, performance and stability. Initially proposed to mitigate internal covariate shift, batch normalization is believed to introduce a regularizing and smoothing effect and promote robustness with respect to different initialization schemes. The OF in equation (1) depends not only on the parameters of interest, x, in our case the MD parameters, but also on a priori unknown, context-providing OF parameters, z =<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mrow><mi>Î¼</mi><mo separator="true">,</mo><mi>Ïƒ</mi></mrow><msub><mo stretchy="false">)</mo><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">({Î¼, Ïƒ})_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.036108em;vertical-align:-0.286108em;"></span><span class="mopen">(</span><span class="mord"><span class="mord mathnormal">Î¼</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">Ïƒ</span></span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>, from the standardization. Their values cannot be deduced from individual OF evaluations, yet fundamentally control OF performance and hence the optimization process.</p><p id="03c9f3bb-f0cb-41a6-9ac6-35d843e92195" class="">å…¶ä¸­ z æ˜¯ OF å‚æ•°é›†ï¼Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Î¼</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">Î¼_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal">Î¼</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> æ˜¯å¹³å‡å€¼ï¼Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ïƒ</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">Ïƒ_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">Ïƒ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> æ˜¯ä½ç½® x å¤„ç²’å­çš„å“åº” <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>R</mi><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">R_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.969438em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.00773em;">R</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.00773em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> çš„æ ‡å‡†åå·®ã€‚æ‰€æœ‰å“åº”éƒ½è¢«è®¤ä¸ºåŒæ ·é‡è¦ï¼Œä½†å¯ä»¥æœ‰ä¸åŒçš„èŒƒå›´å’Œå•ä½ã€‚ä¸ºäº†ä½¿å®ƒä»¬åœ¨ä¸€ä¸ªå…±äº«å°ºåº¦ä¸Šå…·æœ‰å¯æ¯”æ€§ï¼Œæˆ‘ä»¬å°†å‰å‡ ä»£æ”¶é›†çš„æ¯ä¸ªå“åº”çš„å€¼é›†æ ‡å‡†åŒ–ã€‚è¯¥ç­–ç•¥æ¨¡ä»¿æ»šåŠ¨æ‰¹é‡æ ‡å‡†åŒ–çš„æ¦‚å¿µ[35]ã€‚ä¸ºäº†æé«˜ç¥ç»ç½‘ç»œçš„é€Ÿåº¦ã€æ€§èƒ½å’Œç¨³å®šæ€§ï¼Œå¯¹æ¯ä¸€å±‚çš„è¾“å…¥è¿›è¡Œé‡æ–°ç¼–ç å’Œé‡æ–°æ ‡å®šã€‚æ‰¹é‡å½’ä¸€åŒ–æœ€åˆæ˜¯ä¸ºäº†å‡è½»å†…éƒ¨åå˜é‡åç§»è€Œæå‡ºçš„ï¼Œå®ƒè¢«è®¤ä¸ºå¼•å…¥äº†æ­£åˆ™åŒ–å’Œå¹³æ»‘æ•ˆåº”ï¼Œå¹¶æé«˜äº†ä¸åŒåˆå§‹åŒ–æ–¹æ¡ˆçš„é²æ£’æ€§ã€‚ç­‰å¼ (1) ä¸­çš„ OF ä¸ä»…å–å†³äºå‚æ•° xï¼Œåœ¨æˆ‘ä»¬çš„ä¾‹å­ä¸­æ˜¯ MD å‚æ•°ï¼Œè¿˜å–å†³äºæ¥è‡ªæ ‡å‡†åŒ–çš„å…ˆéªŒæœªçŸ¥çš„æä¾›ä¸Šä¸‹æ–‡çš„ OF å‚æ•° z =<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mrow><mi>Î¼</mi><mo separator="true">,</mo><mi>Ïƒ</mi></mrow><msub><mo stretchy="false">)</mo><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">({Î¼, Ïƒ})_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.036108em;vertical-align:-0.286108em;"></span><span class="mopen">(</span><span class="mord"><span class="mord mathnormal">Î¼</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">Ïƒ</span></span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>ã€‚å®ƒä»¬çš„å€¼ä¸èƒ½ä»å•ç‹¬çš„ OF è¯„ä¼°ä¸­æ¨æ–­å‡ºæ¥ï¼Œä½†ä»æ ¹æœ¬ä¸Šæ§åˆ¶äº† OF çš„æ€§èƒ½ï¼Œå› æ­¤ä¹Ÿæ§åˆ¶äº†ä¼˜åŒ–è¿‡ç¨‹ã€‚</p><p id="a9b97a0e-1629-4226-b657-dafe635cfe95" class="">Our self-adapting PSO variant, FLAPS, solves this problem. Provided with a comprehensive history of all previous particles and their responses, OF parameters are learned on the fly. They are continuously refined according to the current state of the optimization, yielding a dynamically evolving and increasingly distinct OF topology. This environmental dynamism may cause convergence problems if the OF fails to approach a stable topology. As more particles are evaluated, the ranges and distributions of individual responses become better understood. Therefore, OF parameters become more accurate, improving OF performance in assessing the suitability of the actual parameters of interest, x. After each generation, the values z =<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mrow><mi>Î¼</mi><mo separator="true">,</mo><mi>Ïƒ</mi></mrow><msub><mo stretchy="false">)</mo><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">({Î¼, Ïƒ})_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.036108em;vertical-align:-0.286108em;"></span><span class="mopen">(</span><span class="mord"><span class="mord mathnormal">Î¼</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">Ïƒ</span></span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> are used to reevaluate the OF for all particles in the history. Personal best positions, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">p_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">p</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>, and the swarmâ€™s global best position, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>, are updated accordingly for propagating particles to the next generation. FLAPS uses a traditional PSO velocity formulation[16]. Strategies to prevent diverging velocities include introducing an inertia weight[18]or a constriction factor[19]. We regulate the velocities by means of a maximum value at each particle update[17]. FLAPSâ€™s pseudo code is shown in Algorithm 1. Inspired by the â€˜simplifying PSOâ€™ paradigm, it builds on a slim standard PSO core and can easily be complemented by concepts such as inertia weight[18]and swarm constriction[19] or diversity increasing mechanisms. Its time complexity is similar to that of a standard PSO with
<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>O</mi><mo stretchy="false">(</mo><mi>s</mi><mi mathvariant="normal">/</mi><mi>p</mi><mo separator="true">â‹…</mo><mi>G</mi><mo separator="true">â‹…</mo><mi>S</mi><mi>i</mi><mi>m</mi><mo>+</mo><mi>S</mi><mo separator="true">â‹…</mo><mi>O</mi><mi>p</mi><mi>t</mi><mo>=</mo><mi>O</mi><mo stretchy="false">(</mo><mi>s</mi><mi mathvariant="normal">/</mi><mi>p</mi><mo separator="true">â‹…</mo><mi>G</mi><mo separator="true">â‹…</mo><mi>S</mi><mi>i</mi><mi>m</mi><mo>+</mo><msup><mi>S</mi><mn>2</mn></msup><mo separator="true">â‹…</mo><mi>G</mi><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">O(s/pÂ·GÂ·Sim+SÂ·Opt=O(s/pÂ·GÂ·Sim+S^2Â·G)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.02778em;">O</span><span class="mopen">(</span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mord mathnormal">p</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mord mathnormal">im</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:0.8777699999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">Opt</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.02778em;">O</span><span class="mopen">(</span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mord mathnormal">p</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mord mathnormal">im</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:1.064108em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mclose">)</span></span></span></span></span><span>ï»¿</span></span> in Landau notation, where <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi></mrow><annotation encoding="application/x-tex">P</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">P</span></span></span></span></span><span>ï»¿</span></span> is the number of simulation processors, â€˜Simâ€™ the maximal simulation time, and all other variables as defined in Algorithm 1.</p><blockquote id="9609c1e2-e03c-479c-b9dd-9d2175356639" class="">æˆ‘ä»¬çš„è‡ªé€‚åº”PSOå˜ä½“FLAPSè§£å†³äº†è¿™ä¸ªé—®é¢˜ã€‚ æä¾›äº†æ‰€æœ‰å…ˆå‰ç²’å­åŠå…¶å“åº”çš„å…¨é¢å†å²è®°å½•ï¼Œå¯ä»¥åŠ¨æ€å­¦ä¹ OFå‚æ•°ã€‚å®ƒä»¬æ ¹æ®å½“å‰çš„ä¼˜åŒ–çŠ¶æ€ä¸æ–­æ”¹è¿›ï¼Œäº§ç”ŸåŠ¨æ€æ¼”åŒ–å’Œæ—¥ç›Šä¸åŒçš„æ‹“æ‰‘ç»“æ„ã€‚å¦‚æœOFæ— æ³•æ¥è¿‘ä¸€ä¸ªç¨³å®šçš„æ‹“æ‰‘ï¼Œè¿™ç§åŠ¨æ€ç¯å¢ƒå¯èƒ½å¯¼è‡´æ”¶æ•›é—®é¢˜ã€‚éšç€æ›´å¤šçš„ç²’å­è¢«è¯„ä¼°ï¼Œä¸ªä½“å“åº”çš„èŒƒå›´å’Œåˆ†å¸ƒå˜å¾—æ›´å¥½ç†è§£ã€‚å› æ­¤ï¼Œå¯¹OFå‚æ•°çš„è¯„ä¼°å˜å¾—æ›´åŠ å‡†ç¡®ï¼Œä»è€Œæé«˜äº†OFåœ¨è¯„ä¼°å®é™…å‚æ•°xçš„é€‚ç”¨æ€§æ–¹é¢çš„æ€§èƒ½ã€‚åœ¨æ¯ä¸€ä»£ä¹‹åï¼Œå€¼z=<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mrow><mi>Î¼</mi><mtext>ï¼Œ</mtext><mi>Ïƒ</mi></mrow><msub><mo stretchy="false">)</mo><mi>j</mi></msub></mrow><annotation encoding="application/x-tex">({Î¼ï¼ŒÏƒ})_j</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.036108em;vertical-align:-0.286108em;"></span><span class="mopen">(</span><span class="mord"><span class="mord mathnormal">Î¼</span><span class="mord cjk_fallback">ï¼Œ</span><span class="mord mathnormal" style="margin-right:0.03588em;">Ïƒ</span></span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>ç”¨äºé‡æ–°è®¡ç®—å†å²ä¸Šæ‰€æœ‰ç²’å­çš„OFã€‚ä¸ªä½“æœ€ä½³ä½ç½®<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">p_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">p</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>å’Œç§ç¾¤å…¨å±€æœ€ä½³ä½ç½®<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>B</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{Best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.32833099999999993em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.05017em;">B</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>ç›¸åº”åœ°æ›´æ–°ï¼Œä»¥ä¾¿å°†ç²’å­ä¼ æ’­åˆ°ä¸‹ä¸€ä»£ã€‚FLAPSä½¿ç”¨ä¼ ç»Ÿçš„PSOé€Ÿåº¦å…¬å¼[16]ã€‚é˜²æ­¢é€Ÿåº¦å‘æ•£çš„ç­–ç•¥åŒ…æ‹¬å¼•å…¥æƒ¯æ€§é‡é‡[18]å’Œæ”¶ç¼©å› å­[19]ã€‚æˆ‘ä»¬é€šè¿‡æ¯ä¸€ä¸ªç²’å­æ›´æ–°æ—¶çš„æœ€å¤§å€¼æ¥è°ƒèŠ‚é€Ÿåº¦[17]ã€‚FLAPSçš„ä¼ªä»£ç å¦‚ç®—æ³•1æ‰€ç¤ºã€‚å—â€œç®€åŒ–ç²’å­ç¾¤ç®—æ³•â€èŒƒå¼çš„å¯å‘ï¼Œå®ƒå»ºç«‹åœ¨ä¸€ä¸ªæ ‡å‡†ç²’å­ç¾¤ç®—æ³•æ ¸å¿ƒä¹‹ä¸Šï¼Œå¹¶å¯ä»¥å¾ˆå®¹æ˜“åœ°é€šè¿‡æƒ¯æ€§æƒé‡[18]å’Œç¾¤ä½“æ”¶ç¼©[19]å’Œå¤šæ ·æ€§å¢åŠ æœºåˆ¶ç­‰æ¦‚å¿µè¿›è¡Œè¡¥å……ã€‚å…¶æ—¶é—´å¤æ‚åº¦ä¸æ ‡å‡†ç²’å­ç¾¤ç®—æ³•ç›¸ä¼¼ï¼Œå…¶Landauè¡¨ç¤ºæ³•ä¸º<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>O</mi><mo stretchy="false">(</mo><mi>s</mi><mi mathvariant="normal">/</mi><mi>p</mi><mo separator="true">â‹…</mo><mi>G</mi><mo separator="true">â‹…</mo><mi>S</mi><mi>i</mi><mi>m</mi><mo>+</mo><mi>S</mi><mo separator="true">â‹…</mo><mi>O</mi><mi>p</mi><mi>t</mi><mo>=</mo><mi>O</mi><mo stretchy="false">(</mo><mi>s</mi><mi mathvariant="normal">/</mi><mi>p</mi><mo separator="true">â‹…</mo><mi>G</mi><mo separator="true">â‹…</mo><mi>S</mi><mi>i</mi><mi>m</mi><mo>+</mo><msup><mi>S</mi><mn>2</mn></msup><mo separator="true">â‹…</mo><mi>G</mi><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">O(s/pÂ·GÂ·Sim+SÂ·Opt=O(s/pÂ·GÂ·Sim+S^2Â·G)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.02778em;">O</span><span class="mopen">(</span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mord mathnormal">p</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mord mathnormal">im</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:0.8777699999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">Opt</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.02778em;">O</span><span class="mopen">(</span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mord mathnormal">p</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="mord mathnormal">im</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:1.064108em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.05764em;">S</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mpunct">â‹…</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal">G</span><span class="mclose">)</span></span></span></span></span><span>ï»¿</span></span>,å…¶ä¸­<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi></mrow><annotation encoding="application/x-tex">P</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">P</span></span></span></span></span><span>ï»¿</span></span>ä¸ºä»¿çœŸå¤„ç†å™¨æ•°ï¼Œ&#x27;Sim&#x27;ä¸ºæœ€å¤§ä»¿çœŸæ—¶é—´ï¼Œä»¥åŠç®—æ³•1ä¸­å®šä¹‰çš„æ‰€æœ‰å˜é‡ã€‚</blockquote><pre id="424a64ea-ddb5-4510-9b38-aad295fa762d" class="code"><code>ç®—æ³• 1 FLAPS ç®—æ³•
åˆ†åˆ«åœ¨æœç´¢ç©ºé—´çš„ä¸Šé™ b_{up} å’Œ ä¸‹é™b_{lo} çš„ä¹‹é—´çš„éšæœºä½ç½® x_p (p = 1, ..., S) å¤„ä½¿ç”¨ç§ç¾¤å¤§å°ä¸º S çš„åˆå§‹ç§ç¾¤ popã€‚
for g â† 1 to maximum generations G do
  for particle in pop do 
  è¯„ä¼°particleåœ¨ä½ç½® x_på¤„çš„å“åº”ï¼š
  particle.fargs = [response_j (x_p )]_j
  end 
  å°†å½“å‰ä¸€ä»£ pop é™„åŠ åˆ°å†å² histp: histp.append(pop) 
  æ ¹æ® histp ä¸­å“åº”çš„å½“å‰çŸ¥è¯†çŠ¶æ€æ›´æ–°å‚æ•° z_gï¼šz_g = updateParams(histp)
  for particle in histp do
  ä½¿ç”¨æœ€è¿‘çš„ z_g é‡æ–°è¯„ä¼°ç›®æ ‡å‡½æ•°ï¼šparticle.fitness = f(x_p ; z_g )
  end 
  for generation in histp do 
    for particle in generation do 
    ç¡®å®šä¸ªä½“æœ€ä½³p_{best} å¹¶ç›¸åº”æ›´æ–°å…¨å±€æœ€ä½³ g_{best}. 
    end
  end
  for particle in pop do
    æ›´æ–°ç²’å­é€Ÿåº¦å’Œä½ç½®ï¼šparticle.speed+= rand (0, Ï•_1) (p_{best} âˆ’particle.position )+ rand (0, Ï•_2)(g_{best} âˆ’particle.position) 
    é€šè¿‡ s_max = 0.7 G_{-1} (b_{up}-b_{lo})è°ƒèŠ‚é€Ÿåº¦:
    if particle.speed &gt; s_{max} then
       particle.speed = s_{max}
    end 
    if particle.speed &lt; -s_{max} then
       particle.speed = -s_{max}
    end
    particle.position+ =particle.speed
  end
end 
ç»“æœï¼šg_{best}</code></pre><h2 id="2988e98c-c18a-44a1-a810-9718f0a2c0d2" class="">Application to data-assisted protein simulations</h2><p id="299c65ab-1129-4e86-a471-8e4d6be58e34" class="">We applied FLAPS to the optimization of MD parameters in SAXS-guided protein simulations[7]. SAXS data are integrated into computationally efficient structure-based models, which probe
dynamics arising from a proteinâ€™s native geometry[36,37]. To assess the utility of different MD parameter sets, we need a metric for simulation quality in terms of physically reasonable structures matching the data. Designing such an OF in advance is non-trivial and has two major aspects: (1) physical plausibility of a simulated ensemble of protein structures and (2) its agreement with the target data, that is, how well the data are reproduced by simulated structures. To represent these aspects, we use the Rosetta energy function 2015 (REF15)[38,39] and the least-squares deviation Ï‡2 of simulated data from the target data[7].</p><blockquote id="9b64fee8-a2f0-4e73-88ef-4446f494dbac" class="">æˆ‘ä»¬å°† FLAPS åº”ç”¨äº SAXS å¼•å¯¼çš„è›‹ç™½è´¨æ¨¡æ‹Ÿä¸­ MD å‚æ•°çš„ä¼˜åŒ–[7]ã€‚SAXS æ•°æ®è¢«é›†æˆåˆ°è®¡ç®—æ•ˆç‡é«˜çš„åŸºäºç»“æ„çš„æ¨¡å‹ä¸­ï¼Œè¯¥æ¨¡å‹æ¢æµ‹äº†è›‹ç™½è´¨çš„å¤©ç„¶å‡ ä½•ç»“æ„æ‰€äº§ç”Ÿçš„åŠ¨æ€å˜åŒ–[36,37]ã€‚ä¸ºäº†è¯„ä¼°ä¸åŒ MD å‚æ•°é›†çš„æ•ˆæœï¼Œæˆ‘ä»¬éœ€è¦ä¸€ä¸ªæ¨¡æ‹Ÿè´¨é‡çš„è¡¡é‡æ ‡å‡†ï¼Œå³ç‰©ç†ä¸Šåˆç†çš„ç»“æ„ä¸æ•°æ®ç›¸åŒ¹é…ã€‚æå‰è®¾è®¡è¿™æ ·çš„ OF å¹¶éæ˜“äº‹ï¼Œä¸»è¦æœ‰ä¸¤ä¸ªæ–¹é¢ï¼šï¼ˆ1ï¼‰æ¨¡æ‹Ÿçš„è›‹ç™½è´¨ç»“æ„é›†åˆçš„ç‰©ç†åˆç†æ€§ä»¥åŠï¼ˆ2ï¼‰å®ƒä¸ç›®æ ‡æ•°æ®çš„ä¸€è‡´æ€§ï¼Œå³æ¨¡æ‹Ÿç»“æ„å¯¹æ•°æ®çš„å†ç°ç¨‹åº¦ã€‚ä¸ºäº†è¡¨ç¤ºè¿™äº›æ–¹é¢ï¼Œæˆ‘ä»¬ä½¿ç”¨ç½—å¡å¡”èƒ½é‡å‡½æ•°2015 (REF15)[38,39]å’Œæ¨¡æ‹Ÿæ•°æ®ä¸ç›®æ ‡æ•°æ®[7]çš„æœ€å°äºŒä¹˜åå·®<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>Ï‡</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">Ï‡^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï‡</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>ã€‚</blockquote><p id="9064275c-96ed-4ac4-8f4b-f64d05e199da" class="">Protein structure determination relies on quick and reliable scoring of many models to select those closest to the native state. Structures are rated by energetic scores associated with their conformational state. REF15 is a weighted sum of energy terms efficiently approximating the energy of a biomolecular conformation as a function of geometric degrees of freedom and chemical identities[38]. With a proteinâ€™s native fold corresponding to the state with minimal free energy, a lower-scoring structure is expected to be more native-like. Because the scores do not have a direct conversion to physical energies, REF15 and structural stability are not correlated across different proteins. Similarly, Ï‡2 values without context are inconclusive and must be compared for each protein system. Both REF15 and Ï‡2 are available from a simulation, yielding a molecular systemâ€™s atom positions over time.</p><blockquote id="34c1aefe-4b3e-4770-8f62-186698ebb210" class="">è›‹ç™½è´¨ç»“æ„çš„æµ‹å®šä¾èµ–äºå¯¹è®¸å¤šæ¨¡å‹è¿›è¡Œå¿«é€Ÿå¯é çš„è¯„åˆ†ï¼Œä»¥é€‰æ‹©æœ€æ¥è¿‘å¤©ç„¶çŠ¶æ€çš„æ¨¡å‹ã€‚ç»“æ„é€šè¿‡ä¸å…¶æ„è±¡çŠ¶æ€ç›¸å…³çš„èƒ½é‡åˆ†æ•°æ¥è¯„å®šã€‚REF15 æ˜¯èƒ½é‡é¡¹çš„åŠ æƒæ€»å’Œï¼Œæœ‰æ•ˆåœ°è¿‘ä¼¼ä½œä¸ºå‡ ä½•è‡ªç”±åº¦å’ŒåŒ–å­¦ç‰¹æ€§çš„ç”Ÿç‰©åˆ†å­æ„è±¡çš„èƒ½é‡ã€‚ç”±äºè›‹ç™½è´¨çš„å¤©ç„¶æŠ˜å å¯¹åº”äºå…·æœ‰æœ€å°è‡ªç”±èƒ½çš„çŠ¶æ€ï¼Œå› æ­¤é¢„è®¡å¾—åˆ†è¾ƒä½çš„ç»“æ„æ›´åƒå¤©ç„¶ç»“æ„ã€‚å› ä¸ºåˆ†æ•°æ²¡æœ‰ç›´æ¥è½¬æ¢ä¸ºç‰©ç†èƒ½é‡ï¼Œæ‰€ä»¥ REF15 å’Œç»“æ„ç¨³å®šæ€§åœ¨ä¸åŒè›‹ç™½è´¨ä¹‹é—´ä¸ç›¸å…³ã€‚åŒæ ·ï¼Œæ²¡æœ‰ä¸Šä¸‹æ–‡çš„ <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>Ï‡</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">Ï‡^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï‡</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> å€¼æ˜¯ä¸ç¡®å®šçš„ï¼Œå¿…é¡»é’ˆå¯¹æ¯ä¸ªè›‹ç™½è´¨ç³»ç»Ÿè¿›è¡Œæ¯”è¾ƒã€‚REF15 å’Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>Ï‡</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex"> Ï‡^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï‡</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> éƒ½å¯ä»¥ä»æ¨¡æ‹Ÿä¸­å¾—åˆ°ï¼Œå¾—å‡ºåˆ†å­ç³»ç»Ÿçš„åŸå­ä½ç½®éšæ—¶é—´çš„å˜åŒ–ã€‚</blockquote><p id="b3fe79fc-0e25-4b1f-827e-54abff3197d3" class="">For SAXS-guided structure-based simulations, two MD parameters are particularly important: bias weight kÏ‡ and temperature <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>T</mi></mrow><annotation encoding="application/x-tex">T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>ï»¿</span></span>. kÏ‡ balances information in the SAXS data with the physical model,
and T is a measure of available thermal energy and controls the systemâ€™s conformational flexibility. Thus, a particle corresponds to a simulation using a particular MD parameter set, x =(<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>Ï‡</mi><mo separator="true">,</mo><mi>T</mi></mrow><annotation encoding="application/x-tex">kÏ‡,T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mord mathnormal">Ï‡</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>ï»¿</span></span>). The
OF is set up as</p><blockquote id="7744dbe8-2f54-402a-a7d2-2bfdb1d304a0" class="">å¯¹äº SAXS å¼•å¯¼çš„åŸºäºç»“æ„çš„æ¨¡æ‹Ÿï¼Œä¸¤ä¸ª MD å‚æ•°å°¤ä¸ºé‡è¦ï¼šåç½®æƒé‡<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>Ï‡</mi></mrow><annotation encoding="application/x-tex"> kÏ‡ </annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mord mathnormal">Ï‡</span></span></span></span></span><span>ï»¿</span></span>å’Œæ¸©åº¦ <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>T</mi></mrow><annotation encoding="application/x-tex">T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>ï»¿</span></span>ã€‚<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>Ï‡</mi></mrow><annotation encoding="application/x-tex">kÏ‡</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mord mathnormal">Ï‡</span></span></span></span></span><span>ï»¿</span></span> å¹³è¡¡ SAXS æ•°æ®ä¸­çš„ä¿¡æ¯ä¸ç‰©ç†æ¨¡å‹ï¼Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>T</mi></mrow><annotation encoding="application/x-tex">T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>ï»¿</span></span> æ˜¯å¯ç”¨çƒ­èƒ½çš„åº¦é‡å¹¶æ§åˆ¶ç³»ç»Ÿçš„æ„è±¡çµæ´»æ€§ã€‚å› æ­¤ï¼Œç²’å­å¯¹åº”äºä½¿ç”¨ç‰¹å®š MD å‚æ•°é›† x =(<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mi>Ï‡</mi><mo separator="true">,</mo><mi>T</mi></mrow><annotation encoding="application/x-tex">kÏ‡,T</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mord mathnormal">Ï‡</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">T</span></span></span></span></span><span>ï»¿</span></span>) çš„æ¨¡æ‹Ÿã€‚OFè®¾ç½®ä¸º</blockquote><p id="88a0ebd4-e12c-4f7b-bac2-ff7a7ded1b92" class="">
</p><figure id="506306e3-b6b3-4c5c-aefb-2bf0842a77cb" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%201.png"><img style="width:288px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%201.png"/></a></figure><p id="714348c6-09b0-4c39-8879-988a69d61772" class=""><mark class="highlight-gray">Eq.(2)</mark></p><p id="90945aec-ba03-4c69-823e-3f3583aeb563" class="">
</p><p id="cdccce32-e780-46d3-aadf-4cbb36cc57b9" class="">                                   </p><p id="09325c7c-6140-40b3-be84-2ee48d0eeb3a" class="">The first response evaluates the average physical stability of simulated structures, the second is the median Ï‡2 deviation of simulated data from the target data. Owing to the ill-posed nature of the SAXS inverse problem, globally distinct protein structures can possess the same scattering intensity. As shown in Fig. 1, this can lead to a pronounced ambiguity in Ï‡2. To resolve the resulting non-injectivity in the OF, we introduce a third response, the inverse average Ï‡2 deviation. This acts as a regularizer, rewarding deviations from the target data and thus preventing possible overfitting. Combining these responses yields a surrogate model of a simulated ensembleâ€™s similarity to the desired target structure. The smaller the OF, the more physical, data-consistent and (likely) similar to the target state the simulated structures are.</p><blockquote id="716503f6-a3a0-485a-ae50-6202ba418822" class="">ç¬¬ä¸€ä¸ªæ˜¯å“åº”è¯„ä¼°æ¨¡æ‹Ÿç»“æ„çš„å¹³å‡ç‰©ç†ç¨³å®šæ€§ï¼Œç¬¬äºŒä¸ªæ˜¯å“åº”è¯„ä¼°æ¨¡æ‹Ÿæ•°æ®ä¸ç›®æ ‡æ•°æ®çš„ä¸­å€¼<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Ï‡</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">Ï‡2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">Ï‡</span><span class="mord">2</span></span></span></span></span><span>ï»¿</span></span>åå·®ã€‚ ç”±äºSAXSé€†é—®é¢˜çš„ä¸é€‚å®šæ€§è´¨ï¼Œå…¨å±€ä¸åŒçš„è›‹ç™½è´¨ç»“æ„å¯ä»¥å…·æœ‰ç›¸åŒçš„æ•£å°„å¼ºåº¦ã€‚ å¦‚å›¾1æ‰€ç¤º ,è¿™ä¼šå¯¼è‡´<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Ï‡</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">Ï‡2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">Ï‡</span><span class="mord">2</span></span></span></span></span><span>ï»¿</span></span>çš„æ˜æ˜¾æ­§ä¹‰ã€‚ ä¸ºäº†è§£å†³åœ¨OFä¸­äº§ç”Ÿçš„éå†…å°„æ€§ï¼Œæˆ‘ä»¬å¼•å…¥äº†ç¬¬ä¸‰ä¸ªå“åº”ï¼Œé€†å¹³å‡<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Ï‡</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">Ï‡2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">Ï‡</span><span class="mord">2</span></span></span></span></span><span>ï»¿</span></span>åå·®ã€‚ è¿™èµ·åˆ°äº†æ­£åˆ™åŒ–çš„ä½œç”¨ï¼Œå¥–åŠ±ä¸ç›®æ ‡æ•°æ®çš„åå·®ï¼Œä»è€Œé˜²æ­¢å¯èƒ½çš„è¿‡æ‹Ÿåˆã€‚ ç»„åˆè¿™äº›å“åº”äº§ç”Ÿäº†ä¸€ä¸ªæ¨¡æ‹Ÿé›†åˆä½“ä¸æœŸæœ›ç›®æ ‡ç»“æ„ç›¸ä¼¼åº¦çš„ä»£ç”¨æ¨¡å‹ã€‚ OFè¶Šå°ï¼Œæ¨¡æ‹Ÿç»“æ„çš„ç‰©ç†ã€æ•°æ®ä¸€è‡´æ€§å’Œï¼ˆå¯èƒ½ï¼‰ä¸ç›®æ ‡çŠ¶æ€è¶Šç›¸ä¼¼ã€‚</blockquote><p id="d35f3675-8b6a-4d56-8049-452cf8feab18" class="">In physico-empirical structure-based models, different combinations of bias weight and temperature can equally yield useful results. There is no MD parameter ground truth for this type of simulation, so a purely evidence-based evaluation according to the similarity of simulated protein structures to the target is to be applied.</p><blockquote id="c679af23-0ae1-4dd6-b6b6-143a25ce3a96" class="">åœ¨åŸºäºç‰©ç†ç»éªŒç»“æ„çš„æ¨¡å‹ä¸­ï¼Œåç½®æƒé‡å’Œæ¸©åº¦çš„ä¸åŒç»„åˆåŒæ ·å¯ä»¥äº§ç”Ÿæœ‰ç”¨çš„ç»“æœã€‚è¿™ç§ç±»å‹çš„æ¨¡æ‹Ÿæ²¡æœ‰ MD å‚æ•°çš„åŸºç¡€çœŸå€¼ï¼Œå› æ­¤å°†æ ¹æ®æ¨¡æ‹Ÿè›‹ç™½è´¨ç»“æ„ä¸ç›®æ ‡çš„ç›¸ä¼¼æ€§è¿›è¡Œçº¯ç²¹çš„åŸºäºè¯æ®çš„è¯„ä¼°ã€‚</blockquote><p id="544dc284-4171-44d3-84d4-a5d27197a8f7" class="">We use the global distance test (GDT)[40] to quantify differences between two conformations of a protein (Methods section â€˜Root-mean-square deviationâ€™). To estimate how similar two superimposed structures are, the displacement of each alpha carbon is compared to various distance cutoffs. Percentages, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>P</mi><mi>x</mi></msub></mrow><annotation encoding="application/x-tex">P_x</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.83333em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.151392em;"><span style="top:-2.5500000000000003em;margin-left:-0.13889em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">x</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> , of alpha carbons with displacements below cutoffs of x â„« are used to calculate the total score:</p><blockquote id="aa22a34a-0c1b-469d-a412-45a9852762ca" class="">æˆ‘ä»¬ä½¿ç”¨å…¨å±€è·ç¦»æµ‹è¯• (GDT)[40]æ¥é‡åŒ–è›‹ç™½è´¨çš„ä¸¤ç§æ„è±¡ä¹‹é—´çš„å·®å¼‚ï¼ˆæ–¹æ³•éƒ¨åˆ†â€œå‡æ–¹æ ¹åå·®â€ï¼‰ã€‚ä¸ºäº†ä¼°è®¡ä¸¤ä¸ªå åŠ ç»“æ„çš„ç›¸ä¼¼ç¨‹åº¦ï¼Œå°†æ¯ä¸ª <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Î±</mi></mrow><annotation encoding="application/x-tex">Î±</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.0037em;">Î±</span></span></span></span></span><span>ï»¿</span></span> ç¢³çš„ä½ç§»ä¸ä¸åŒè·ç¦»æˆªæ­¢å€¼è¿›è¡Œæ¯”è¾ƒã€‚ä½ç§»å€¼å°äºæˆªæ­¢å€¼ x Ã… çš„ <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Î±</mi></mrow><annotation encoding="application/x-tex"> Î± </annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.0037em;">Î±</span></span></span></span></span><span>ï»¿</span></span> ç¢³çš„ç™¾åˆ†æ¯”<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>P</mi><mi>x</mi></msub></mrow><annotation encoding="application/x-tex"> P_x </annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.83333em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.151392em;"><span style="top:-2.5500000000000003em;margin-left:-0.13889em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">x</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>ç”¨äºè®¡ç®—æ€»åˆ†ï¼š</blockquote><p id="fff38e97-e74c-47c7-827c-5dc0395346e0" class="">
</p><figure id="b8956a06-c147-4e76-91ad-f394a3fae944" class="image" style="text-align:center"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%202.png"><img style="width:419px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%202.png"/></a></figure><p id="09a10690-8b7f-4c6b-a970-3b0691501abf" class=""><mark class="highlight-gray">Eq.(3)</mark></p><p id="c5bea738-be6c-47ac-8503-97529b206a94" class="">                  </p><p id="03bae6ee-5d24-4098-8e31-f1a5f82be98b" class="">Higher GDT values indicate a stronger similarity between two models. Structures with GDT &gt; 50 are considered topologically accurate[41]. The GDT is used to validate the OF as a surrogate model of
an ensembleâ€™s similarity to the target structure.</p><blockquote id="fa7094a8-8d18-4c53-af59-c9d5ebc0c3db" class="">è¾ƒé«˜çš„GDTå€¼è¡¨æ˜ä¸¤ä¸ªæ¨¡å‹ä¹‹é—´çš„ç›¸ä¼¼æ€§æ›´å¼ºã€‚ GDT&gt;50çš„ç»“æ„è¢«è®¤ä¸ºåœ¨æ‹“æ‰‘å­¦ä¸Šæ˜¯å‡†ç¡®çš„[41]ã€‚ GDTç”¨äºéªŒè¯OFä½œä¸ºä¸€ä¸ªä»£ç”¨æ¨¡å‹çš„é›†åˆä½“ä¸ç›®æ ‡ç»“æ„çš„ç›¸ä¼¼æ€§ã€‚</blockquote><figure id="6262188f-427a-4d80-a4ef-af6c97836338" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%203.png"><img style="width:583px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%203.png"/></a></figure><p id="7d1220cc-1a62-4122-8679-76c9c7a84256" class=""><mark class="highlight-gray">Fig. 1 | Ambiguous Ï‡2. The ill-posedness of the SAXS inverse problem results in a pronounced ambiguity in the Ï‡2 deviation of simulated data from the target data. This manifests in a non-injective â€˜two-branchâ€™ behaviour of the second response as a function of similarity of simulated structures to the desired target state, here quantified by the GDT. GDTsys, GDT between initial and target protein structure; kÏ‡, bias weight; </mark><mark class="highlight-gray"><style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Îµ</mi></mrow><annotation encoding="application/x-tex">Îµ</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal">Îµ</span></span></span></span></span><span>ï»¿</span></span></mark><mark class="highlight-gray">, energy scale of the structure-based model. Data from all lysine-, arginine-, ornithine-binding protein holo-to-apo runs are combined.</mark></p><h2 id="13ac445b-9e1b-46e4-8e32-8ab3b407400b" class="">Results</h2><p id="c13104f7-f3ad-49e6-aca3-76e5a4540c4a" class="">We optimized MD parameters of SAXS-guided structure-based simulations for two well-characterized proteins: lysine-, arginine-, ornithine-binding (LAO) protein and adenylate kinase (ADK).</p><blockquote id="fd2131b5-1446-446c-b41f-ef2d6180b19d" class="">æˆ‘ä»¬ä¼˜åŒ–äº† SAXS å¼•å¯¼çš„åŸºäºç»“æ„æ¨¡æ‹Ÿçš„ MD å‚æ•°ï¼ŒåŒ…æ‹¬ä¸¤ç§ç‰¹å¾æ˜ç¡®çš„è›‹ç™½è´¨ï¼šèµ–æ°¨é…¸ã€ç²¾æ°¨é…¸ã€é¸Ÿæ°¨é…¸ç»“åˆ (LAO) è›‹ç™½å’Œè…ºè‹·é…¸æ¿€é…¶ (ADK)ã€‚</blockquote><p id="e952b2e5-2827-49dc-a287-12b307daea61" class="">Small ligands such as sugars and amino acids are actively transported into bacteria across cell membranes[42]. Dedicated transport systems comprise a receptor (that is, the binding protein) and a membrane-bound protein complex. Interactions of the ligated binding protein with the membrane components induce conformational changes in the latter, forming an entry pathway for the ligand. We study LAO protein (Fig. 2a), which undergoes a conformational change from an apo (unligated; Protein Data Bank43 (PDB) code 2LAO44) to a holo (ligated; PDB code 1LST44) state upon ligand binding. The structures have a GDT of 39.39. SAXS-guided simulations started from the holo state and aimed at the apo state.</p><blockquote id="2247bb75-6c33-402c-8e81-d2a9932df5ad" class="">ç³–å’Œæ°¨åŸºé…¸ç­‰å°é…ä½“é€šè¿‡ç»†èƒè†œ[42]ç§¯æåœ°è¢«è½¬è¿åˆ°ç»†èŒä¸­ã€‚ä¸“ç”¨è½¬è¿ç³»ç»ŸåŒ…æ‹¬å—ä½“ï¼ˆå³ç»“åˆè›‹ç™½ï¼‰å’Œè†œç»“åˆè›‹ç™½å¤åˆç‰©ã€‚è¿æ¥çš„ç»“åˆè›‹ç™½ä¸è†œæˆåˆ†çš„ç›¸äº’ä½œç”¨è¯±å¯¼åè€…çš„æ„è±¡å˜åŒ–ï¼Œå½¢æˆé…ä½“çš„è¿›å…¥é€”å¾„ã€‚æˆ‘ä»¬ç ”ç©¶ LAOäº† è›‹ç™½ï¼ˆå›¾ 2aï¼‰ï¼Œå®ƒåœ¨é…ä½“ç»“åˆæ—¶ç»å†äº†ä» APOï¼ˆæœªè¿æ¥ï¼›è›‹ç™½è´¨æ•°æ®åº“[43]ï¼ˆPDBï¼‰ä»£ç  2LAO[44]ï¼‰åˆ°holoï¼ˆè¿æ¥ï¼›PDB ä»£ç  1LST[44]ï¼‰çŠ¶æ€çš„æ„è±¡å˜åŒ–ã€‚è¿™äº›ç»“æ„çš„ GDT ä¸º 39.39ã€‚ SAXS å¼•å¯¼çš„æ¨¡æ‹Ÿä»holoçŠ¶æ€å¼€å§‹ä»¥ apo çŠ¶æ€ä¸ºç›®æ ‡ã€‚</blockquote><p id="3a8c5807-3acd-44ec-8970-a7938ca586f5" class="">Adenosine triphosphate (ATP) is the universal energy source in cells and is vital to processes such as muscle contraction and nerve impulse propagation. By continuously checking ATP levels, ADK provides the cell with a mechanism to monitor energetic levels and metabolic processes. The transition between an open (PDB code 4AKE45) and closed (PDB code 1AKE46) state is quintessen-
tial to its catalytic function47. The structures have a GDT of 33.06. SAXS-guided simulations started from the open conformation and aimed at the closed one.</p><blockquote id="2ad7537e-0b53-4838-a7e7-78430422abc9" class="">ä¸‰ç£·é…¸è…ºè‹·(ATP)æ˜¯ç»†èƒä¸­æ™®éå­˜åœ¨çš„èƒ½é‡æ¥æºï¼Œå¯¹è‚Œè‚‰æ”¶ç¼©å’Œç¥ç»å†²åŠ¨ä¼ æ’­ç­‰è¿‡ç¨‹è‡³å…³é‡è¦ã€‚é€šè¿‡ä¸æ–­æ£€æµ‹ATPæ°´å¹³ï¼ŒADKä¸ºç»†èƒæä¾›äº†ä¸€ç§ç›‘æµ‹èƒ½é‡æ°´å¹³å’Œä»£è°¢è¿‡ç¨‹çš„æœºåˆ¶ã€‚æ‰“å¼€çŠ¶æ€ï¼ˆPDBä»£ç 4AKE[45])å’Œå…³é—­çŠ¶æ€ï¼ˆPDBä»£ç 1AKE[46])ä¹‹é—´çš„è½¬æ¢æ˜¯å…¶å‚¬åŒ–åŠŸèƒ½çš„å…¸å‹ç‰¹å¾[47]ã€‚è¿™äº›ç»“æ„çš„GDTä¸º33.06ã€‚SAXSå¼•å¯¼çš„æ¨¡æ‹Ÿä»å¼€æ”¾æ„è±¡å¼€å§‹ï¼Œä»¥å°é—­æ„è±¡ä¸ºç›®æ ‡ã€‚</blockquote><p id="760099ea-fa39-4022-bfed-252103fed0c6" class="">Artificial target data were calculated from known structures with CRYSOL48. Statistical uncertainties were modelled following ref. [49].</p><blockquote id="7ddf5a14-594b-414b-9772-a2168ad23afb" class="">äººå·¥ç›®æ ‡æ•°æ®æ˜¯ç”¨CRYSOL[48]çš„å·²çŸ¥ç»“æ„è®¡ç®—å‡ºæ¥çš„ã€‚ç»Ÿè®¡ä¸Šçš„ä¸ç¡®å®šæ€§æ˜¯æŒ‰ç…§å‚è€ƒæ–‡çŒ®å»ºç«‹çš„[49]ã€‚</blockquote><figure id="db110c5d-dac6-415f-94df-272fdd1d7578" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%204.png"><img style="width:601px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%204.png"/></a></figure><p id="be89b13e-ad77-473e-a2d2-2c5cc629d142" class=""><mark class="highlight-gray">Fig. 2 | Test systems. a, Lysine-, arginine-, ornithine-binding protein in its holo (coloured) and apo (grey) state, with a GDT of 39.39. b, Adenylate kinase in its open (coloured) and closed (grey) state, with a GDT of 33.06. The colouring indicates the displacement of each alpha carbon in the initial structure with respect to the target state. The average alpha-carbon displacement in each coloured structure with respect to each grey structure is given. Structures are visualized with PyMOL58.</mark></p><p id="fa018a68-4101-4463-a63e-3042771f94dd" class="">
</p><p id="89a7760b-1e0f-4d96-ada6-47f45059fb3b" class="">We performed seven FLAPS runs with different initial conditions for each protein. Swarm-based metaheuristics such as FLAPS have hyperparameters influencing the optimization behaviour, and
their efficacy can only be demonstrated empirically by a finite number of computational experiments. We present results for a swarm of 10 particles and 15 generations as a workable trade-off between optimization performance and compute time for the considered application. This set-up was found to be sufficient for convergence in preceding trial runs. Calculations were performed on 1,000 cores of a supercomputer. One run cost ~40,000 core hours. The results of the three best runs are listed in Table 1 (complete results are provided in Supplementary Tables 1 and 2).</p><blockquote id="8a6f0382-4134-42ea-b528-666fc292b5e6" class="">æˆ‘ä»¬å¯¹æ¯ç§è›‹ç™½è´¨ä½¿ç”¨ä¸åŒçš„åˆå§‹æ¡ä»¶è¿›è¡Œäº†ä¸ƒæ¬¡ FLAPS è¿è¡Œã€‚åŸºäºç¾¤ä½“çš„å…ƒå¯å‘å¼ç®—æ³•ï¼ˆä¾‹å¦‚ FLAPSï¼‰å…·æœ‰å½±å“ä¼˜åŒ–è¡Œä¸ºçš„è¶…å‚æ•°ï¼Œå¹¶ä¸”å®ƒä»¬çš„åŠŸæ•ˆåªèƒ½é€šè¿‡æœ‰é™æ•°é‡çš„è®¡ç®—å®éªŒæ¥è¯æ˜ã€‚æˆ‘ä»¬å±•ç¤ºäº†ä¸€ç»„ ç”±10 ä¸ªç²’å­å’Œ 15 ä»£ç»„æˆçš„ç»“æœï¼Œä½œä¸ºæ‰€è€ƒè™‘åº”ç”¨ç¨‹åºçš„ä¼˜åŒ–æ€§èƒ½å’Œè®¡ç®—æ—¶é—´ä¹‹é—´çš„å¯è¡Œæƒè¡¡ã€‚å‘ç°è¿™ç§è®¾ç½®è¶³ä»¥å®ç°æ”¶æ•›ã€‚è®¡ç®—æ˜¯åœ¨ä¸€å°è¶…çº§è®¡ç®—æœºçš„ 1,000 ä¸ªæ ¸å¿ƒä¸Šè¿›è¡Œçš„ã€‚ä¸€æ¬¡è¿è¡Œæˆæœ¬çº¦ä¸º 40,000 æ ¸å¿ƒå°æ—¶ã€‚è¡¨1åˆ—å‡ºäº†ä¸‰ä¸ªæœ€ä½³è¿è¡Œçš„ç»“æœï¼ˆå®Œæ•´çš„ç»“æœè§è¡¥å……è¡¨ 1 å’Œ 2ï¼‰ã€‚</blockquote><figure id="5328cb32-fc1c-49ef-8426-bbecf88d0eb8" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%205.png"><img style="width:1198px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%205.png"/></a></figure><p id="28f9f094-6a0f-4d95-8b89-3ca9b0fef4cf" class="">As shown for LAO protein in Fig. 3, the OF consistently converged to a stable topology (Supplementary section â€˜Analyzing swarm convergenceâ€™).</p><blockquote id="c097d37c-2405-445f-bbc1-d181805ee4f5" class="">LAO è›‹ç™½å¦‚å›¾ 3 æ‰€ç¤ºï¼ŒOF å§‹ç»ˆæ”¶æ•›åˆ°ç¨³å®šçš„æ‹“æ‰‘ç»“æ„ï¼ˆè¡¥å……éƒ¨åˆ†â€œåˆ†æç¾¤æ”¶æ•›â€ï¼‰ã€‚</blockquote><figure id="1d890102-b703-4397-819f-9ba73ea5e610" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%206.png"><img style="width:1027px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%206.png"/></a></figure><p id="e2d6a06e-d8ce-4dbe-9ee1-33e97a7c819e" class=""><mark class="highlight-gray">Fig. 3 | Dynamically evolving OF topology. Results are shown for LAO protein (seed 1790954). The current global best position is marked by a star. </mark><mark class="highlight-gray"><style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Îµ</mi></mrow><annotation encoding="application/x-tex">Îµ</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal">Îµ</span></span></span></span></span><span>ï»¿</span></span></mark><mark class="highlight-gray"> is the energy scale of the structure-based model.</mark></p><p id="2b3957cd-e448-43ca-a4bb-82a3ac67e8bc" class="">
</p><p id="38f6eea4-f0e4-4789-bcf6-378d1c31e8c3" class="">For each simulation, we calculated the median GDT with respect to the target state from all structures in the trajectory. To validate the OF, we state its Pearson correlation Ï with the median GDT as a measure of linear correlation. Because minimizing the OF should be equivalent to maximizing the GDT, negative correlations, ideally âˆ’1, are expected. The OFâ€™s suitability is confirmed for both LAO protein and ADK with correlations up to âˆ’0.94 and âˆ’0.85, respectively.</p><blockquote id="131fa4ac-a60a-4b88-b884-4c7108370e59" class="">å¯¹äºæ¯ä¸€æ¬¡æ¨¡æ‹Ÿï¼Œæˆ‘ä»¬è®¡ç®—äº†è½¨è¿¹ä¸­æ‰€æœ‰ç»“æ„ç›¸å¯¹äºç›®æ ‡çŠ¶æ€çš„ GDTä¸­å€¼ã€‚ä¸ºäº†éªŒè¯ OFï¼Œæˆ‘ä»¬å°†å…¶ä¸Pearson ç›¸å…³åº¦<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Ï</mi></mrow><annotation encoding="application/x-tex"> Ï</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">Ï</span></span></span></span></span><span>ï»¿</span></span> ä½œä¸ºçº¿æ€§ç›¸å…³çš„åº¦é‡ã€‚å› ä¸ºæœ€å°åŒ– OF åº”è¯¥ç­‰åŒäºæœ€å¤§åŒ– GDTï¼Œæ‰€ä»¥é¢„è®¡ä¼šå‡ºç°è´Ÿç›¸å…³ï¼ˆç†æƒ³æƒ…å†µä¸‹ä¸º -1ï¼‰ã€‚OF å¯¹ LAO è›‹ç™½å’Œ ADK çš„é€‚ç”¨æ€§éƒ½å¾—åˆ°äº†è¯å®ï¼Œç›¸å…³æ€§åˆ†åˆ«è¾¾åˆ°-0.94 å’Œ -0.85ã€‚</blockquote><p id="341f0729-fe31-4ad7-9f0e-55b87bc5a67c" class="">As the Pearson correlation only reflects linearity, we also studied the exact relations between the OF and the median GDT (Fig. 4). To identify the best MD parameter combinations, the OF must have
low values for large GDTs, irrespective of the actual relationshipâ€™s complexity. This is the case for both test systems.</p><blockquote id="eb6b4ef9-877d-48ca-b531-ad018982be48" class="">ç”±äº Pearson ç›¸å…³ä»…åæ˜ çº¿æ€§ï¼Œæˆ‘ä»¬è¿˜ç ”ç©¶äº† OF å’Œ GDTä¸­å€¼ä¹‹é—´çš„ç²¾ç¡®å…³ç³»ï¼ˆå›¾ 4ï¼‰ã€‚ä¸ºäº†ç¡®å®šæœ€ä½³çš„ MD å‚æ•°ç»„åˆï¼Œæ— è®ºå®é™…å…³ç³»çš„å¤æ‚æ€§å¦‚ä½•ï¼Œå¯¹äºå¤§å‹ GDTï¼ŒOF å¿…é¡»å…·æœ‰è¾ƒä½çš„å€¼ï¼Œè¿™ä¸¤ä¸ªæµ‹è¯•ç³»ç»Ÿéƒ½æ˜¯è¿™æ ·ã€‚</blockquote><figure id="80137984-c829-4836-aff6-786c66cb2af9" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%207.png"><img style="width:1026px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%207.png"/></a></figure><p id="12373b4f-3957-4b1d-a4f1-0f65ad6d0765" class=""><mark class="highlight-gray">Fig. 4 | OF versus median GDT. a,b, OF versus median GDT for LAO protein (a) and ADK (b). GDTsys
, GDT between initial and target structure of each test system; Îµ, energy scale of the structure-based model.</mark></p><p id="3f6ac5d3-3434-49c8-ab23-880c3c06a008" class="">
</p><p id="f896714f-277c-4391-950b-f060d0b9c157" class="">Global best positions, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>, returned functional MD parameter combinations throughout. For LAO protein, <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> median GDTs were consistently of the order of 70 and correspond well to the best values reached. This means that for half of the simulated structures, at least 70% of all alpha carbons lie within a small radius from their positions in the target state. These results indicate the structural accuracy of the simulated ensemble for <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> and thus convergence to the target state and successful refinement against the data. The same is true for ADK, where <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> and the best median GDTs are around 63 and more similar than those of LAO protein. Example structures from <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> simulations, shown in Fig. 5, are in nearly perfect accordance with the target states.</p><blockquote id="8dbe8b81-518f-429f-a073-13085e727bd5" class="">å…¨å±€æœ€ä½³ä½ç½®<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>ï¼Œè¿”å›æ•´ä¸ªåŠŸèƒ½MDå‚æ•°ç»„åˆã€‚å¯¹äºLAOè›‹ç™½ï¼Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> çš„ä¸­å€¼GDTsä¸€ç›´åœ¨70å·¦å³ï¼Œå¾ˆå¥½çš„å¯¹åº”äºè¾¾åˆ°çš„æœ€ä½³å€¼ã€‚è¿™æ„å‘³ç€åœ¨ä¸€åŠçš„æ¨¡æ‹Ÿç»“æ„ä¸­ï¼Œè‡³å°‘æœ‰70%çš„<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Î±</mi></mrow><annotation encoding="application/x-tex">Î±</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.0037em;">Î±</span></span></span></span></span><span>ï»¿</span></span>ç¢³åŸå­ä½äºç›®æ ‡çŠ¶æ€ä½ç½®çš„å°åŠå¾„èŒƒå›´å†…ã€‚è¿™äº›ç»“æœè¡¨æ˜äº†<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>æ¨¡æ‹Ÿé›†åˆçš„ç»“æ„å‡†ç¡®æ€§ï¼Œä»è€Œæ”¶æ•›åˆ°ç›®æ ‡çŠ¶æ€ï¼Œå¹¶æˆåŠŸåœ°å¯¹æ•°æ®è¿›è¡Œç»†åŒ–ã€‚ADKçš„æƒ…å†µä¹Ÿæ˜¯å¦‚æ­¤ï¼Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>å’Œæœ€ä½³ä¸­å€¼GDTsåœ¨63å·¦å³ï¼Œæ¯”LAOè›‹ç™½çš„GDTsæ›´ç›¸ä¼¼ã€‚æ¥è‡ª<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>æ¨¡æ‹Ÿçš„ç»“æ„å®ä¾‹å¦‚å›¾5æ‰€ç¤ºï¼Œä¸ç›®æ ‡çŠ¶æ€å‡ ä¹å®Œå…¨ä¸€è‡´ã€‚</blockquote><p id="35c5a361-a686-432f-8ae0-ecdd96ec776f" class="">Additionally, we considered the reversed conformational transitions, that is, from apo to holo state for LAO protein and closed to open state for ADK (Supplementary Tables 3 and 4, respectively).
With <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Ï</mi></mrow><annotation encoding="application/x-tex">Ï</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">Ï</span></span></span></span></span><span>ï»¿</span></span> between âˆ’0.56 and âˆ’0.88, FLAPS could also identify functional MD parameters for SAXS-guided apo-to-holo simulations of LAO protein. <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> and the best median GDTs were consistently slightly below 70, indicating high similarity of the simulated structures and the desired target state for the MD parameter combinations found.</p><blockquote id="14c5118a-6825-4762-b491-4b8eed7c1d5f" class="">æ­¤å¤–ï¼Œæˆ‘ä»¬è€ƒè™‘äº†åå‘æ„è±¡è½¬å˜ï¼Œå³ LAO è›‹ç™½ä» apo åˆ°å…¨æ¯çŠ¶æ€ï¼ŒADK å…³é—­åˆ°å¼€æ”¾çŠ¶æ€ï¼ˆåˆ†åˆ«è¡¥å……è¡¨ 3 å’Œ 4ï¼‰ã€‚<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Ï</mi></mrow><annotation encoding="application/x-tex">Ï</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">Ï</span></span></span></span></span><span>ï»¿</span></span>åœ¨-0.56å’Œ-0.88ä¹‹é—´ï¼ŒFLAPSä¹Ÿå¯ä»¥ä¸ºSAXSå¼•å¯¼çš„LAOè›‹ç™½çš„apo-to-holoæ¨¡æ‹Ÿç¡®å®šåŠŸèƒ½æ€§MDå‚æ•°ã€‚<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>å’Œæœ€ä½³ä¸­å€¼GDTså§‹ç»ˆç•¥ä½äº70ï¼Œè¡¨æ˜æ¨¡æ‹Ÿç»“æ„å’Œæ‰¾åˆ°çš„MDå‚æ•°ç»„åˆçš„æœŸæœ›ç›®æ ‡çŠ¶æ€é«˜åº¦ç›¸ä¼¼ã€‚</blockquote><p id="4909cf97-586d-4b67-9dd9-8d9b6698f637" class="">By comparison, Pearson correlations up to âˆ’0.53 and <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> median GDTs of ~45 indicate only average structural accuracy for the closed-to-open transition of ADK. However, overall best median
GDTs were around 50. With only half of all alpha carbons within a small radius from their target positions, the observed behaviour is not a problem of FLAPS, but is due to the limits of the underlying simulation method[7]. The information in the coarse-grained structure-based
model and the low-resolution SAXS data seems to be insufficient to determine the molecular structure with the same accuracy as for the other test cases. However, even under these circumstances, FLAPS was capable of determining acceptable parameters.</p><blockquote id="af74d2b1-195d-465e-93af-4bc5dbaecf11" class="">ç›¸æ¯”ä¹‹ä¸‹ï¼Œé«˜è¾¾ -0.53 çš„ Pearson ç›¸å…³æ€§å’Œçº¦ 45 çš„ <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>g</mi><mrow><mi>b</mi><mi>e</mi><mi>s</mi><mi>t</mi></mrow></msub></mrow><annotation encoding="application/x-tex">g_{best}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.03588em;">g</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em;"><span style="top:-2.5500000000000003em;margin-left:-0.03588em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">b</span><span class="mord mathnormal mtight">es</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> ä¸­å€¼ GDT ä»…è¡¨æ˜ ADK çš„å°é—­åˆ°å¼€æ”¾è½¬å˜çš„å¹³å‡ç»“æ„ç²¾åº¦ã€‚ç„¶è€Œï¼Œæ€»ä½“æœ€ä½³ GDTs ä¸­å€¼çº¦ä¸º 50ã€‚ç”±äºåªæœ‰ä¸€åŠçš„ <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Î±</mi></mrow><annotation encoding="application/x-tex">Î±</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal" style="margin-right:0.0037em;">Î±</span></span></span></span></span><span>ï»¿</span></span> ç¢³åœ¨å…¶ç›®æ ‡ä½ç½®çš„å¾ˆå°çš„åŠå¾„èŒƒå›´å†…ï¼Œè§‚å¯Ÿåˆ°çš„è¡Œä¸ºä¸æ˜¯ FLAPS çš„é—®é¢˜ï¼Œè€Œæ˜¯ç”±äºåŸºç¡€æ¨¡æ‹Ÿæ–¹æ³•çš„é™åˆ¶[7]ã€‚åŸºäºç²—ç²’åº¦ç»“æ„çš„æ¨¡å‹å’Œä½åˆ†è¾¨ç‡ SAXS æ•°æ®ä¸­çš„ä¿¡æ¯ä¼¼ä¹ä¸è¶³ä»¥ä¸å…¶ä»–æµ‹è¯•ç”¨ä¾‹ç›¸åŒçš„ç²¾åº¦ç¡®å®šåˆ†å­ç»“æ„ã€‚ç„¶è€Œï¼Œå³ä½¿åœ¨è¿™äº›æƒ…å†µä¸‹ï¼ŒFLAPS ä¹Ÿèƒ½å¤Ÿç¡®å®šå¯æ¥å—çš„å‚æ•°ã€‚</blockquote><p id="200f7a78-b343-4e22-86d7-853521a8b6d9" class="">To evaluate FLAPSâ€™s efficiency, we performed comparative grid-search optimizations where we found superior performance of FLAPS for all considered protein systems (Supplementary section
â€˜Comparison to grid searchâ€™).</p><blockquote id="c948318f-c3c4-4985-a5fe-da11755a3b0f" class="">ä¸ºäº†è¯„ä¼° FLAPS çš„æ•ˆç‡ï¼Œæˆ‘ä»¬è¿›è¡Œäº†ç½‘æ ¼æœç´¢æ¯”è¾ƒä¼˜åŒ–ï¼Œæˆ‘ä»¬å‘ç° FLAPS å¯¹æ‰€æœ‰è€ƒè™‘çš„è›‹ç™½è´¨ç³»ç»Ÿéƒ½æœ‰ä¼˜è¶Šçš„æ€§èƒ½ï¼ˆè¡¥å……éƒ¨åˆ†â€œä¸ç½‘æ ¼æœç´¢çš„æ¯”è¾ƒâ€ï¼‰ã€‚</blockquote><figure id="bf8c89e9-2d44-4396-a10c-b1626ff2998c" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%208.png"><img style="width:585px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%208.png"/></a></figure><p id="e56bc253-c93a-44ee-a109-6cfddafa0f12" class=""><mark class="highlight-gray">Fig. 5 | Representative structures from global best simulations. a,b, LAO protein (seed 1790954) (a) and ADK (seed 1795691) (b). Structures with maximum GDT are shown (coloured) and are almost identical to the respective target states (grey). The colouring indicates the displacement of each alpha carbon in the simulated structure with respect to the target state. The average alpha-carbon displacement in each coloured structure with respect to each grey structure is given. Structures are visualized with PyMOL[58].</mark></p><h2 id="ce042da4-1648-4c51-85e5-32d72dbf198e" class="">Discussion</h2><p id="e0908ac8-631b-4776-a943-6bff1dbfbf1e" class="">The inverse problem of reconstructing molecular structures from low-resolution SAXS data is still unsolved. Biomolecular simulations are among the most powerful tools for eliminating the arising
ambiguity and access the valuable structural information content of such data. However, data-assisted simulations rely on MD parameters, where, most importantly, experimental information must be weighted accurately with respect to the physical model.</p><blockquote id="7cedfb06-2f9d-4304-8052-9c15fc4a3ca8" class="">ä»ä½åˆ†è¾¨ç‡ SAXS æ•°æ®é‡å»ºåˆ†å­ç»“æ„çš„é€†é—®é¢˜ä»æœªè§£å†³ã€‚ç”Ÿç‰©åˆ†å­æ¨¡æ‹Ÿæ˜¯æ¶ˆé™¤äº§ç”Ÿçš„æ­§ä¹‰å’Œè·å–æ­¤ç±»æ•°æ®çš„æœ‰ä»·å€¼çš„ç»“æ„ä¿¡æ¯çš„æœ€å¼ºå¤§å·¥å…·ä¹‹ä¸€ã€‚ç„¶è€Œï¼Œæ•°æ®è¾…åŠ©æ¨¡æ‹Ÿä¾èµ–äº MD å‚æ•°ï¼Œå…¶ä¸­æœ€é‡è¦çš„æ˜¯ï¼Œå®éªŒä¿¡æ¯å¿…é¡»ç›¸å¯¹äºç‰©ç†æ¨¡å‹å‡†ç¡®åŠ æƒã€‚</blockquote><p id="4ef30628-60b5-43c3-8a23-ccc3e55df2bd" class="">Here, we have shown how computational intelligence can be used to systematically explore MD parameter spaces and optimize the performance of complex physics-based simulation techniques.
We introduced FLAPS, a data-driven solution for a fully automatic and reproducible parameter search based on particle swarms.</p><blockquote id="4a7632c6-5ebb-4f43-bc5c-4ab9b1c1e5c1" class="">åœ¨è¿™é‡Œï¼Œæˆ‘ä»¬å±•ç¤ºäº†å¦‚ä½•ä½¿ç”¨è®¡ç®—æ™ºèƒ½æ¥ç³»ç»Ÿåœ°æ¢ç´¢ MD å‚æ•°ç©ºé—´å¹¶ä¼˜åŒ–åŸºäºç‰©ç†çš„å¤æ‚æ¨¡æ‹ŸæŠ€æœ¯çš„æ€§èƒ½ã€‚æˆ‘ä»¬ä»‹ç»äº† FLAPSï¼Œè¿™æ˜¯ä¸€ç§åŸºäºç²’å­ç¾¤çš„å…¨è‡ªåŠ¨ä¸”å¯é‡å¤çš„å‚æ•°æœç´¢çš„æ•°æ®é©±åŠ¨çš„è§£å†³æ–¹æ¡ˆã€‚</blockquote><p id="824f8194-ef0e-473b-a080-d4df8448f53e" class="">To identify the best MD parameters for SAXS-guided protein simulations, we designed an OF as an accurate surrogate of simulation quality in terms of physical structures matching the target data. A suitable OF will typically depend on multiple quality features of different scales to equally reflect a data-assisted simulationâ€™s physical plausibility and its agreement with the data. To handle multiple responses in classical PSO, they need to be mapped to a scalar score via multiplication by fixed weights. These additional OF parameters must either be chosen manually (and probably suboptimally) in advance or be absorbed into the search space, resulting in a massive increase in dimensionality. FLAPS solves this problem by intelligently learning OF parameters in the optimization process, avoiding the need to set them as â€˜magic numbersâ€™, while reducing the search-space dimensionality to a minimum. Various responses are automatically balanced with respect to each other to enable a meaningful and unbiased comparison on a shared scale. Implemented in FLAPS, our conceptual OF reliably identified useful MD parameters for two different proteins, where we observed convergence of simulated structures to the target state. Owing to the dedicated in situ processing, the algorithm leverages available computing resources and transparently scales from a laptop to current supercomputers.</p><blockquote id="1dbd793a-020b-4b43-8754-acffc3b81901" class="">ä¸ºäº†ç¡®å®š SAXS å¼•å¯¼çš„è›‹ç™½è´¨æ¨¡æ‹Ÿçš„æœ€ä½³ MD å‚æ•°ï¼Œæˆ‘ä»¬æ ¹æ®ä¸ç›®æ ‡æ•°æ®åŒ¹é…çš„ç‰©ç†ç»“æ„è®¾è®¡äº†ä¸€ä¸ª OF ä½œä¸ºæ¨¡æ‹Ÿè´¨é‡çš„å‡†ç¡®æ›¿ä»£ç‰©ã€‚åˆé€‚çš„ OF é€šå¸¸å–å†³äºä¸åŒå°ºåº¦çš„å¤šä¸ªè´¨é‡ç‰¹å¾ï¼Œä»¥åŒæ ·åæ˜ æ•°æ®è¾…åŠ©æ¨¡æ‹Ÿçš„ç‰©ç†åˆç†æ€§åŠå…¶ä¸æ•°æ®çš„ä¸€è‡´æ€§ã€‚ä¸ºäº†åœ¨ç»å…¸ PSO ä¸­å¤„ç†å¤šä¸ªå“åº”ï¼Œéœ€è¦é€šè¿‡ä¹˜ä»¥å›ºå®šæƒé‡å°†å®ƒä»¬æ˜ å°„åˆ°æ ‡é‡åˆ†æ•°ã€‚è¿™äº›é¢å¤–çš„ OF å‚æ•°è¦ä¹ˆå¿…é¡»æå‰äººå·¥æ‰‹åŠ¨é€‰æ‹©ï¼ˆå¯èƒ½æ˜¯æ¬¡ä¼˜çš„ï¼‰ï¼Œè¦ä¹ˆè¢«å¸æ”¶åˆ°æœç´¢ç©ºé—´ä¸­ï¼Œä»è€Œå¯¼è‡´ç»´åº¦çš„å¤§é‡å¢åŠ ã€‚FLAPSé€šè¿‡åœ¨ä¼˜åŒ–è¿‡ç¨‹ä¸­æ™ºèƒ½åœ°å­¦ä¹ OFå‚æ•°æ¥è§£å†³è¿™ä¸ªé—®é¢˜ï¼Œé¿å…å°†å®ƒä»¬è®¾ç½®ä¸º &quot;ç¥å¥‡çš„æ•°å­—&quot;ï¼ŒåŒæ—¶å°†æœç´¢ç©ºé—´çš„ç»´åº¦é™åˆ°æœ€ä½ã€‚å„ç§åæ˜ è¢«è‡ªåŠ¨åœ°ç›¸äº’å¹³è¡¡ï¼Œä»¥ä¾¿åœ¨ä¸€ä¸ªå…±åŒçš„å°ºåº¦ä¸Šè¿›è¡Œæœ‰æ„ä¹‰çš„å’Œæ— åè§çš„æ¯”è¾ƒã€‚åœ¨FLAPSå®æ–½ä¸­ï¼Œæˆ‘ä»¬çš„æ¦‚å¿µOFå¯é åœ°ç¡®å®šäº†ä¸¤ç§ä¸åŒè›‹ç™½è´¨çš„æœ‰ç”¨çš„MDå‚æ•°ï¼Œæˆ‘ä»¬è§‚å¯Ÿåˆ°æ¨¡æ‹Ÿç»“æ„å‘ç›®æ ‡çŠ¶æ€çš„æ”¶æ•›ã€‚ç”±äºä¸“é—¨çš„åŸä½å¤„ç†ï¼Œè¯¥ç®—æ³•å……åˆ†åˆ©ç”¨äº†å¯ç”¨çš„è®¡ç®—èµ„æºï¼Œå¹¶é€æ˜åœ°ä»ç¬”è®°æœ¬ç”µè„‘æ‰©å±•åˆ°å½“å‰çš„è¶…çº§è®¡ç®—æœºã€‚</blockquote><p id="354f9d3e-3218-44ca-aa22-1d08754381cb" class="">In previous studies, the bias weight is conveniently chosen as the smallest value yielding satisfactory Ï‡2 (refs. [7,8]). This purely data-based selection criterion neglects the physical information provided by the molecular simulations, risking the selection of physically dysfunctional values due to the ill-posedness of the SAXS inverse problem. The flexible composite OF in FLAPS allows us to include multiple selection criteria, yielding a direct surrogate of simulation quality in terms of not only data conformity but also the physical plausibility of simulated structures. In contrast to grid search, the optimization does not rely on a predefined list of manually chosen and a priori fixed candidates. This circumvents either missing an optimum, spanning large search spaces due to fine-grained grids, or both. Instead, a directional search led by the most current best solutions provides guidance in the selection process and a valid context for meaningful interpretation of multi-layer
criteria. Additionally, the compute time to find useful parameters is decreased by reducing the dimensionality.</p><blockquote id="31b9f1cf-f085-41df-a457-bf63727d2861" class="">åœ¨ä»¥å‰çš„ç ”ç©¶ä¸­ï¼Œåå·®æƒé‡è¢«æ–¹ä¾¿åœ°é€‰æ‹©ä¸ºäº§ç”Ÿä»¤äººæ»¡æ„çš„ <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Ï‡</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">Ï‡2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em;"></span><span class="mord mathnormal">Ï‡</span><span class="mord">2</span></span></span></span></span><span>ï»¿</span></span> çš„æœ€å°å€¼ï¼ˆå‚è€ƒæ–‡çŒ® [7,8]) ã€‚è¿™ç§çº¯ç²¹åŸºäºæ•°æ®çš„é€‰æ‹©æ ‡å‡†å¿½ç•¥äº†åˆ†å­æ¨¡æ‹Ÿæä¾›çš„ç‰©ç†ä¿¡æ¯ï¼Œç”±äº SAXS é€†é—®é¢˜çš„ä¸é€‚å®šæ€§ï¼Œå†’ç€é€‰æ‹©ç‰©ç†åŠŸèƒ½å¤±è°ƒå€¼çš„é£é™©ã€‚FLAPS ä¸­çš„çµæ´»ç»„åˆ OF å…è®¸æˆ‘ä»¬åŒ…å«å¤šä¸ªé€‰æ‹©æ ‡å‡†ï¼Œä»è€Œåœ¨æ•°æ®ä¸€è‡´æ€§å’Œæ¨¡æ‹Ÿç»“æ„çš„ç‰©ç†åˆç†æ€§æ–¹é¢ç›´æ¥æ›¿ä»£æ¨¡æ‹Ÿè´¨é‡ã€‚ä¸ç½‘æ ¼æœç´¢ç›¸æ¯”ï¼Œä¼˜åŒ–ä¸ä¾èµ–äºæ‰‹åŠ¨é€‰æ‹©çš„é¢„å®šä¹‰åˆ—è¡¨å’Œå…ˆéªŒå›ºå®šå€™é€‰è€…ã€‚è¿™é¿å…äº†ç”±äºç»†ç²’åº¦ç½‘æ ¼è€Œé”™è¿‡æœ€ä½³ã€è·¨è¶Šå¤§å‹æœç´¢ç©ºé—´ï¼Œæˆ–ä¸¤è€…å…¼è€Œæœ‰ä¹‹ã€‚ç›¸åï¼Œç”±æœ€æ–°çš„æœ€ä½³è§£å†³æ–¹æ¡ˆé¢†å¯¼çš„å®šå‘æœç´¢ä¸ºé€‰æ‹©è¿‡ç¨‹æä¾›äº†æŒ‡å¯¼ï¼Œå¹¶ä¸ºæœ‰æ„ä¹‰åœ°è§£é‡Šå¤šå±‚æ ‡å‡†æä¾›äº†æœ‰æ•ˆçš„ä¸Šä¸‹æ–‡ã€‚æ­¤å¤–ï¼Œé€šè¿‡å‡å°‘ç»´åº¦æ¥å‡å°‘å¯»æ‰¾æœ‰ç”¨å‚æ•°çš„è®¡ç®—æ—¶é—´ã€‚</blockquote><p id="8fdaf7fd-e0d1-4187-85c4-77036a8e9918" class="">FLAPS can be transferred easily to optimize other life-sciences applications, such as simulation methods incorporating data from various experimental sources[11]. In a more general context, it solves the problem of weighting different contributions of composite OFs in multi-response optimization. Such problems often occur in industrial manufacturing, processing and design, where manually chosen weights are commonly used[32,33]. In FLAPS, OF parameters can be learned in a self-improving manner according to any desired scheme, for example, standardization, rescaling, mean normalization or a relative weighting after one of the aforementioned steps. FLAPS shows how computational-intelligence concepts can be harnessed successfully for practical optimization problems at the forefront of life sciences.</p><blockquote id="43507803-2406-447b-b7ff-2c18c4e849ae" class="">FLAPS å¯ä»¥è½»æ¾è½¬æ¢ä»¥ä¼˜åŒ–å…¶å®ƒç”Ÿå‘½ç§‘å­¦åº”ç”¨ï¼Œä¾‹å¦‚ç»“åˆæ¥è‡ªå„ç§å®éªŒæ¥æºçš„æ•°æ®çš„æ¨¡æ‹Ÿæ–¹æ³•[11]ã€‚åœ¨æ›´å¹¿æ³›çš„æƒ…å†µä¸‹ï¼Œå®ƒè§£å†³äº†åœ¨å¤šå“åº”ä¼˜åŒ–ä¸­åŠ æƒå¤åˆ OFs çš„ä¸åŒè´¡çŒ®çš„é—®é¢˜ã€‚æ­¤ç±»é—®é¢˜ç»å¸¸å‡ºç°åœ¨å·¥ä¸šåˆ¶é€ ã€åŠ å·¥å’Œè®¾è®¡ä¸­ï¼Œé€šå¸¸ä½¿ç”¨äººå·¥æ‰‹åŠ¨é€‰æ‹©æƒé‡[32,33]ã€‚åœ¨ FLAPS ä¸­ï¼ŒOF å‚æ•°å¯ä»¥æ ¹æ®ä»»ä½•æ‰€éœ€çš„æ–¹æ¡ˆä»¥è‡ªæˆ‘æ”¹è¿›çš„æ–¹å¼å­¦ä¹ ï¼Œä¾‹å¦‚ï¼Œæ ‡å‡†åŒ–ã€é‡æ–°ç¼©æ”¾ã€å‡å€¼å½’ä¸€åŒ–æˆ–åœ¨ä¸Šè¿°æ­¥éª¤ä¹‹ä¸€ä¹‹åçš„ç›¸å¯¹åŠ æƒã€‚FLAPS å±•ç¤ºäº†å¦‚ä½•æˆåŠŸåˆ©ç”¨è®¡ç®—æ™ºèƒ½æ¦‚å¿µè§£å†³ç”Ÿå‘½ç§‘å­¦å‰æ²¿çš„å®é™…ä¼˜åŒ–é—®é¢˜ã€‚</blockquote><h2 id="4284660b-3655-4d02-8e55-47952d30926e" class="">Methods</h2><p id="655096c5-54b8-43ee-bb10-d241ce0bff64" class="">SAXS-guided protein simulations. SAXS-guided protein simulations were set up as described in ref. [7] and run with the MD engine GROMACS 5, including the scattering-guided MD extension[12].  Simulation input files are available on GitHub (<a href="http://github.com/FLAPS-NMI/FLAPS-sim_setups">github.com/FLAPS-NMI/FLAPS-sim_setups</a>). We used the popular CRYSOL software[48] in default mode to compute artificial SAXS intensities from known initial and target structures. Each intensity contained 700 equidistant data points up to a momentum transfer of <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>q</mi><mo>=</mo><mn>0.35</mn><mtext>â„«âˆ’</mtext></mrow><annotation encoding="application/x-tex">q = 0.35 â„«âˆ’</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">q</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.72777em;vertical-align:-0.08333em;"></span><span class="mord">0.35â„«âˆ’</span></span></span></span></span><span>ï»¿</span></span>1. During a simulation, the SAXS intensities of simulated structures are calculated internally in GROMACS[7,12] using the Debye equation[50]with amino-acid scattering factors corrected for displaced solvent[51]. The CRYSOL intensities were rescaled such that the extrapolation of the forward scattering matches the internally calculated GROMACS value.
Uncertainties were modelled as described in ref. [49]. Target SAXS data were calculated as the difference of rescaled CRYSOL intensities, including uncertainties, of known initial and target structures. The initial structureâ€™s rescaled CRYSOL intensity, including uncertainties, served as the absolute reference scattering in the simulations[7]. We included 17 data points selected as the difference curveâ€™s local extrema and interjacent points centred between each two extrema. For both
proteins, this corresponds well to the number of independent Shannon channels giving the number of independent data points in a SAXS curve[10].</p><blockquote id="e35ce56f-06c1-477a-93a9-71bb75ab2c2a" class="">SAXS å¼•å¯¼çš„è›‹ç™½è´¨æ¨¡æ‹Ÿã€‚å¦‚å‚è€ƒæ–‡çŒ®[7]ä¸­æ‰€è¿°è®¾ç½® SAXS å¼•å¯¼çš„è›‹ç™½è´¨æ¨¡æ‹Ÿã€‚ä¸ MD å¼•æ“ GROMACS 5 ä¸€èµ·è¿è¡Œï¼ŒåŒ…æ‹¬æ•£å°„å¼•å¯¼çš„ MD æ‰©å±•[12]ã€‚æ¨¡æ‹Ÿè¾“å…¥æ–‡ä»¶å¯åœ¨ GitHub (<a href="http://github.com/FLAPS-NMI/FLAPS-sim_setups">github.com/FLAPS-NMI/FLAPS-sim_setups</a>) ä¸Šè·å¾—ã€‚æˆ‘ä»¬åœ¨é»˜è®¤æ¨¡å¼ä¸‹ä½¿ç”¨æµè¡Œçš„ CRYSOL è½¯ä»¶ [48] ä»å·²çŸ¥çš„åˆå§‹ç»“æ„å’Œç›®æ ‡ç»“æ„è®¡ç®—äººå·¥ SAXS å¼ºåº¦ã€‚æ¯ä¸ªå¼ºåº¦åŒ…å« 700 ä¸ªç­‰è·æ•°æ®ç‚¹ï¼Œç›´åˆ°åŠ¨é‡è½¬ç§» <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>q</mi><mo>=</mo><mn>0.35</mn><msup><mover accent="true"><mi>A</mi><mo>Ëš</mo></mover><mrow><mo>âˆ’</mo><mn>1</mn></mrow></msup></mrow><annotation encoding="application/x-tex">q = 0.35 Ã…^{-1}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">q</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.9467699999999999em;vertical-align:0em;"></span><span class="mord">0.35</span><span class="mord"><span class="mord accent"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.9467699999999999em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="mord mathnormal">A</span></span><span style="top:-3.25233em;"><span class="pstrut" style="height:3em;"></span><span class="accent-body" style="left:-0.23611em;"><span class="mord">Ëš</span></span></span></span></span></span></span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">âˆ’</span><span class="mord mtight">1</span></span></span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>ã€‚åœ¨æ¨¡æ‹Ÿè¿‡ç¨‹ä¸­ï¼Œæ¨¡æ‹Ÿç»“æ„çš„SAXSå¼ºåº¦åœ¨GROMACS[7,12]å†…éƒ¨è®¡ç®—ä½¿ç”¨Debyeæ–¹ç¨‹[50]ï¼Œå¹¶å¯¹ç½®æ¢æº¶å‰‚çš„æ°¨åŸºé…¸æ•£å°„ç³»æ•°è¿›è¡Œæ ¡æ­£[51]ã€‚CRYSOLçš„å¼ºåº¦è¢«é‡æ–°è°ƒæ•´ï¼Œä½¿æ­£å‘æ•£å°„çš„å¤–æ¨å€¼ä¸GROMACSçš„å†…éƒ¨è®¡ç®—å€¼ä¸€è‡´ã€‚ä¸ç¡®å®šå› ç´ çš„å»ºæ¨¡å¦‚å‚è€ƒæ–‡çŒ®[49]æ‰€è¿°ã€‚ç›®æ ‡ SAXS æ•°æ®è®¡ç®—æ˜¯å·²çŸ¥åˆå§‹ç»“æ„å’Œç›®æ ‡ç»“æ„çš„é‡æ–°è°ƒæ•´ CRYSOL å¼ºåº¦ï¼ˆåŒ…æ‹¬ä¸ç¡®å®šæ€§ï¼‰çš„å·®å¼‚ã€‚åˆå§‹ç»“æ„é‡æ–°è°ƒæ•´åçš„ CRYSOL å¼ºåº¦ï¼ˆåŒ…æ‹¬ä¸ç¡®å®šæ€§ï¼‰åœ¨æ¨¡æ‹Ÿä¸­ç”¨ä½œç»å¯¹å‚è€ƒæ•£å°„[7]ã€‚æˆ‘ä»¬å°†17 ä¸ªæ•°æ®ç‚¹åŒ…æ‹¬é€‰æ‹©ä½œä¸ºå·®å¼‚æ›²çº¿çš„å±€éƒ¨æå€¼å’Œä»¥æ¯ä¸¤ä¸ªæå€¼ä¸ºä¸­å¿ƒçš„ä¸­é—´ç‚¹ã€‚å¯¹äºè¿™ä¸¤ç§è›‹ç™½è´¨ï¼Œè¿™å¾ˆå¥½åœ°å¯¹åº”äºç‹¬ç«‹é¦™å†œé€šé“çš„æ•°é‡ï¼Œç»™å‡ºäº† SAXS æ›²çº¿ä¸­ç‹¬ç«‹æ•°æ®ç‚¹çš„æ•°é‡[10]ã€‚</blockquote><p id="d7eca8c7-9d00-4fe7-9e5a-389d341d0386" class="">Root-mean-square deviation. In addition to the GDT analysis presented in the main text, we considered the more common root-mean-square deviation (RMSD) as a structural similarity measure. The results are presented in Supplementary Tables 1â€“4.</p><blockquote id="9353ee58-a4ec-4961-a74c-dd0dd281001e" class="">å‡æ–¹æ ¹åå·®ã€‚é™¤äº†æ­£æ–‡ä¸­ä»‹ç»çš„ GDT åˆ†æä¹‹å¤–ï¼Œæˆ‘ä»¬è¿˜è€ƒè™‘å°†æ›´å¸¸è§çš„å‡æ–¹æ ¹åå·® (RMSD) ä½œä¸ºç»“æ„ç›¸ä¼¼æ€§åº¦é‡ã€‚ç»“æœåˆ—äºè¡¥å……è¡¨ 1-4 ä¸­ã€‚ </blockquote><p id="cf0ad769-5269-4dfb-85e1-f14488f85b9f" class="">
</p><figure id="514bf761-5a38-4231-adce-461c4ac04e36" class="image"><a href="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%209.png"><img style="width:324px" src="%E5%8A%A8%E6%80%81%E7%B2%92%E5%AD%90%E7%BE%A4%E4%BC%98%E5%8C%96%208a0c985a5a914a269ffa8a06f5d751c9/Untitled%209.png"/></a></figure><p id="31d582c2-b0f7-46e0-bb8d-171909802047" class=""><mark class="highlight-gray">Eq.(4)</mark></p><p id="b7e3e759-d4ff-438f-b5cb-e55fb1834b6e" class="">                              </p><p id="8b513387-bd1b-4268-96ce-322f7df7de82" class="">The RMSD is the minimal mass-weighted average distance between N atoms (usually backbone or alpha carbon) of two superimposed structures over all possible spatial translations and rotations,</p><blockquote id="5731b174-8cdd-4934-82e6-335765711d4d" class="">RMSD æ˜¯åœ¨æ‰€æœ‰å¯èƒ½çš„ç©ºé—´å¹³ç§»å’Œæ—‹è½¬ä¸­ï¼Œä¸¤ä¸ªå åŠ ç»“æ„çš„ N åŸå­ï¼ˆé€šå¸¸æ˜¯éª¨æ¶æˆ– Î± ç¢³ï¼‰ä¹‹é—´çš„æœ€å°è´¨é‡åŠ æƒå¹³å‡è·ç¦»ã€‚</blockquote><p id="9583855f-251c-45c4-81c6-25cab0894f0b" class="">where M = Î£Ni=1 mi and mi is the mass of atom i. ri and ri,0 are the positions of atom i in the mobile and reference structure, respectively. Holo/apo LAO protein and open/closed ADK have alpha-carbon RMSD values of 4.7 â„« and 7.1 â„«, respectively.</p><blockquote id="e5300bb2-9b58-4318-b7ce-eab1d98055af" class="">å…¶ä¸­ï¼Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>=</mo><mrow><msubsup><mo>âˆ‘</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>N</mi></msubsup><msub><mi>m</mi><mi>i</mi></msub></mrow></mrow><annotation encoding="application/x-tex">m={\sum_{i=1}^Nm_{i}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathnormal">m</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:1.2809409999999999em;vertical-align:-0.29971000000000003em;"></span><span class="mord"><span class="mop"><span class="mop op-symbol small-op" style="position:relative;top:-0.0000050000000000050004em;">âˆ‘</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.981231em;"><span style="top:-2.40029em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span><span class="mrel mtight">=</span><span class="mord mtight">1</span></span></span></span><span style="top:-3.2029em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.10903em;">N</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.29971000000000003em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord"><span class="mord mathnormal">m</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span> å’Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>m</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">m_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.58056em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal">m</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>æ˜¯åŸå­<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi></mrow><annotation encoding="application/x-tex">i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.65952em;vertical-align:0em;"></span><span class="mord mathnormal">i</span></span></span></span></span><span>ï»¿</span></span>çš„è´¨é‡ã€‚<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>r</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex"> r_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.58056em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.02778em;">r</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.31166399999999994em;"><span style="top:-2.5500000000000003em;margin-left:-0.02778em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>å’Œ<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>r</mi><mrow><mi>i</mi><mo separator="true">,</mo><mn>0</mn></mrow></msub></mrow><annotation encoding="application/x-tex">r_{i,0}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.716668em;vertical-align:-0.286108em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.02778em;">r</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.311664em;"><span style="top:-2.5500000000000003em;margin-left:-0.02778em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span><span class="mpunct mtight">,</span><span class="mord mtight">0</span></span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em;"><span></span></span></span></span></span></span></span></span></span></span><span>ï»¿</span></span>åˆ†åˆ«æ˜¯åŸå­<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi></mrow><annotation encoding="application/x-tex">i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.65952em;vertical-align:0em;"></span><span class="mord mathnormal">i</span></span></span></span></span><span>ï»¿</span></span>åœ¨ç§»åŠ¨ç»“æ„å’Œå‚è€ƒç»“æ„ä¸­çš„ä½ç½®ã€‚ Holo/apo LAOè›‹ç™½å’Œå¼€æ”¾/å°é—­ADKçš„Î±-ç¢³RMSDå€¼åˆ†åˆ«ä¸º 4.7 â„« å’Œ 7.1 â„«ã€‚</blockquote><p id="cf8e504a-ea66-45e9-83e4-6c5f90f85ab4" class="">A disadvantage is that RMSD correlates strongly with the largest displacement between two structures, and small numbers of displaced atoms induce large changes. We use GDT[40,52,53]
as the main target metric as it more accurately accounts for local misalignments.</p><blockquote id="3d480a89-d41c-4f7d-ac25-605ccc276cce" class="">ç¼ºç‚¹æ˜¯RMSDä¸ä¸¤ä¸ªç»“æ„ä¹‹é—´çš„æœ€å¤§ä½ç§»å¯†åˆ‡ç›¸å…³ï¼Œå°‘é‡çš„ä½ç§»åŸå­ä¼šå¼•èµ·å¾ˆå¤§çš„å˜åŒ–ã€‚ æˆ‘ä»¬ä½¿ç”¨GDT[40,52,53]ä½œä¸ºä¸»è¦çš„ç›®æ ‡æŒ‡æ ‡ï¼Œå› ä¸ºå®ƒæ›´å‡†ç¡®åœ°è¯´æ˜äº†å±€éƒ¨é”™ä½ã€‚</blockquote><p id="a5c0d707-29c5-4ae4-b4ce-02116f6df851" class="">Implementation. FLAPS is implemented as a stand-alone solver in Hyppopy, a Python-based hyperparameter optimization package available at <a href="https://github/">https://github</a>.com/MIC-DKFZ/Hyppopy. Hyppopy provides tools for blackbox optimization. It has a simple, unified application programming interface (API) that can be used to access a collection of solver libraries. Our implementation of FLAPS is available on GitHub[54,55]. We implemented a Message-Passing-Interface (MPI)-parallel
version of the code using a sophisticated parallelization architecture as described in Supplementary Fig. 13. Available compute nodes comprising a given number of processors are divided into blocks, each of which corresponds to one particle in the swarm. Within one block, the simulation itself runs on a single core, while all the other cores process the generated frames in the trajectory on the fly. This results in a massive reduction in runtime.</p><blockquote id="5844ee1b-5472-4d47-b71c-fcc647fb398f" class="">æ‰§è¡Œã€‚ FLAPS åœ¨ Hyppopy ä¸­ä½œä¸ºç‹¬ç«‹æ±‚è§£å™¨å®ç°ï¼ŒHyppopy æ˜¯ä¸€ä¸ªåŸºäº Python çš„è¶…å‚æ•°ä¼˜åŒ–åŒ…ï¼Œå¯åœ¨ <a href="https://github/">https://github</a>.com/MIC-DKFZ/Hyppopyè·å¾—ã€‚Hyppopy æä¾›äº†é»‘ç›’ä¼˜åŒ–å·¥å…·ã€‚å®ƒæœ‰ä¸€ä¸ªç®€å•ã€ç»Ÿä¸€çš„åº”ç”¨ç¨‹åºç¼–ç¨‹æ¥å£ (API)ï¼Œå¯ç”¨äºè®¿é—®ä¸€ç»„æ±‚è§£å™¨åº“ã€‚æˆ‘ä»¬çš„ FLAPS å®ç°å¯åœ¨ GitHub[54,55] ä¸Šæ‰¾åˆ°ã€‚å¦‚è¡¥å……å›¾ 13 æ‰€è¿°ï¼Œæˆ‘ä»¬ä½¿ç”¨å¤æ‚çš„å¹¶è¡ŒåŒ–æ¶æ„å®ç°äº†ä»£ç çš„æ¶ˆæ¯ä¼ é€’æ¥å£ï¼ˆMPIï¼‰å¹¶è¡Œç‰ˆæœ¬ã€‚åŒ…å«ç»™å®šæ•°é‡å¤„ç†å™¨çš„å¯ç”¨è®¡ç®—èŠ‚ç‚¹è¢«åˆ’åˆ†ä¸ºå—ï¼Œæ¯ä¸ªå—å¯¹åº”ä¸€ä¸ªç¾¤ä½“ä¸­çš„ç²’å­ã€‚åœ¨ä¸€ä¸ªå—å†…ï¼Œæ¨¡æ‹Ÿæœ¬èº«åœ¨å•ä¸ªæ ¸ä¸Šè¿è¡Œï¼Œè€Œæ‰€æœ‰å…¶å®ƒæ ¸åœ¨è¿è¡Œä¸­å¤„ç†è½¨è¿¹ä¸­ç”Ÿæˆçš„å¸§ã€‚è¿™å¯¼è‡´è¿è¡Œæ—¶é—´å¤§å¹…å‡å°‘ã€‚</blockquote><p id="946afb80-caa2-4678-ae13-9afe6b2f83ef" class="">The experiments were run on the ForHLR II cluster system located at the Steinbuch Centre for Computing at Karlsruhe Institute of Technology. The system comprises 1,152 thin, that is, solely central processing unit (CPU)-based, compute nodes. Each node is equipped with two 10-core Intel Xeon E5-2660 v3 Haswell CPUs at 3.3 GHz, 64 GB of DDR3 main memory and 4x Mellanox 100-Gbit EDR InfiniBand links. The software packages used were a RHEL Linux with kernel version 4.18.0 and Python 3.6.8.</p><blockquote id="37d92224-9c97-4d36-b84d-fbd261b6ce6d" class="">å®éªŒåœ¨ä½äºå¡å°”æ–¯é²å„ç†å·¥å­¦é™¢ Steinbuch è®¡ç®—ä¸­å¿ƒçš„ ForHLR II é›†ç¾¤ç³»ç»Ÿä¸Šè¿è¡Œã€‚è¯¥ç³»ç»ŸåŒ…æ‹¬ 1,152 ä¸ªç˜¦è®¡ç®—èŠ‚ç‚¹ï¼Œå³ä»…åŸºäºä¸­å¤®å¤„ç†å•å…ƒ (CPU) çš„è®¡ç®—èŠ‚ç‚¹ã€‚æ¯ä¸ªèŠ‚ç‚¹é…å¤‡ä¸¤ä¸ª 3.3 GHz çš„ 10 æ ¸ Intel Xeon E5-2660 v3 Haswell CPUã€64 GB DDR3 ä¸»å†…å­˜å’Œ 4 ä¸ª Mellanox 100-Gbit EDR InfiniBand é“¾è·¯ã€‚ä½¿ç”¨çš„è½¯ä»¶åŒ…æ˜¯ RHEL Linuxçš„å†…æ ¸ç‰ˆæœ¬ 4.18.0 å’Œ Python 3.6.8 ã€‚</blockquote><p id="bf9316af-2f8d-4902-aa44-1ce2017ee458" class="">Each run used 51 compute nodes (1,020 cores in total). Owing to the magnitude of metadata and I/O operations, we used a private on-demand file system (BeeGFS On-Demand) with a stripe count of 1, where one node was reserved for the metadata server[56]. Each block in the underlying simulator-worker scheme consisted of five nodes, that is, 100 cores (one simulator, 99 workers). Each run cost ~40,000 CPU hours. For the presented application, we used cognitive acceleration coefficient<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï•</mi><mn>1</mn></msub><mo>=</mo><mn>2.0</mn></mrow><annotation encoding="application/x-tex"> Ï•_1= 2.0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï•</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.30110799999999993em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">1</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">2.0</span></span></span></span></span><span>ï»¿</span></span> and social acceleration coefficient<style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï•</mi><mn>2</mn></msub><mo>=</mo><mn>1.5</mn></mrow><annotation encoding="application/x-tex"> Ï•_2= 1.5</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï•</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.30110799999999993em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">1.5</span></span></span></span></span><span>ï»¿</span></span> in the particle update (Algorithm 1). The complete set-up, including all PSO hyperparameters used, is available on GitHub[57].</p><blockquote id="85b6c910-5109-443a-97c9-18c1b7e21f87" class="">æ¯æ¬¡è¿è¡Œä½¿ç”¨ 51 ä¸ªè®¡ç®—èŠ‚ç‚¹ï¼ˆæ€»å…± 1,020 ä¸ªå†…æ ¸ï¼‰ã€‚ç”±äºå…ƒæ•°æ®å’Œ I/O æ“ä½œçš„è§„æ¨¡ï¼Œæˆ‘ä»¬ä½¿ç”¨äº†æ¡å¸¦è®¡æ•°ä¸º 1 çš„ç§æœ‰æŒ‰éœ€æ–‡ä»¶ç³»ç»Ÿ (BeeGFS On-Demand)ï¼Œå…¶ä¸­ä¸ºå…ƒæ•°æ®æœåŠ¡å™¨ä¿ç•™äº†ä¸€ä¸ªèŠ‚ç‚¹[56]ã€‚åº•å±‚simulator-worker æ–¹æ¡ˆä¸­çš„æ¯ä¸ªblock ç”±5 ä¸ªèŠ‚ç‚¹ç»„æˆï¼Œå³100 ä¸ªæ ¸å¿ƒï¼ˆä¸€ä¸ªæ¨¡æ‹Ÿå™¨ï¼Œ99 ä¸ªworkerï¼‰ã€‚æ¯æ¬¡è¿è¡ŒèŠ±è´¹çº¦ 40,000 ä¸ª CPU å°æ—¶ã€‚å¯¹äºæ‰€æå‡ºçš„åº”ç”¨ç¨‹åºï¼Œæˆ‘ä»¬åœ¨ç²’å­æ›´æ–°ï¼ˆç®—æ³• 1ï¼‰ä¸­ä½¿ç”¨äº†è®¤çŸ¥åŠ é€Ÿç³»æ•° <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï•</mi><mn>1</mn></msub><mo>=</mo><mn>2.0</mn></mrow><annotation encoding="application/x-tex"> Ï•_1= 2.0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï•</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.30110799999999993em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">1</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">2.0</span></span></span></span></span><span>ï»¿</span></span> å’Œç¤¾ä¼šåŠ é€Ÿç³»æ•° <style>@import url('https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.2/katex.min.css')</style><span data-token-index="0" contenteditable="false" class="notion-text-equation-token" style="user-select:all;-webkit-user-select:all;-moz-user-select:all"><span></span><span><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>Ï•</mi><mn>2</mn></msub><mo>=</mo><mn>1.5</mn></mrow><annotation encoding="application/x-tex"> Ï•_2= 1.5</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord mathnormal">Ï•</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.30110799999999993em;"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span><span class="vlist-s">â€‹</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">1.5</span></span></span></span></span><span>ï»¿</span></span>ã€‚å®Œæ•´çš„è®¾ç½®ï¼ŒåŒ…æ‹¬ä½¿ç”¨çš„æ‰€æœ‰ PSO è¶…å‚æ•°ï¼Œå¯åœ¨ GitHub[57] ä¸Šæ‰¾åˆ°ã€‚</blockquote><h2 id="0975005c-8c36-4471-bf4a-1620c3cb7394" class="">Data availability</h2><p id="b6ee4a79-adef-46f8-8fb1-976808f790e2" class="">The software for the metaheuristic molecular dynamics parameter optimization of SAXS-guided structure-based protein simulations used in this work54,55 is publicly available at <a href="https://github.com/FLAPS-NMI?tab=repositories">https://github.com/FLAPS-NMI?tab=repositories</a>. A minimal dataset to reproduce the presented results[57] is publicly available at <a href="https://github/">https://github</a>.com/FLAPS-NMI/FLAPS sim_setups/releases/tag/v1.0 and published under the Creative Commons Attribution 4.0 International Public License.</p><blockquote id="76d7fcfc-8967-4c4d-b8c0-9ebfd4414600" class="">æœ¬å·¥ä½œä¸­ä½¿ç”¨çš„ SAXS å¼•å¯¼çš„åŸºäºç»“æ„çš„è›‹ç™½è´¨æ¨¡æ‹Ÿçš„å…ƒå¯å‘å¼åˆ†å­åŠ¨åŠ›å­¦å‚æ•°ä¼˜åŒ–è½¯ä»¶[ 54,55] å¯åœ¨ <a href="https://github.com/FLAPS-NMI?tab=repositories">https://github.com/FLAPS-NMI?tab=repositories</a> ä¸Šè·å¾—ã€‚é‡ç°æ‰€å‘ˆç°ç»“æœçš„æœ€å°æ•°æ®é›†[57] å¯åœ¨ <a href="https://github/">https://github</a> com/FLAPS-NMI/FLAPS-sim_setups/releases/tag/v1.0 ä¸Šå…¬å¼€è·å¾—ã€‚å¹¶æ ¹æ®çŸ¥è¯†å…±äº«ç½²å 4.0 å›½é™…å…¬å…±è®¸å¯è¯å‘å¸ƒã€‚</blockquote><h2 id="abc336ef-beff-471d-9560-346a993a582b" class="">Code availability</h2><p id="ee11ab2d-dd69-4cbe-b6b4-25098f1ff554" class="">All code used in this work is publicly available at <a href="https://github.com/">https://github.com/</a>FLAPS-NMI?tab=repositories and published under the New BSD Licence. The MPI-parallelized FLAPS solver is implemented in Optunity, a hyperparameter tuning package for Python. Our extended version of Optunity55
is available at <a href="https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0">https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0</a> and integrated into Hyppopy, a Python-based toolbox for blackbox optimization. Our extended version of Hyppopy[54]
is available at <a href="https://github.com/FLAPS-NMI/">https://github.com/FLAPS-NMI/</a>FLAPS-Hyppopy/releases/tag/v1.0.</p><blockquote id="ef54cd68-12e4-413e-af0b-e7843c4dfd0d" class="">è¿™é¡¹å·¥ä½œä¸­ä½¿ç”¨çš„æ‰€æœ‰ä»£ç éƒ½å¯ä»¥åœ¨ <a href="https://github.com/FLAPS-NMI?tab=repositories">https://github.com/FLAPS-NMI?tab=repositories</a> ä¸Šå…¬å¼€è·å¾—ï¼Œå¹¶åœ¨æ–° BSD è®¸å¯è¯ä¸‹å‘å¸ƒã€‚ MPI å¹¶è¡ŒåŒ– FLAPS æ±‚è§£å™¨åœ¨ Optunity ä¸­å®ç°ï¼ŒOptunity æ˜¯ Python çš„è¶…å‚æ•°è°ƒæ•´åŒ…ã€‚æˆ‘ä»¬çš„ Optunity[55] æ‰©å±•ç‰ˆæœ¬å¯åœ¨ <a href="https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0">https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0</a> è·å¾—ï¼Œå¹¶é›†æˆåˆ°åŸºäº Python çš„é»‘ç›’ä¼˜åŒ–å·¥å…·ç®± Hyppopy ä¸­ã€‚æˆ‘ä»¬çš„ Hyppopy[54] æ‰©å±•ç‰ˆå¯åœ¨ <a href="https://github.com/FLAPS-NMI/FLAPS-Hyppopy/releases/tag/v1.0">https://github.com/FLAPS-NMI/FLAPS Hyppopy/releases/tag/v1.0</a> è·å¾—ã€‚</blockquote><h2 id="6563f67b-19bb-4464-a444-6800eaf7e045" class="">References</h2><ol type="1" id="eb6535e4-6a26-4fbf-9a77-16795bc5f4ab" class="numbered-list" start="1"><li>Selkoe, D. Folding proteins in fatal ways. Nature 426, 900â€“904 (2003).</li></ol><ol type="1" id="60417bc1-69f2-431e-9c0f-b3086b920717" class="numbered-list" start="2"><li>Selkoe, D. J. Cell biology of protein misfolding: the examples of Alzheimerâ€™s and Parkinsonâ€™s diseases. Nat. Cell Biol. 6, 1054â€“1061 (2004).</li></ol><ol type="1" id="37e725b7-d4a2-43bc-91b8-07cca2184503" class="numbered-list" start="3"><li>Soto, C. Unfolding the role of protein misfolding in neurodegenerative diseases. Nat. Rev. Neurosci. 4, 49â€“60 (2003).</li></ol><ol type="1" id="46205c3d-6fc0-4c9f-ae44-ea9e9187f139" class="numbered-list" start="4"><li>Mukherjee, A., Morales-Scheihing, D., Butler, P. C. &amp; Soto, C. Type 2 diabetes as a protein misfolding disease. Trends Mol. Med. 21, 439â€“449 (2015).</li></ol><ol type="1" id="6375c8b3-82c9-4f5d-a656-3f8dc45d72e0" class="numbered-list" start="5"><li>Dobson, C. M. Protein-misfolding diseases: getting out of shape. Nature 418,729â€“730 (2002).</li></ol><ol type="1" id="9c7ca279-2276-47d9-b5c1-f4ecdd393511" class="numbered-list" start="6"><li>Karaca, E., Rodrigues, J. P., Graziadei, A., Bonvin, A. M. &amp; Carlomagno, T.M3: an integrative framework for structure determination of molecular machines. Nat. Methods 14, 897â€“902 (2017).</li></ol><ol type="1" id="9f2e1fd8-78bd-4958-90fe-5f3a33885766" class="numbered-list" start="7"><li>Weiel, M., Reinartz, I. &amp; Schug, A. Rapid interpretation of small-angle X-ray scattering data. PLoS Comput. Biol. 15, e1006900 (2019).</li></ol><ol type="1" id="ae811d77-ca67-461b-b877-8c62dd835887" class="numbered-list" start="8"><li>Hermann, M. R. &amp; Hub, J. S. SAXS-restrained ensemble simulations of intrinsically disordered proteins with commitment to the principle of maximum entropy. J. Chem. Theory Comput. 15, 5103â€“5115 (2019).</li></ol><ol type="1" id="b0be9ab0-a094-4d84-b678-ef378bfef9c0" class="numbered-list" start="9"><li>Chen, P. et al. Combined small-angle X-ray and neutron scattering restraints in molecular dynamics simulations. J. Chem. Theory Comput. 15, 4687â€“4698 (2019).</li></ol><ol type="1" id="e9565981-4f2c-485d-b803-d886d1bb4e73" class="numbered-list" start="10"><li>Chen, P. &amp; Hub, J. S. Interpretation of solution X-ray scattering by explicit-solvent molecular dynamics. Biophys. J. 108, 2573â€“2584(2015).</li></ol><ol type="1" id="d52d066a-ad25-453c-85ac-eecc61be8f51" class="numbered-list" start="11"><li>Whitford, P. C. et al. Excited states of ribosome translocation revealed through integrative molecular modeling. Proc. Natl Acad. Sci. USA 108,18943â€“18948 (2011).</li></ol><ol type="1" id="49f710c1-5ae2-4d2b-b6f8-0e67299a60f1" class="numbered-list" start="12"><li>BjÃ¶rling, A., Niebling, S., Marcellini, M., van der Spoel, D. &amp; Westenhoff, S. Deciphering solution scattering data with experimentally guided molecular dynamics simulations. J. Chem. Theory Comput. 11, 780â€“787 (2015).</li></ol><ol type="1" id="d82dfaa0-b23a-42ed-b212-050345519e14" class="numbered-list" start="13"><li>Shevchuk, R. &amp; Hub, J. S. Bayesian refinement of protein structures and ensembles against SAXS data using molecular dynamics. PLoS Comput. Biol.13, e1005800 (2017).</li></ol><ol type="1" id="c3425827-d499-445d-b3ab-7b438e8ca716" class="numbered-list" start="14"><li>Hummer, G. &amp; KÃ¶finger, J. Bayesian ensemble refinement by replica simulations and reweighting. J. Chem. Phys. 143, 243150 (2015).</li></ol><ol type="1" id="3d857413-3609-40a0-ae6f-c6a07cae61f4" class="numbered-list" start="15"><li>Hub, J. S. Interpreting solution X-ray scattering data using molecular simulations. Curr. Opin. Struct. Biol. 49, 18â€“26 (2018).</li></ol><ol type="1" id="adf9c6f8-5328-465e-81d9-520c2f9e3110" class="numbered-list" start="16"><li>Kennedy, J. &amp; Eberhart, R. Particle swarm optimization. In Proc. ICNNâ€™95â€”International Conference on Neural Networks Vol. 4, 1942â€“1948 (IEEE, 1995);
<a href="https://doi.org/10.1109/ICNN.1995.488968">https://doi.org/10.1109/ICNN.1995.488968</a></li></ol><ol type="1" id="00b899b4-fb7c-443b-b344-f15f8056f7ac" class="numbered-list" start="17"><li>Kennedy, J. The particle swarm: social adaptation of knowledge. In Proc. 1997 IEEE International Conference on Evolutionary Computation (ICECâ€™97) 303â€“308 (IEEE, 1997); <a href="https://doi.org/10.1109/ICEC.1997.592326">https://doi.org/10.1109/ICEC.1997.592326</a></li></ol><ol type="1" id="f3c1ba3e-e4fe-4779-98e9-026120faf913" class="numbered-list" start="18"><li>Shi, Y. &amp; Eberhart, R. A modified particle swarm optimizer. In Proc. 1998 IEEE International Conference on Evolutionary Computation. IEEE World Congress on Computational Intelligence (cat. no. 98TH8360) 69â€“73 (IEEE,1998); <a href="https://doi.org/10.1109/ICEC.1998.699146">https://doi.org/10.1109/ICEC.1998.699146</a></li></ol><ol type="1" id="a881887f-1638-4cb8-b453-5d9bdd62cccd" class="numbered-list" start="19"><li>Clerc, M. &amp; Kennedy, J. The particle swarmâ€”explosion, stability and convergence in a multidimensional complex space. IEEE Trans. Evol. Comput.6, 58â€“73 (2002).</li></ol><ol type="1" id="15b0f19a-1651-4b3c-b51e-b58a43c9231e" class="numbered-list" start="20"><li>Parsopoulos, K. E. &amp; Vrahatis, M. N. Recent approaches to global optimization problems through particle swarm optimization. Nat. Comput. 1,235â€“306 (2002).</li></ol><ol type="1" id="c10f21ad-61de-4bcb-a20a-e83929153e37" class="numbered-list" start="21"><li>Jordehi, A. R. Particle swarm optimisation for dynamic optimisation problems: a review. Neural Comput. Appl. 25, 1507â€“1516 (2014).</li></ol><ol type="1" id="deb6f2bf-3b00-48e1-aa91-890cf74ee118" class="numbered-list" start="22"><li>Blackwell, T. in Evolutionary Computation in Dynamic and Uncertain Environments (eds Yang, S. et al.) 29â€“49 (Springer, 2007); <a href="https://doi/">https://doi</a>.org/10.1007/978-3-540-49774-5_2</li></ol><ol type="1" id="fee9faa2-94eb-4731-9150-38e1b19933ea" class="numbered-list" start="23"><li>Stanley, K. O., Clune, J., Lehman, J. &amp; Miikkulainen, R. Designing neural networks through neuroevolution. Nat. Mach. Intell. 1, 24â€“35 (2019).</li></ol><ol type="1" id="17fed1f4-3e67-4b36-b9b6-6d145a873e60" class="numbered-list" start="24"><li>Taherkhani, M. &amp; Safabakhsh, R. A novel stability-based adaptive inertia weight for particle swarm optimization. Appl. Soft Comput. 38,281â€“295 (2016).</li></ol><ol type="1" id="11ea2433-ebc1-423a-a0f3-ab90b6c40800" class="numbered-list" start="25"><li>Eberhart, R. C. &amp; Shi, Y. Comparing inertia weights and constriction factors in particle swarm optimization. In Proc. 2000 Congress on Evolutionary Computation. CEC00 (cat. no. 00TH8512) Vol. 1, 84â€“88 (IEEE, 2000);<a href="https://doi.org/10.1109/CEC.2000.870279">https://doi.org/10.1109/CEC.2000.870279</a></li></ol><ol type="1" id="0d266271-c70a-4bec-b8e7-d622bb7aa7eb" class="numbered-list" start="26"><li>Pedersen, M. E. H. &amp; Chipperfield, A. J. Simplifying particle swarm optimization. Appl. Soft Comput. 10, 618â€“628 (2010).</li></ol><ol type="1" id="7cac1ba4-d695-4d2a-9666-22b51402131a" class="numbered-list" start="27"><li>Meissner, M., Schmuker, M. &amp; Schneider, G. Optimized particle swarm optimization (OPSO) and its application to artificial neural network training.BMC Bioinformatics 7, 125 (2006).</li></ol><ol type="1" id="2f6d5a57-d86d-47b9-a47f-87aa955b7448" class="numbered-list" start="28"><li>Nobile, M. S. et al. Fuzzy self-tuning PSO: a settings-free algorithm for global optimization. Swarm Evol. Comput. 39, 70â€“85 (2018).</li></ol><ol type="1" id="a74bb90f-6ad4-45f6-8c98-9ac1fdd619e1" class="numbered-list" start="29"><li>Poli, R. Analysis of the publications on the applications of particle swarm optimisation. J. Artif. Evol. Appl. 2008, 685175 (2008).</li></ol><ol type="1" id="8ceb6d77-f50a-4efa-96ef-d93365c86429" class="numbered-list" start="30"><li>Sengupta, S., Basak, S. &amp; Peters, R. A. Particle swarm optimization: a survey of historical and recent developments with hybridization perspectives. Mach.Learn. Knowledge Extraction 1, 157â€“191 (2019).</li></ol><ol type="1" id="0215636b-adc7-4e49-8d72-ff5342dc4d64" class="numbered-list" start="31"><li>Navalertporn, T. &amp; Afzulpurkar, N. V. Optimization of tile manufacturing process using particle swarm optimization. Swarm Evol. Comput. 1,97â€“109 (2011).</li></ol><ol type="1" id="6d35edb4-ca51-4b74-b89f-7f8da52c8914" class="numbered-list" start="32"><li>Pawar, P., Rao, R. &amp; Davim, J. Multiobjective optimization of grinding process parameters using particle swarm optimization algorithm. Mater. Manuf. Process. 25, 424â€“431 (2010).</li></ol><ol type="1" id="b4c49e42-1ff6-4f04-9e9e-6b253a273fc3" class="numbered-list" start="33"><li>Ma, C. &amp; Qu, L. Multiobjective optimization of switched reluctance motors based on design of experiments and particle swarm optimization. IEEE Trans. Energy Conver. 30, 1144â€“1153 (2015).</li></ol><ol type="1" id="882fed13-5dc4-4826-9431-e896f382df29" class="numbered-list" start="34"><li>Zhang, C., Chen, Z., Mei, Q. &amp; Duan, J. Application of particle swarm optimization combined with response surface methodology to transverse flux permanent magnet motor optimization. IEEE Trans. Magn. 53,8113107 (2017).</li></ol><ol type="1" id="d15b2d42-452d-4c82-9480-78b693af5c22" class="numbered-list" start="35"><li>Ioffe, S. &amp; Szegedy, C. Batch normalization: accelerating deep network training by reducing internal covariate shift. In Proc. 32nd International Conference on Machine Learning Vol. 37 (eds. Bach, F. &amp; Blei, D.) 448â€“456 (PMLR, 2015).</li></ol><ol type="1" id="c7989c8f-4ba8-4dd0-862d-bfef55d790b4" class="numbered-list" start="36"><li>Onuchic, J. N. &amp; Wolynes, P. G. Theory of protein folding. Curr. Opin. Struct. Biol. 14, 70â€“75 (2004).</li></ol><ol type="1" id="cc252d2c-0be1-49db-b9ca-cb5654c9ec41" class="numbered-list" start="37"><li>Schug, A. &amp; Onuchic, J. N. From protein folding to protein function and biomolecular binding by energy landscape theory. Curr. Opin. Pharmacol. 10,709â€“714 (2010).</li></ol><ol type="1" id="ad25bf71-9194-4b36-9a4e-12c2d04811a3" class="numbered-list" start="38"><li>Alford, R. F. et al. The Rosetta all-atom energy function for macromolecular modeling and design. J. Chem. Theory Comput. 13,3031â€“3048 (2017).</li></ol><ol type="1" id="1df558e5-f6ef-4b95-a93a-7634a95004bf" class="numbered-list" start="39"><li>Leman, J. K. et al. Macromolecular modeling and design in Rosetta: recent methods and frameworks. Nat. Methods 17, 665â€“680 (2020).</li></ol><ol type="1" id="4d355e7f-40e3-4ffd-ad18-9c714a54c753" class="numbered-list" start="40"><li>Zemla, A. LGA: a method for finding 3D similarities in protein structures. Nucleic Acids Res. 31, 3370â€“3374 (2003).</li></ol><ol type="1" id="1d9658a1-e1f0-4cec-b255-d4e773dfe605" class="numbered-list" start="41"><li>Moult, J., Fidelis, K., Kryshtafovych, A., Schwede, T. &amp; Tramontano, A.Critical assessment of methods of protein structure prediction (CASP)â€“round XII. Proteins 86, 7â€“15 (2018).</li></ol><ol type="1" id="f0e9873c-2d6f-4690-889c-4f1e370146d3" class="numbered-list" start="42"><li>Ames, G. F.-L. Bacterial periplasmic transport systems: structure, mechanism and evolution. Annu. Rev. Biochem. 55, 397â€“425 (1986).</li></ol><ol type="1" id="13741a20-a8c5-4fa3-abc4-a137e4885f19" class="numbered-list" start="43"><li>Berman, H. M. et al. The Protein Data Bank. Nucleic Acids Res. 28,235â€“242 (2000).</li></ol><ol type="1" id="830dcf4e-019e-4bdd-82cd-bcd984c66558" class="numbered-list" start="44"><li>Oh, B.-H. et al. Three-dimensional structures of the periplasmic lysine/arginine/ornithine-binding protein with and without a ligand. J. Biol. Chem.268, 11348â€“11355 (1993).</li></ol><ol type="1" id="05a179d3-407a-4454-bdac-8dfa1f956a1d" class="numbered-list" start="45"><li>MÃ¼ller, C., Schlauderer, G., Reinstein, J. &amp; Schulz, G. E. Adenylate kinase motions during catalysis: an energetic counterweight balancing substrate binding. Structure 4, 147â€“156 (1996).</li></ol><ol type="1" id="10f68c24-5384-4b52-bbe6-180b669d1374" class="numbered-list" start="46"><li>MÃ¼ller, C. W. &amp; Schulz, G. E. Structure of the complex between adenylate kinase from Escherichia coli and the inhibitor Ap5 A refined at 1.9-â„« resolution: a model for a catalytic transition state. J. Mol. Biol. 224,159â€“177 (1992).</li></ol><ol type="1" id="eb6fb9bd-46fa-4020-8e87-3e12aedc5dc9" class="numbered-list" start="47"><li>Whitford, P. C., Miyashita, O., Levy, Y. &amp; Onuchic, J. N. Conformational transitions of adenylate kinase: switching by cracking. J. Mol. Biol. 366, 1661â€“1671 (2007).</li></ol><ol type="1" id="e30cb62d-f558-4706-b724-77404d163f8b" class="numbered-list" start="48"><li>Svergun, D., Barberato, C. &amp; Koch, M. H. CRYSOLâ€”a program to evaluate X-ray solution scattering of biological macromolecules from atomic coordinates. J. Appl. Crystallogr. 28, 768â€“773 (1995).</li></ol><ol type="1" id="9525083a-4276-49da-80a4-13be24324644" class="numbered-list" start="49"><li>Sedlak, S. M., Bruetzel, L. K. &amp; Lipfert, J. Quantitative evaluation of statistical errors in small-angle X-ray scattering measurements. J. Appl. Crystallogr. 50, 621â€“630 (2017).</li></ol><ol type="1" id="960a1a6c-638d-4335-ad4e-62dce6905535" class="numbered-list" start="50"><li>Debye, P. Zerstreuung von RÃ¶ntgenstrahlen. Annal. Phys. 351,809â€“823 (1915).</li></ol><ol type="1" id="2a157ddb-49c1-4da1-946a-38d95d6a2cc0" class="numbered-list" start="51"><li>Yang, S., Park, S., Makowski, L. &amp; Roux, B. A rapid coarse residue-based computational method for X-ray solution scattering characterization of protein folds and multiple conformational states of large protein complexes.Biophys. J. 96, 4449â€“4463 (2009).</li></ol><ol type="1" id="676fff88-d1f2-48c6-a487-f7ac083530d3" class="numbered-list" start="52"><li>Kufareva, I. &amp; Abagyan, R. in Homology Modeling (eds Orry, A. J. W. &amp; Abagyan, R.) 231â€“257 (Springer, 2011).</li></ol><ol type="1" id="4ec26413-2f59-43b6-acab-f334a1db9dfb" class="numbered-list" start="53"><li>Modi, V., Xu, Q., Adhikari, S. &amp; Dunbrack Jr, R. L. Assessment of template-based modeling of protein structure in CASP11. Proteins 84,200â€“220 (2016).</li></ol><ol type="1" id="4037dad5-73b5-4999-bfbf-861c47971d0d" class="numbered-list" start="54"><li>Weiel, M. et al. FLAPS Optunity code repository (FLAPS-NMI@Github/
FLAPS-optunity, 2021); <a href="https://doi.org/10.5281/zenodo.4773992">https://doi.org/10.5281/zenodo.4773992</a>,<a href="https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0">https://github.com/FLAPS-NMI/FLAPS-optunity/releases/tag/v1.0</a></li></ol><ol type="1" id="2d967381-00bb-4023-bda5-602152897d42" class="numbered-list" start="55"><li>Soysal, M. et al. Using on-demand file systems in HPC environments. In 2019 International Conference on High Performance Computing &amp; Simulation (HPCS) 390â€“398 (IEEE, 2019); <a href="https://doi.org/10.1109/">https://doi.org/10.1109/</a>HPCS48598.2019.9188216</li></ol><ol type="1" id="a21fb0be-45da-43cd-8d95-90970915a7f5" class="numbered-list" start="56"><li>Weiel, M. et al. Minimal dataset repository for reproduction of presented results (FLAPS-NMI@Github/FLAPS-sim_setups, 2021); <a href="https://doi.org/">https://doi.org/</a>10.5281/zenodo.4773999, <a href="https://github.com/FLAPS-NMI/FLAPS-sim_setups/">https://github.com/FLAPS-NMI/FLAPS-sim_setups/</a>releases/tag/v1.0</li></ol><ol type="1" id="5cb38a41-618f-4245-b846-427c50734d0c" class="numbered-list" start="57"><li>The PyMOL Molecular Graphics System, Version 1.8 (SchrÃ¶dinger, 2015).</li></ol><p id="aceb260a-2c23-4196-96b6-72766b8e3ecb" class="">
</p></div></article></body></html>